diff --git a/src/Corrade/Utility/TweakableParser.cpp b/src/Corrade/Utility/TweakableParser.cpp
--- a/src/Corrade/Utility/TweakableParser.cpp
+++ b/src/Corrade/Utility/TweakableParser.cpp
@@ -37,12 +37,12 @@ namespace Corrade { namespace Utility {
 namespace {
     std::pair<const char*, int> integerBase(Containers::ArrayView<const char> value) {
         if(String::viewBeginsWith(value, "0x") || String::viewBeginsWith(value, "0X"))
-            return {value + 2, 16};
+            return {value.data() + 2, 16};
         if(String::viewBeginsWith(value, "0b") || String::viewBeginsWith(value, "0B"))
-            return {value + 2, 2};
+            return {value.data() + 2, 2};
         if(String::viewBeginsWith(value, "0"))
-            return {value + 1, 8};
-        return {value, 10};
+            return {value.data() + 1, 8};
+        return {value.data(), 10};
     }
 }
 
