--- 3rdparty/dawn/src/tint/lang/core/ir/transform/multiplanar_external_texture.cc
+++ 3rdparty/dawn/src/tint/lang/core/ir/transform/multiplanar_external_texture.cc
@@ -46,6 +46,9 @@
     using Ts::operator()...;
 };
 
+template<class... Ts>
+overloaded(Ts...) -> overloaded<Ts...>;
+
 using MultiplanarTexture = tint::transform::multiplanar::MultiplanarTexture;
 using YCBCRTexture = tint::transform::multiplanar::YCBCRTexture;
 
