diff --git a/cmake/dependencies.cmake b/cmake/dependencies.cmake
index d06a276..c138d44 100644
--- a/cmake/dependencies.cmake
+++ b/cmake/dependencies.cmake
@@ -134,7 +134,14 @@ if(USE_HDF5)
   #####
   # First, find the C and HL libraries.
   #####
-  find_package(HDF5 COMPONENTS C HL REQUIRED)
+  find_package(HDF5 NO_MODULE COMPONENTS C HL REQUIRED)
+  if(TARGET hdf5::hdf5-static)
+    add_library(HDF5::HDF5 ALIAS hdf5::hdf5-static)
+    add_library(hdf5::hdf5_hl ALIAS hdf5::hdf5_hl-static)
+  else()
+    add_library(HDF5::HDF5 ALIAS hdf5::hdf5-shared)
+    add_library(hdf5::hdf5_hl ALIAS hdf5::hdf5_hl-shared)
+  endif()
 
   message(STATUS "Found HDF5 version: ${HDF5_VERSION}")
   if(${HDF5_VERSION} VERSION_LESS ${HDF5_VERSION_REQUIRED})
@@ -173,6 +180,13 @@ if(USE_HDF5)
     message(STATUS "HDF5 has zlib.")
   endif()
 
+  if(NOT TARGET ZLIB::ZLIB)
+    add_library(ZLIB::ZLIB UNKNOWN IMPORTED)
+    set_target_properties(ZLIB::ZLIB PROPERTIES
+      IMPORTED_LOCATION "${ZLIB_LIBRARY}"
+      INTERFACE_INCLUDE_DIRECTORIES "${ZLIB_INCLUDE_DIR}"
+    )
+  endif()
 
   # Check to see if H5Z_SZIP exists in HDF5_Libraries. If so, we must use szip library.
   check_hdf5_feature(HAVE_H5Z_SZIP H5_HAVE_FILTER_SZIP)
