diff --git a/cmake/config-ix.cmake b/cmake/config-ix.cmake
index f58d48253..10733561e 100644
--- a/cmake/config-ix.cmake
+++ b/cmake/config-ix.cmake
@@ -519,7 +519,7 @@ else()
   find_program(GO_EXECUTABLE NAMES go DOC "go executable")
   if(GO_EXECUTABLE STREQUAL "GO_EXECUTABLE-NOTFOUND")
     message(STATUS "Go bindings disabled.")
-  else()
+  elseif(EXISTS ${CMAKE_SOURCE_DIR}/bindings/go/conftest.go)
     execute_process(COMMAND ${GO_EXECUTABLE} run ${CMAKE_SOURCE_DIR}/bindings/go/conftest.go
                     RESULT_VARIABLE GO_CONFTEST)
     if(GO_CONFTEST STREQUAL "0")
@@ -528,6 +528,8 @@ else()
     else()
       message(STATUS "Go bindings disabled, need at least Go 1.2.")
     endif()
+  else()
+    message(STATUS "Missing ${CMAKE_SOURCE_DIR}/bindings/go/conftest.go, Go bindings disabled.")
   endif()
 endif()
 
