diff --git a/ThirdParty/CMakeLists.txt b/ThirdParty/CMakeLists.txt
index 5486b48..9dfa47e 100644
--- a/ThirdParty/CMakeLists.txt
+++ b/ThirdParty/CMakeLists.txt
@@ -10,6 +10,10 @@ include(FindPkgConfig)
 
 if (GL_SUPPORTED AND (PLATFORM_WIN32 OR PLATFORM_LINUX OR PLATFORM_MACOS) AND (NOT glew_FOUND))
     pkg_check_modules(glew REQUIRED glew)
 endif()
+if (PLATFORM_LINUX AND glew_FOUND)
+    set(glew_LIBRARIES ${glew_STATIC_LIBRARIES} CACHE INTERNAL "" FORCE)
+    set(glew_LIBRARY_DIRS ${glew_STATIC_LIBRARY_DIRS} CACHE INTERNAL "" FORCE)
+endif()
 
 if (VULKAN_SUPPORTED)
@@ -30,10 +34,18 @@ if (VULKAN_SUPPORTED OR METAL_SUPPORTED OR WEBGPU_SUPPORTED OR (ARCHIVER_SUPPORT
 
     if (NOT (${DILIGENT_NO_GLSLANG} AND ${DILIGENT_NO_HLSL}) AND (NOT SPIRV-Tools_FOUND))
         pkg_check_modules(SPIRV-Tools REQUIRED SPIRV-Tools)
     endif()
+    if (PLATFORM_LINUX AND SPIRV-Tools_FOUND)
+        set(SPIRV-Tools_LIBRARIES ${SPIRV-Tools_STATIC_LIBRARIES} CACHE INTERNAL "" FORCE)
+        set(SPIRV-Tools_LIBDIR ${SPIRV-Tools_STATIC_LIBRARY_DIRS} CACHE INTERNAL "" FORCE)
+    endif()
 
     if (NOT ${DILIGENT_NO_GLSLANG} AND (NOT glslang_FOUND))
         pkg_check_modules(glslang REQUIRED glslang)
     endif()
+    if (PLATFORM_LINUX AND glslang_FOUND)
+        set(glslang_LIBRARIES ${glslang_STATIC_LIBRARIES} CACHE INTERNAL "" FORCE)
+        set(glslang_LIBDIR ${glslang_STATIC_LIBRARY_DIRS} CACHE INTERNAL "" FORCE)
+    endif()
 
     if (NOT spirv-cross-c_FOUND)
diff --git a/Graphics/Archiver/CMakeLists.txt b/Graphics/Archiver/CMakeLists.txt
--- a/Graphics/Archiver/CMakeLists.txt
+++ b/Graphics/Archiver/CMakeLists.txt
@@ -150,7 +150,8 @@ if(GL_SUPPORTED OR GLES_SUPPORTED)
-        target_link_libraries(Diligent-Archiver-static PRIVATE ${glew_LIBRARIES})
+        target_link_libraries(Diligent-Archiver-static PRIVATE ${glew_LIBRARIES} opengl32)
     elseif(PLATFORM_LINUX)
         find_package(X11 REQUIRED)
+        find_package(OpenGL REQUIRED)
         target_link_directories(Diligent-Archiver-static PRIVATE ${glew_LIBRARY_DIRS})
-        target_link_libraries(Diligent-Archiver-static PRIVATE ${glew_LIBRARIES} X11::X11)
+        target_link_libraries(Diligent-Archiver-static PRIVATE ${glew_LIBRARIES} OpenGL::GL X11::X11)
     elseif(PLATFORM_MACOS)
         target_link_directories(Diligent-Archiver-static PRIVATE ${glew_LIBRARY_DIRS})
diff --git a/Graphics/GraphicsEngineOpenGL/CMakeLists.txt b/Graphics/GraphicsEngineOpenGL/CMakeLists.txt
--- a/Graphics/GraphicsEngineOpenGL/CMakeLists.txt
+++ b/Graphics/GraphicsEngineOpenGL/CMakeLists.txt
@@ -203,9 +203,10 @@ if(PLATFORM_WIN32)
-    set(PRIVATE_DEPENDENCIES ${PRIVATE_DEPENDENCIES} ${glew_LIBRARIES})
+    set(PRIVATE_DEPENDENCIES ${PRIVATE_DEPENDENCIES} ${glew_LIBRARIES} opengl32)
 elseif(PLATFORM_ANDROID)
     set(PRIVATE_DEPENDENCIES ${PRIVATE_DEPENDENCIES} GLESv3 EGL)
 elseif(PLATFORM_LINUX)
     find_package(X11 REQUIRED)
+    find_package(OpenGL REQUIRED)
     target_link_directories(Diligent-GraphicsEngineOpenGL-static PRIVATE ${glew_LIBRARY_DIRS})
-    set(PRIVATE_DEPENDENCIES ${PRIVATE_DEPENDENCIES} ${glew_LIBRARIES} X11::X11)
+    set(PRIVATE_DEPENDENCIES ${PRIVATE_DEPENDENCIES} ${glew_LIBRARIES} OpenGL::GL X11::X11)
 elseif(PLATFORM_MACOS)
     target_link_directories(Diligent-GraphicsEngineOpenGL-static PRIVATE ${glew_LIBRARY_DIRS})
diff --git a/Graphics/ShaderTools/CMakeLists.txt b/Graphics/ShaderTools/CMakeLists.txt
index 42bf3be..9057f7c 100644
--- a/Graphics/ShaderTools/CMakeLists.txt
+++ b/Graphics/ShaderTools/CMakeLists.txt
@@ -186,7 +186,7 @@ if(ENABLE_SPIRV)
     if (${USE_SPIRV_TOOLS})
         target_link_libraries(Diligent-ShaderTools
         PRIVATE
-            SPIRV-Tools-opt
+            ${SPIRV-Tools_LIBRARIES}
         )
         target_compile_definitions(Diligent-ShaderTools PRIVATE USE_SPIRV_TOOLS=1)
     endif()
@@ -194,8 +194,7 @@ if(ENABLE_SPIRV)
     if (${USE_GLSLANG})
         target_link_libraries(Diligent-ShaderTools
         PRIVATE
-            SPIRV
-            glslang
+            ${glslang_LIBRARIES}
         )
     endif()
 endif()
