diff --git a/BuildTools/CMake/BuildUtils.cmake b/BuildTools/CMake/BuildUtils.cmake
index ee394acab..da9d7b69e 100644
--- a/BuildTools/CMake/BuildUtils.cmake
+++ b/BuildTools/CMake/BuildUtils.cmake
@@ -335,14 +335,19 @@ function(install_core_lib _TARGET)
     if(TARGET_TYPE STREQUAL STATIC_LIBRARY)
         list(APPEND DILIGENT_CORE_INSTALL_LIBS_LIST ${_TARGET})
         set(DILIGENT_CORE_INSTALL_LIBS_LIST ${DILIGENT_CORE_INSTALL_LIBS_LIST} CACHE INTERNAL "Core libraries installation list")
+        install(TARGETS                 ${_TARGET}
+                ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+                LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+                RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
+        )
     elseif(TARGET_TYPE STREQUAL SHARED_LIBRARY)
         install(TARGETS                 ${_TARGET}
-                ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}/${DILIGENT_CORE_DIR}/$<CONFIG>"
-                LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}/${DILIGENT_CORE_DIR}/$<CONFIG>"
-                RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}/${DILIGENT_CORE_DIR}/$<CONFIG>"
+                ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+                LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+                RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
         )
         if (DILIGENT_INSTALL_PDB)
-            install(FILES $<TARGET_PDB_FILE:${_TARGET}> DESTINATION "${CMAKE_INSTALL_BINDIR}/${DILIGENT_CORE_DIR}/$<CONFIG>" OPTIONAL)
+            install(FILES $<TARGET_PDB_FILE:${_TARGET}> DESTINATION "${CMAKE_INSTALL_BINDIR}" OPTIONAL)
         endif()
     endif()
 
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 37baf199b..caf3d0867 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -595,7 +595,6 @@ add_subdirectory(Primitives)
 add_subdirectory(Platforms)
 add_subdirectory(Common)
 add_subdirectory(Graphics)
-add_subdirectory(Tests)
 
 
 # Installation instructions
@@ -606,7 +605,7 @@ if(DILIGENT_INSTALL_CORE)
         "${DILIGENT_CORE_INSTALL_LIBS_LIST}"
         DiligentCore-static                     # Custom target name
         DiligentCore                            # Folder
-        "${CMAKE_INSTALL_LIBDIR}/${DILIGENT_CORE_DIR}/$<CONFIG>"    # Install destination
+        "${CMAKE_INSTALL_LIBDIR}"    # Install destination
     )
 
     install(FILES License.txt DESTINATION "Licenses" RENAME DiligentEngine-License.txt)
