diff --git a/CMakeLists.txt b/CMakeLists.txt
index 92f7114..a56a7b1 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -306,11 +306,10 @@ if(WITH_BORINGSSL)
     list(APPEND DYNAMIC_LIB ${BORINGSSL_SSL_LIBRARY})
     list(APPEND DYNAMIC_LIB ${BORINGSSL_CRYPTO_LIBRARY})
 else()
-    list(APPEND DYNAMIC_LIB ${OPENSSL_CRYPTO_LIBRARY})
     if(WITH_MESALINK)
         list(APPEND DYNAMIC_LIB ${MESALINK_LIB})
     else()
-        list(APPEND DYNAMIC_LIB ${OPENSSL_SSL_LIBRARY})
+        list(APPEND DYNAMIC_LIB ${OPENSSL_SSL_LIBRARY} ${OPENSSL_CRYPTO_LIBRARY})
     endif()
 endif()
 
