diff --git a/include/webdriverxx.h b/include/webdriverxx.h
index 6b9627f..5ce2bfd 100644
--- a/include/webdriverxx.h
+++ b/include/webdriverxx.h
@@ -5,6 +5,7 @@
 #include "webdriverxx/browsers/chrome.h"
 #include "webdriverxx/browsers/firefox.h"
 #include "webdriverxx/browsers/ie.h"
+#include "webdriverxx/browsers/phantom.h"
 #include "webdriverxx/wait.h"
 #include "webdriverxx/wait_match.h"
 
diff --git a/include/webdriverxx/browsers/phantom.h b/include/webdriverxx/browsers/phantom.h
new file mode 100644
index 0000000..8223e4e
--- /dev/null
+++ b/include/webdriverxx/browsers/phantom.h
@@ -0,0 +1,24 @@
+#ifndef WEBDRIVERXX_BROWSERS_PHANTOM_H
+#define WEBDRIVERXX_BROWSERS_PHANTOM_H
+
+#include "../capabilities.h"
+
+namespace webdriverxx {
+
+struct Phantom : Capabilities { // copyable
+    Phantom(const Capabilities& defaults = Capabilities())
+		: Capabilities(defaults) {
+		SetBrowserName(browser::Phantom);
+		SetVersion("");
+		SetPlatform(platform::Any);
+	}
+
+	WEBDRIVERXX_PROPERTIES_BEGIN(Phantom)
+	// Profile is a profile folder, zipped and base64 encoded.
+	WEBDRIVERXX_PROPERTY(LoggingPrefs, "loggingPrefs", LoggingPrefs)
+	WEBDRIVERXX_PROPERTIES_END()
+};
+
+} // namespace webdriverxx
+
+#endif
