diff --git a/include/webdriverxx/detail/types.h b/include/webdriverxx/detail/types.h
index 544a069..bad0d24 100644
--- a/include/webdriverxx/detail/types.h
+++ b/include/webdriverxx/detail/types.h
@@ -29,6 +29,24 @@ inline
 void CustomFromJson(const picojson::value& value, ElementRef& result) {
 	WEBDRIVERXX_CHECK(value.is<picojson::object>(), "ElementRef is not an object");
 	result.ref = FromJson<std::string>(value.get("ELEMENT"));
+	if (result.ref == "null") {
+		std::stringstream element(value.serialize());
+		std::string segment;
+		std::vector<std::string> strArr;
+		while (std::getline(element, segment, ':')) {
+			segment.erase(std::remove(segment.begin(), segment.end(), '{'), segment.end());
+			segment.erase(std::remove(segment.begin(), segment.end(), '"'), segment.end());
+			segment.erase(std::remove(segment.begin(), segment.end(), '}'), segment.end());
+			strArr.push_back(segment);
+		}
+		if (strArr.size() >= 1) {
+			std::stringstream ss;
+			ss << "{\"ELEMENT\":" << "\"" << strArr[1] << "\"}";
+			picojson::value jSon;
+			picojson::parse(jSon, ss);
+			result.ref = FromJson<std::string>(jSon.get("ELEMENT"));
+		}
+	}
 }
 
 inline
