diff --git a/src/hb.hh b/src/hb.hh
index 052cabed9f5..07e68886de5 100644
--- a/src/hb.hh
+++ b/src/hb.hh
@@ -179,6 +179,9 @@
 #include "hb-aat.h"
 #define HB_AAT_H_IN
 
+#if 0
+#  For Android & SDL, it doesn't compile with ndk-r21d
+#  https://github.com/harfbuzz/harfbuzz/commit/7cb22ba7ebf6ef053790a3201cb7014bacd51e46
 #include <cassert>
 #include <cfloat>
 #include <climits>
@@ -188,6 +191,17 @@
 #include <cstdio>
 #include <cstdlib>
 #include <cstring>
+#else
+#include <limits.h>
+#include <math.h>
+#include <float.h>
+#include <stdlib.h>
+#include <stddef.h>
+#include <string.h>
+#include <assert.h>
+#include <stdio.h>
+#include <stdarg.h>
+#endif
 
 #if (defined(_MSC_VER) && _MSC_VER >= 1500) || defined(__MINGW32__)
 #ifdef __MINGW32_VERSION
@@ -398,8 +412,14 @@ extern "C" void  hb_free_impl(void *ptr);
 #endif
 
 #ifndef HB_NO_ERRNO
+#if 0
+#  For Android & SDL, it doesn't compile with ndk-r21d
+#  https://github.com/harfbuzz/harfbuzz/commit/7cb22ba7ebf6ef053790a3201cb7014bacd51e46
 #  include <cerrno>
 #else
+#  include <errno.h>
+#endif
+#else
 static int HB_UNUSED _hb_errno = 0;
 #  undef errno
 #  define errno _hb_errno
