diff --git a/BuildUtils.cmake b/BuildUtils.cmake
index 0dcdf1e..60028c2 100644
--- a/BuildUtils.cmake
+++ b/BuildUtils.cmake
@@ -1,6 +1,7 @@
 
 # Converts shaders to headers and generates master header with the list of all files
 function(convert_shaders_to_headers _SHADERS _SHADER_OUTPUT_DIR _SHADERS_LIST_FILE _SHADERS_INC_LIST)
+    set(FILE2STRING_PATH "${CMAKE_CURRENT_SOURCE_DIR}/script.py" CACHE INTERNAL "File2String utility")
     if(NOT FILE2STRING_PATH STREQUAL "")
         find_package(Python3 REQUIRED)
 
diff --git a/CMakeLists.txt b/CMakeLists.txt
index d6a9bb2..7b2690c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,6 +1,46 @@
 cmake_minimum_required (VERSION 3.6)
 
 include(BuildUtils.cmake)
+include(DiligentCoreBuildUtils.cmake)
+
+add_library(Diligent-PublicBuildSettings INTERFACE)
+if(PLATFORM_WIN32)
+    set(GL_SUPPORTED       TRUE CACHE INTERNAL "OpenGL is supported on Win32 platform")
+    set(VULKAN_SUPPORTED   TRUE CACHE INTERNAL "Vulkan is supported on Win32 platform")
+    set(WEBGPU_SUPPORTED   TRUE CACHE INTERNAL "WebGPU is supported on Win32 platform")
+    set(ARCHIVER_SUPPORTED TRUE CACHE INTERNAL "Archiver is supported on Win32 platform")
+    target_compile_definitions(Diligent-PublicBuildSettings INTERFACE PLATFORM_WIN32=1)
+elseif(PLATFORM_UNIVERSAL_WINDOWS)
+    set(ARCHIVER_SUPPORTED TRUE CACHE INTERNAL "Archiver is supported on Universal Windows platform")
+    target_compile_definitions(Diligent-PublicBuildSettings INTERFACE PLATFORM_UNIVERSAL_WINDOWS=1)
+elseif(PLATFORM_ANDROID)
+    set(GLES_SUPPORTED     TRUE CACHE INTERNAL "OpenGLES is supported on Android platform")
+    set(VULKAN_SUPPORTED   TRUE CACHE INTERNAL "Vulkan is supported on Android platform")
+    set(ARCHIVER_SUPPORTED TRUE CACHE INTERNAL "Archiver is supported on Android platform")
+    target_compile_definitions(Diligent-PublicBuildSettings INTERFACE PLATFORM_ANDROID=1)
+elseif(PLATFORM_LINUX)
+    set(GL_SUPPORTED       TRUE CACHE INTERNAL "OpenGL is supported on Linux platform")
+    set(VULKAN_SUPPORTED   TRUE CACHE INTERNAL "Vulkan is supported on Linux platform")
+    set(ARCHIVER_SUPPORTED TRUE CACHE INTERNAL "Archiver is supported on Linux platform")
+    target_compile_definitions(Diligent-PublicBuildSettings INTERFACE PLATFORM_LINUX=1)
+elseif(PLATFORM_MACOS)
+    set(GL_SUPPORTED       TRUE CACHE INTERNAL "OpenGL is supported on MacOS platform")
+    set(VULKAN_SUPPORTED   TRUE CACHE INTERNAL "Vulkan is enabled through MoltenVK on MacOS platform")
+    set(ARCHIVER_SUPPORTED TRUE CACHE INTERNAL "Archiver is supported on MacOS platform")
+    target_compile_definitions(Diligent-PublicBuildSettings INTERFACE PLATFORM_MACOS=1 PLATFORM_APPLE=1)
+elseif(PLATFORM_IOS)
+    set(GLES_SUPPORTED TRUE CACHE INTERNAL "OpenGLES is supported on iOS platform")
+    target_compile_definitions(Diligent-PublicBuildSettings INTERFACE PLATFORM_IOS=1 PLATFORM_APPLE=1)
+elseif(PLATFORM_TVOS)
+    target_compile_definitions(Diligent-PublicBuildSettings INTERFACE PLATFORM_TVOS=1 PLATFORM_APPLE=1)
+elseif(PLATFORM_WEB)
+    set(GLES_SUPPORTED     TRUE CACHE INTERNAL "OpenGLES is supported on Web platform")
+    set(WEBGPU_SUPPORTED   TRUE CACHE INTERNAL "WebGPU is supported on Web platform")
+    set(ARCHIVER_SUPPORTED TRUE CACHE INTERNAL "Archiver is supported on Web platform")
+    target_compile_definitions(Diligent-PublicBuildSettings INTERFACE PLATFORM_WEB=1 PLATFORM_EMSCRIPTEN=1)
+else()
+    message(FATAL_ERROR "No PLATFORM_XXX variable defined. Make sure that 'DiligentCore' folder is processed first")
+endif()
 
 set_property(GLOBAL PROPERTY USE_FOLDERS ON)
 
@@ -17,10 +57,17 @@ if(PLATFORM_WIN32 OR PLATFORM_UNIVERSAL_WINDOWS OR PLATFORM_LINUX OR PLATFORM_MA
 else()
     set(DILIGENT_INSTALL_FX OFF)
 endif()
+include(FindPkgConfig)
+pkg_check_modules(diligentcore REQUIRED diligentcore)
+pkg_check_modules(diligenttools REQUIRED diligenttools)
+pkg_check_modules(imgui REQUIRED imgui)
+target_link_directories(DiligentFX PUBLIC ${diligentcore_LIBDIR} ${diligenttools_LIBDIR})
 
 target_link_libraries(DiligentFX 
 PRIVATE
+    Diligent-PublicBuildSettings
     Diligent-BuildSettings
+    ${imgui_LINK_LIBRARIES}
 PUBLIC
     Diligent-GraphicsEngine
     Diligent-GraphicsTools
@@ -29,7 +76,8 @@ PUBLIC
     Diligent-Imgui
 )
 
-if (TARGET Diligent-HLSL2GLSLConverterLib)
+if (GL_SUPPORTED OR GLES_SUPPORTED OR VULKAN_SUPPORTED)
+    message(STATUS "Building with HLSL2GLSL_CONVERTER_SUPPORTED=1")
     target_link_libraries(DiligentFX PRIVATE Diligent-HLSL2GLSLConverterLib)
     target_compile_definitions(DiligentFX PRIVATE HLSL2GLSL_CONVERTER_SUPPORTED=1)
 endif()
@@ -39,6 +87,9 @@ set_common_target_properties(DiligentFX)
 target_include_directories(DiligentFX
 PUBLIC
     .
+    ${diligentcore_INCLUDEDIR}
+    ${diligenttools_INCLUDEDIR}
+    ${imgui_INCLUDEDIR}
 )
 
 add_subdirectory(Utilities)
@@ -50,7 +101,6 @@ if(DILIGENT_USD_PATH)
     add_subdirectory(Hydrogent)
 endif()
 
-add_subdirectory(Tests)
 
 get_target_property(SOURCE DiligentFX SOURCES)
 
@@ -86,35 +136,35 @@ source_group("generated" FILES
     ${SHADERS_INC_LIST}
 )
 
-if(DILIGENT_INSTALL_FX)
+if(1)
     install(TARGETS				 DiligentFX
-            ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}/${DILIGENT_FX_DIR}/$<CONFIG>"
-            LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}/${DILIGENT_FX_DIR}/$<CONFIG>"
-            RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}/${DILIGENT_FX_DIR}/$<CONFIG>"
+            ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+            LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+            RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
     )
-    install(DIRECTORY    PostProcess/EpipolarLightScattering/interface
-            DESTINATION  "${CMAKE_INSTALL_INCLUDEDIR}/${DILIGENT_FX_DIR}/PostProcess/EpipolarLightScattering"
+    install(DIRECTORY    "${CMAKE_CURRENT_SOURCE_DIR}/PostProcess/EpipolarLightScattering/interface"
+            DESTINATION  "include/DiligentFX/PostProcess/EpipolarLightScattering"
     )
-    install(DIRECTORY    PostProcess/TemporalAntiAliasing/interface
-            DESTINATION  "${CMAKE_INSTALL_INCLUDEDIR}/${DILIGENT_FX_DIR}/PostProcess/TemporalAntiAliasing"
+    install(DIRECTORY    "${CMAKE_CURRENT_SOURCE_DIR}/PostProcess/TemporalAntiAliasing/interface"
+            DESTINATION  "include/DiligentFX/PostProcess/TemporalAntiAliasing"
     )
-    install(DIRECTORY    PostProcess/ScreenSpaceReflection/interface
-            DESTINATION  "${CMAKE_INSTALL_INCLUDEDIR}/${DILIGENT_FX_DIR}/PostProcess/ScreenSpaceReflection"
+    install(DIRECTORY    "${CMAKE_CURRENT_SOURCE_DIR}/PostProcess/ScreenSpaceReflection/interface"
+            DESTINATION  "include/DiligentFX/PostProcess/ScreenSpaceReflection"
     )
-    install(DIRECTORY   PostProcess/ScreenSpaceAmbientOcclusion/interface
-            DESTINATION  "${CMAKE_INSTALL_INCLUDEDIR}/${DILIGENT_FX_DIR}/PostProcess/ScreenSpaceAmbientOcclusion"
+    install(DIRECTORY   "${CMAKE_CURRENT_SOURCE_DIR}/PostProcess/ScreenSpaceAmbientOcclusion/interface"
+            DESTINATION  "include/DiligentFX/PostProcess/ScreenSpaceAmbientOcclusion"
     )
-    install(DIRECTORY   PostProcess/Bloom/interface
-            DESTINATION  "${CMAKE_INSTALL_INCLUDEDIR}/${DILIGENT_FX_DIR}/PostProcess/Bloom"
+    install(DIRECTORY   "${CMAKE_CURRENT_SOURCE_DIR}/PostProcess/Bloom/interface"
+            DESTINATION  "include/DiligentFX/PostProcess/Bloom"
     )
-    install(DIRECTORY    Components/interface
-            DESTINATION  "${CMAKE_INSTALL_INCLUDEDIR}/${DILIGENT_FX_DIR}/Components"
+    install(DIRECTORY    "${CMAKE_CURRENT_SOURCE_DIR}/Components/interface"
+            DESTINATION  "include/DiligentFX/Components"
     )
-    install(DIRECTORY    PBR/interface
-            DESTINATION  "${CMAKE_INSTALL_INCLUDEDIR}/${DILIGENT_FX_DIR}/PBR"
+    install(DIRECTORY    "${CMAKE_CURRENT_SOURCE_DIR}/PBR/interface"
+            DESTINATION  "include/DiligentFX/PBR"
     )
     install(DIRECTORY    Shaders
-            DESTINATION  "."
+            DESTINATION  "include/DiligentFX"
             FILES_MATCHING PATTERN "public/*.*"
             PATTERN "private" EXCLUDE
     )
@@ -125,4 +175,3 @@ set_target_properties(DiligentFX PROPERTIES
 )
 
 # Create a custom target to run source code formatting validation command
-add_format_validation_target(DiligentFX "${CMAKE_CURRENT_SOURCE_DIR}" DiligentFX)
diff --git a/Components/interface/BoundBoxRenderer.hpp b/Components/interface/BoundBoxRenderer.hpp
index 039f569..1ed4419 100644
--- a/Components/interface/BoundBoxRenderer.hpp
+++ b/Components/interface/BoundBoxRenderer.hpp
@@ -28,12 +28,12 @@
 #include <vector>
 #include <memory>
 
-#include "../../../DiligentCore/Graphics/GraphicsEngine/interface/RenderDevice.h"
-#include "../../../DiligentCore/Graphics/GraphicsEngine/interface/DeviceContext.h"
-#include "../../../DiligentCore/Graphics/GraphicsTools/interface/RenderStateCache.h"
-#include "../../../DiligentCore/Common/interface/RefCntAutoPtr.hpp"
-#include "../../../DiligentCore/Common/interface/HashUtils.hpp"
-#include "../../../DiligentCore/Common/interface/BasicMath.hpp"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/RenderDevice.h"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/DeviceContext.h"
+#include "DiligentCore/Graphics/GraphicsTools/interface/RenderStateCache.h"
+#include "DiligentCore/Common/interface/RefCntAutoPtr.hpp"
+#include "DiligentCore/Common/interface/HashUtils.hpp"
+#include "DiligentCore/Common/interface/BasicMath.hpp"
 
 namespace Diligent
 {
diff --git a/Components/interface/CoordinateGridRenderer.hpp b/Components/interface/CoordinateGridRenderer.hpp
index f02cb29..0ab3557 100644
--- a/Components/interface/CoordinateGridRenderer.hpp
+++ b/Components/interface/CoordinateGridRenderer.hpp
@@ -29,14 +29,14 @@
 #include <unordered_map>
 #include <vector>
 
-#include "../../../DiligentCore/Graphics/GraphicsEngine/interface/RenderDevice.h"
-#include "../../../DiligentCore/Graphics/GraphicsEngine/interface/DeviceContext.h"
-#include "../../../DiligentCore/Graphics/GraphicsTools/interface/RenderStateCache.h"
-#include "../../../DiligentCore/Graphics/GraphicsTools/interface/ResourceRegistry.hpp"
-#include "../../../DiligentCore/Graphics/GraphicsTools/interface/ShaderMacroHelper.hpp"
-#include "../../../DiligentCore/Common/interface/RefCntAutoPtr.hpp"
-#include "../../../DiligentCore/Common/interface/BasicMath.hpp"
-#include "../../../DiligentCore/Common/interface/HashUtils.hpp"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/RenderDevice.h"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/DeviceContext.h"
+#include "DiligentCore/Graphics/GraphicsTools/interface/RenderStateCache.h"
+#include "DiligentCore/Graphics/GraphicsTools/interface/ResourceRegistry.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/ShaderMacroHelper.hpp"
+#include "DiligentCore/Common/interface/RefCntAutoPtr.hpp"
+#include "DiligentCore/Common/interface/BasicMath.hpp"
+#include "DiligentCore/Common/interface/HashUtils.hpp"
 
 namespace Diligent
 {
diff --git a/Components/interface/EnvMapRenderer.hpp b/Components/interface/EnvMapRenderer.hpp
index e4e61f4..18815ce 100644
--- a/Components/interface/EnvMapRenderer.hpp
+++ b/Components/interface/EnvMapRenderer.hpp
@@ -29,12 +29,12 @@
 #include <memory>
 #include <array>
 
-#include "../../../DiligentCore/Graphics/GraphicsEngine/interface/RenderDevice.h"
-#include "../../../DiligentCore/Graphics/GraphicsEngine/interface/DeviceContext.h"
-#include "../../../DiligentCore/Graphics/GraphicsTools/interface/RenderStateCache.h"
-#include "../../../DiligentCore/Common/interface/RefCntAutoPtr.hpp"
-#include "../../../DiligentCore/Common/interface/BasicMath.hpp"
-#include "../../../DiligentCore/Common/interface/HashUtils.hpp"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/RenderDevice.h"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/DeviceContext.h"
+#include "DiligentCore/Graphics/GraphicsTools/interface/RenderStateCache.h"
+#include "DiligentCore/Common/interface/RefCntAutoPtr.hpp"
+#include "DiligentCore/Common/interface/BasicMath.hpp"
+#include "DiligentCore/Common/interface/HashUtils.hpp"
 
 namespace Diligent
 {
diff --git a/Components/interface/GBuffer.hpp b/Components/interface/GBuffer.hpp
index 4aa8dd0..a235bdc 100644
--- a/Components/interface/GBuffer.hpp
+++ b/Components/interface/GBuffer.hpp
@@ -26,8 +26,8 @@
 
 #include <vector>
 
-#include "../../../DiligentCore/Graphics/GraphicsEngine/interface/RenderDevice.h"
-#include "../../../DiligentCore/Common/interface/RefCntAutoPtr.hpp"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/RenderDevice.h"
+#include "DiligentCore/Common/interface/RefCntAutoPtr.hpp"
 
 namespace Diligent
 {
diff --git a/Components/interface/ShadowMapManager.hpp b/Components/interface/ShadowMapManager.hpp
index 2897ad0..bde4dee 100644
--- a/Components/interface/ShadowMapManager.hpp
+++ b/Components/interface/ShadowMapManager.hpp
@@ -30,13 +30,13 @@
 #include <vector>
 #include <array>
 
-#include "../../../DiligentCore/Graphics/GraphicsEngine/interface/RenderDevice.h"
-#include "../../../DiligentCore/Graphics/GraphicsEngine/interface/DeviceContext.h"
-#include "../../../DiligentCore/Graphics/GraphicsEngine/interface/Texture.h"
-#include "../../../DiligentCore/Graphics/GraphicsEngine/interface/TextureView.h"
-#include "../../../DiligentCore/Graphics/GraphicsTools/interface/RenderStateCache.h"
-#include "../../../DiligentCore/Common/interface/RefCntAutoPtr.hpp"
-#include "../../../DiligentCore/Common/interface/BasicMath.hpp"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/RenderDevice.h"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/DeviceContext.h"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/Texture.h"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/TextureView.h"
+#include "DiligentCore/Graphics/GraphicsTools/interface/RenderStateCache.h"
+#include "DiligentCore/Common/interface/RefCntAutoPtr.hpp"
+#include "DiligentCore/Common/interface/BasicMath.hpp"
 
 namespace Diligent
 {
diff --git a/Components/interface/ToneMapping.hpp b/Components/interface/ToneMapping.hpp
index ff899c0..bf0e790 100644
--- a/Components/interface/ToneMapping.hpp
+++ b/Components/interface/ToneMapping.hpp
@@ -24,7 +24,7 @@
  *  of the possibility of such damages.
  */
 
-#include "../../../DiligentCore/Common/interface/BasicMath.hpp"
+#include "DiligentCore/Common/interface/BasicMath.hpp"
 
 namespace Diligent
 {
diff --git a/Components/interface/VectorFieldRenderer.hpp b/Components/interface/VectorFieldRenderer.hpp
index 68eb87a..6537408 100644
--- a/Components/interface/VectorFieldRenderer.hpp
+++ b/Components/interface/VectorFieldRenderer.hpp
@@ -27,12 +27,12 @@
 #include <unordered_map>
 #include <vector>
 
-#include "../../../DiligentCore/Graphics/GraphicsEngine/interface/RenderDevice.h"
-#include "../../../DiligentCore/Graphics/GraphicsEngine/interface/DeviceContext.h"
-#include "../../../DiligentCore/Graphics/GraphicsTools/interface/RenderStateCache.h"
-#include "../../../DiligentCore/Common/interface/RefCntAutoPtr.hpp"
-#include "../../../DiligentCore/Common/interface/BasicMath.hpp"
-#include "../../../DiligentCore/Common/interface/HashUtils.hpp"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/RenderDevice.h"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/DeviceContext.h"
+#include "DiligentCore/Graphics/GraphicsTools/interface/RenderStateCache.h"
+#include "DiligentCore/Common/interface/RefCntAutoPtr.hpp"
+#include "DiligentCore/Common/interface/BasicMath.hpp"
+#include "DiligentCore/Common/interface/HashUtils.hpp"
 
 namespace Diligent
 {
diff --git a/Components/src/BoundBoxRenderer.cpp b/Components/src/BoundBoxRenderer.cpp
index a4aa4cd..b1beecc 100644
--- a/Components/src/BoundBoxRenderer.cpp
+++ b/Components/src/BoundBoxRenderer.cpp
@@ -27,13 +27,13 @@
 #include "BoundBoxRenderer.hpp"
 
 #include "Utilities/interface/DiligentFXShaderSourceStreamFactory.hpp"
-#include "ShaderMacroHelper.hpp"
-#include "RenderStateCache.hpp"
-#include "CommonlyUsedStates.h"
-#include "MapHelper.hpp"
-#include "GraphicsUtilities.h"
-#include "ShaderSourceFactoryUtils.hpp"
-#include "GraphicsAccessories.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/ShaderMacroHelper.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/RenderStateCache.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/CommonlyUsedStates.h"
+#include "DiligentCore/Graphics/GraphicsTools/interface/MapHelper.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/GraphicsUtilities.h"
+#include "DiligentCore/Graphics/GraphicsTools/interface/ShaderSourceFactoryUtils.hpp"
+#include "DiligentCore/Graphics/GraphicsAccessories/interface/GraphicsAccessories.hpp"
 
 namespace Diligent
 {
diff --git a/Components/src/CoordinateGridRenderer.cpp b/Components/src/CoordinateGridRenderer.cpp
index ab7b833..1ca8c03 100644
--- a/Components/src/CoordinateGridRenderer.cpp
+++ b/Components/src/CoordinateGridRenderer.cpp
@@ -25,17 +25,17 @@
  */
 
 #include "imgui.h"
-#include "ImGuiUtils.hpp"
+#include "DiligentTools/Imgui/interface/ImGuiUtils.hpp"
 
 #include "CoordinateGridRenderer.hpp"
-#include "GraphicsUtilities.h"
-#include "ShaderMacroHelper.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/GraphicsUtilities.h"
+#include "DiligentCore/Graphics/GraphicsTools/interface/ShaderMacroHelper.hpp"
 #include "PostFXRenderTechnique.hpp"
-#include "GraphicsTypesX.hpp"
-#include "RenderStateCache.hpp"
-#include "CommonlyUsedStates.h"
-#include "MapHelper.hpp"
-#include "ScopedDebugGroup.hpp"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/GraphicsTypesX.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/RenderStateCache.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/CommonlyUsedStates.h"
+#include "DiligentCore/Graphics/GraphicsTools/interface/MapHelper.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/ScopedDebugGroup.hpp"
 
 namespace Diligent
 {
diff --git a/Components/src/EnvMapRenderer.cpp b/Components/src/EnvMapRenderer.cpp
index 762bcf1..7b069e1 100644
--- a/Components/src/EnvMapRenderer.cpp
+++ b/Components/src/EnvMapRenderer.cpp
@@ -27,12 +27,12 @@
 #include "EnvMapRenderer.hpp"
 
 #include "Utilities/interface/DiligentFXShaderSourceStreamFactory.hpp"
-#include "ShaderMacroHelper.hpp"
-#include "RenderStateCache.hpp"
-#include "CommonlyUsedStates.h"
-#include "MapHelper.hpp"
-#include "GraphicsUtilities.h"
-#include "ShaderSourceFactoryUtils.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/ShaderMacroHelper.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/RenderStateCache.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/CommonlyUsedStates.h"
+#include "DiligentCore/Graphics/GraphicsTools/interface/MapHelper.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/GraphicsUtilities.h"
+#include "DiligentCore/Graphics/GraphicsTools/interface/ShaderSourceFactoryUtils.hpp"
 
 namespace Diligent
 {
diff --git a/Components/src/GBuffer.cpp b/Components/src/GBuffer.cpp
index a5b5493..d20c8de 100644
--- a/Components/src/GBuffer.cpp
+++ b/Components/src/GBuffer.cpp
@@ -26,8 +26,8 @@
 
 #include "GBuffer.hpp"
 
-#include "GraphicsAccessories.hpp"
-#include "PlatformMisc.hpp"
+#include "DiligentCore/Graphics/GraphicsAccessories/interface/GraphicsAccessories.hpp"
+#include "DiligentCore/Platforms/interface/PlatformMisc.hpp"
 
 #include <array>
 
diff --git a/Components/src/ShadowMapManager.cpp b/Components/src/ShadowMapManager.cpp
index 06a9e2f..15317b0 100644
--- a/Components/src/ShadowMapManager.cpp
+++ b/Components/src/ShadowMapManager.cpp
@@ -28,13 +28,13 @@
 #include <cfloat>
 
 #include "ShadowMapManager.hpp"
-#include "AdvancedMath.hpp"
+#include "DiligentCore/Common/interface/AdvancedMath.hpp"
 #include "Utilities/interface/DiligentFXShaderSourceStreamFactory.hpp"
-#include "GraphicsUtilities.h"
-#include "MapHelper.hpp"
-#include "CommonlyUsedStates.h"
-#include "RenderStateCache.hpp"
-#include "GraphicsAccessories.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/GraphicsUtilities.h"
+#include "DiligentCore/Graphics/GraphicsTools/interface/MapHelper.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/CommonlyUsedStates.h"
+#include "DiligentCore/Graphics/GraphicsTools/interface/RenderStateCache.hpp"
+#include "DiligentCore/Graphics/GraphicsAccessories/interface/GraphicsAccessories.hpp"
 
 namespace Diligent
 {
diff --git a/Components/src/ToneMapping.cpp b/Components/src/ToneMapping.cpp
index 5751484..262cd75 100644
--- a/Components/src/ToneMapping.cpp
+++ b/Components/src/ToneMapping.cpp
@@ -25,7 +25,7 @@
  */
 
 #include "imgui.h"
-#include "ImGuiUtils.hpp"
+#include "DiligentTools/Imgui/interface/ImGuiUtils.hpp"
 
 #include "ToneMapping.hpp"
 
diff --git a/Components/src/VectorFieldRenderer.cpp b/Components/src/VectorFieldRenderer.cpp
index 8fdf164..8204ad7 100644
--- a/Components/src/VectorFieldRenderer.cpp
+++ b/Components/src/VectorFieldRenderer.cpp
@@ -27,12 +27,12 @@
 #include "VectorFieldRenderer.hpp"
 
 #include "Utilities/interface/DiligentFXShaderSourceStreamFactory.hpp"
-#include "ShaderMacroHelper.hpp"
-#include "RenderStateCache.hpp"
-#include "CommonlyUsedStates.h"
-#include "MapHelper.hpp"
-#include "GraphicsUtilities.h"
-#include "ShaderSourceFactoryUtils.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/ShaderMacroHelper.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/RenderStateCache.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/CommonlyUsedStates.h"
+#include "DiligentCore/Graphics/GraphicsTools/interface/MapHelper.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/GraphicsUtilities.h"
+#include "DiligentCore/Graphics/GraphicsTools/interface/ShaderSourceFactoryUtils.hpp"
 
 namespace Diligent
 {
diff --git a/Hydrogent/CMakeLists.txt b/Hydrogent/CMakeLists.txt
index d18254a..498ea58 100644
--- a/Hydrogent/CMakeLists.txt
+++ b/Hydrogent/CMakeLists.txt
@@ -2,16 +2,7 @@ cmake_minimum_required (VERSION 3.13)
 
 project(Diligent-Hydrogent CXX)
 
-include(FetchContent)
-
-message("Fetching entt repository...")
-FetchContent_Declare(
-    entt
-    GIT_REPOSITORY https://github.com/skypjack/entt
-    GIT_TAG        v3.13.1
-)
-FetchContent_MakeAvailable(entt)
-set_directory_root_folder(${entt_SOURCE_DIR} "DiligentFX/ThirdParty/entt")
+pkg_check_modules(entt REQUIRED entt)
 
 set(SOURCE
     src/HnMaterial.cpp
@@ -100,10 +91,9 @@ add_library(Diligent-Hydrogent STATIC ${HYDROGENT_SOURCE} ${SHADERS_LIST_FILE} $
 
 set_common_target_properties(Diligent-Hydrogent)
 
-add_library(USD-Libraries INTERFACE)
+pkg_check_modules(usd REQUIRED usd)
 
-if(${CMAKE_GENERATOR} MATCHES "Visual Studio" OR
-   ${CMAKE_GENERATOR} MATCHES "Xcode")
+if(0)
     # Since we can't use wildcards in generator expressions, we have to
     # find boost directory for each configuration
     file(GLOB BOOST_INCLUDE_DEBUG "${DILIGENT_USD_PATH}/Debug/include/boost*")
@@ -122,7 +112,6 @@ if(${CMAKE_GENERATOR} MATCHES "Visual Studio" OR
     INTERFACE
         ${DILIGENT_USD_PATH}/$<CONFIG>/lib
     )
-else()
     # Load the pxr package, which will bring all USD libraries
     set(pxr_DIR ${DILIGENT_USD_PATH})
     find_package(pxr)
@@ -166,7 +155,7 @@ set(USD_LIBS
     usdLux
 )
 
-if (NOT pxr_FOUND)
+if (0)
     # Add usd_ prefix to all USD libraries
     set(_USD_LIBS "")
     foreach(lib ${USD_LIBS})
@@ -192,7 +181,6 @@ if (NOT pxr_FOUND)
     target_link_libraries(USD-Libraries INTERFACE Python3::Python)
 endif()
 
-target_link_libraries(USD-Libraries INTERFACE ${USD_LIBS})
 
 target_include_directories(Diligent-Hydrogent
 PUBLIC
@@ -200,6 +188,8 @@ PUBLIC
 PRIVATE
     include
     ${SHADER_OUTPUT_DIR}
+    ${entt_INCLUDEDIR}
+    ${usd_INCLUDEDIR}
 )
 
 target_link_libraries(Diligent-Hydrogent
@@ -213,8 +203,8 @@ PRIVATE
     Diligent-Common
     DiligentFX
 PUBLIC
-    EnTT
-    USD-Libraries
+    ${entt_LINK_LIBRARIES}
+    ${usd_LINK_LIBRARIES}
 )
 
 set_target_properties(Diligent-Hydrogent PROPERTIES
diff --git a/Hydrogent/include/HnDrawItem.hpp b/Hydrogent/include/HnDrawItem.hpp
index f86083b..ba34486 100644
--- a/Hydrogent/include/HnDrawItem.hpp
+++ b/Hydrogent/include/HnDrawItem.hpp
@@ -35,7 +35,7 @@
 #include "Buffer.h"
 #include "PipelineState.h"
 #include "ShaderResourceBinding.h"
-#include "RefCntAutoPtr.hpp"
+#include "DiligentCore/Common/interface/RefCntAutoPtr.hpp"
 
 namespace Diligent
 {
diff --git a/Hydrogent/include/HnShaderSourceFactory.hpp b/Hydrogent/include/HnShaderSourceFactory.hpp
index 9aec7c5..84e0157 100644
--- a/Hydrogent/include/HnShaderSourceFactory.hpp
+++ b/Hydrogent/include/HnShaderSourceFactory.hpp
@@ -26,8 +26,8 @@
 
 #pragma once
 
-#include "../../../DiligentCore/Graphics/GraphicsEngine/interface/Shader.h"
-#include "../../../DiligentCore/Common/interface/RefCntAutoPtr.hpp"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/Shader.h"
+#include "DiligentCore/Common/interface/RefCntAutoPtr.hpp"
 
 namespace Diligent
 {
diff --git a/Hydrogent/include/HnShadowMapManager.hpp b/Hydrogent/include/HnShadowMapManager.hpp
index ec349b9..227dda1 100644
--- a/Hydrogent/include/HnShadowMapManager.hpp
+++ b/Hydrogent/include/HnShadowMapManager.hpp
@@ -30,7 +30,7 @@
 
 #include "RenderDevice.h"
 #include "DeviceContext.h"
-#include "RefCntAutoPtr.hpp"
+#include "DiligentCore/Common/interface/RefCntAutoPtr.hpp"
 
 namespace Diligent
 {
diff --git a/Hydrogent/include/HnTextureIdentifier.hpp b/Hydrogent/include/HnTextureIdentifier.hpp
index 8ea53fb..55984c7 100644
--- a/Hydrogent/include/HnTextureIdentifier.hpp
+++ b/Hydrogent/include/HnTextureIdentifier.hpp
@@ -30,7 +30,7 @@
 #include "pxr/base/tf/token.h"
 #include "pxr/imaging/hd/enums.h"
 
-#include "../../../DiligentCore/Graphics/GraphicsEngine/interface/TextureView.h"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/TextureView.h"
 
 
 namespace Diligent
diff --git a/Hydrogent/include/HnTextureUtils.hpp b/Hydrogent/include/HnTextureUtils.hpp
index aeb09d6..aba6dc5 100644
--- a/Hydrogent/include/HnTextureUtils.hpp
+++ b/Hydrogent/include/HnTextureUtils.hpp
@@ -27,7 +27,7 @@
 #pragma once
 
 #include "TextureLoader.h"
-#include "RefCntAutoPtr.hpp"
+#include "DiligentCore/Common/interface/RefCntAutoPtr.hpp"
 
 namespace Diligent
 {
diff --git a/Hydrogent/interface/GfTypeConversions.hpp b/Hydrogent/interface/GfTypeConversions.hpp
index bdce614..73050cb 100644
--- a/Hydrogent/interface/GfTypeConversions.hpp
+++ b/Hydrogent/interface/GfTypeConversions.hpp
@@ -26,8 +26,8 @@
 
 #pragma once
 
-#include "../../../DiligentCore/Common/interface/BasicMath.hpp"
-#include "../../../DiligentCore/Common/interface/AdvancedMath.hpp"
+#include "DiligentCore/Common/interface/BasicMath.hpp"
+#include "DiligentCore/Common/interface/AdvancedMath.hpp"
 
 #include "pxr/base/gf/vec2f.h"
 #include "pxr/base/gf/vec3f.h"
diff --git a/Hydrogent/interface/HnBuffer.hpp b/Hydrogent/interface/HnBuffer.hpp
index e482aff..894fd49 100644
--- a/Hydrogent/interface/HnBuffer.hpp
+++ b/Hydrogent/interface/HnBuffer.hpp
@@ -28,8 +28,8 @@
 
 #include "pxr/imaging/hd/bprim.h"
 
-#include "../../../DiligentCore/Graphics/GraphicsEngine/interface/Buffer.h"
-#include "../../../DiligentCore/Common/interface/RefCntAutoPtr.hpp"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/Buffer.h"
+#include "DiligentCore/Common/interface/RefCntAutoPtr.hpp"
 
 namespace Diligent
 {
diff --git a/Hydrogent/interface/HnCamera.hpp b/Hydrogent/interface/HnCamera.hpp
index b066fe6..ae717d5 100644
--- a/Hydrogent/interface/HnCamera.hpp
+++ b/Hydrogent/interface/HnCamera.hpp
@@ -28,7 +28,7 @@
 
 #include "pxr/imaging/hd/camera.h"
 
-#include "../../../../DiligentCore/Common/interface/BasicMath.hpp"
+#include "DiligentCore/Common/interface/BasicMath.hpp"
 
 namespace Diligent
 {
diff --git a/Hydrogent/interface/HnFrameRenderTargets.hpp b/Hydrogent/interface/HnFrameRenderTargets.hpp
index 8541940..c161ea7 100644
--- a/Hydrogent/interface/HnFrameRenderTargets.hpp
+++ b/Hydrogent/interface/HnFrameRenderTargets.hpp
@@ -28,7 +28,7 @@
 
 #include <array>
 
-#include "../../../DiligentCore/Graphics/GraphicsEngine/interface/TextureView.h"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/TextureView.h"
 
 namespace Diligent
 {
diff --git a/Hydrogent/interface/HnLight.hpp b/Hydrogent/interface/HnLight.hpp
index 46d2317..b563e5a 100644
--- a/Hydrogent/interface/HnLight.hpp
+++ b/Hydrogent/interface/HnLight.hpp
@@ -28,9 +28,9 @@
 
 #include "pxr/imaging/hd/light.h"
 
-#include "../../../DiligentCore/Common/interface/BasicMath.hpp"
-#include "../../../DiligentCore/Common/interface/AdvancedMath.hpp"
-#include "../../../DiligentTools/AssetLoader/interface/GLTFLoader.hpp"
+#include "DiligentCore/Common/interface/BasicMath.hpp"
+#include "DiligentCore/Common/interface/AdvancedMath.hpp"
+#include "DiligentTools/AssetLoader/interface/GLTFLoader.hpp"
 
 namespace Diligent
 {
diff --git a/Hydrogent/interface/HnMaterial.hpp b/Hydrogent/interface/HnMaterial.hpp
index 4cfca7d..ce1dc2f 100644
--- a/Hydrogent/interface/HnMaterial.hpp
+++ b/Hydrogent/interface/HnMaterial.hpp
@@ -35,11 +35,11 @@
 
 #include "pxr/imaging/hd/material.h"
 
-#include "../../../DiligentCore/Graphics/GraphicsEngine/interface/RenderDevice.h"
-#include "../../../DiligentCore/Graphics/GraphicsEngine/interface/ShaderResourceBinding.h"
-#include "../../../DiligentCore/Common/interface/RefCntAutoPtr.hpp"
-#include "../../../DiligentCore/Common/interface/BasicMath.hpp"
-#include "../../../DiligentTools/AssetLoader/interface/GLTFLoader.hpp"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/RenderDevice.h"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/ShaderResourceBinding.h"
+#include "DiligentCore/Common/interface/RefCntAutoPtr.hpp"
+#include "DiligentCore/Common/interface/BasicMath.hpp"
+#include "DiligentTools/AssetLoader/interface/GLTFLoader.hpp"
 #include "../../../DiligentFX/PBR/interface/PBR_Renderer.hpp"
 
 namespace Diligent
diff --git a/Hydrogent/interface/HnMesh.hpp b/Hydrogent/interface/HnMesh.hpp
index acd9d30..4cebca9 100644
--- a/Hydrogent/interface/HnMesh.hpp
+++ b/Hydrogent/interface/HnMesh.hpp
@@ -35,12 +35,12 @@
 #include "pxr/imaging/hd/mesh.h"
 #include "pxr/base/tf/token.h"
 
-#include "../../../DiligentCore/Graphics/GraphicsEngine/interface/RenderDevice.h"
-#include "../../../DiligentCore/Graphics/GraphicsEngine/interface/Buffer.h"
-#include "../../../DiligentCore/Graphics/GraphicsEngine/interface/GraphicsTypesX.hpp"
-#include "../../../DiligentCore/Common/interface/RefCntAutoPtr.hpp"
-#include "../../../DiligentCore/Common/interface/BasicMath.hpp"
-#include "../../../DiligentCore/Common/interface/STDAllocator.hpp"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/RenderDevice.h"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/Buffer.h"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/GraphicsTypesX.hpp"
+#include "DiligentCore/Common/interface/RefCntAutoPtr.hpp"
+#include "DiligentCore/Common/interface/BasicMath.hpp"
+#include "DiligentCore/Common/interface/STDAllocator.hpp"
 
 #include "entt/entity/entity.hpp"
 
diff --git a/Hydrogent/interface/HnRenderBuffer.hpp b/Hydrogent/interface/HnRenderBuffer.hpp
index 1b78069..1e31c73 100644
--- a/Hydrogent/interface/HnRenderBuffer.hpp
+++ b/Hydrogent/interface/HnRenderBuffer.hpp
@@ -28,8 +28,8 @@
 
 #include "pxr/imaging/hd/renderBuffer.h"
 
-#include "../../../DiligentCore/Graphics/GraphicsEngine/interface/Texture.h"
-#include "../../../DiligentCore/Common/interface/RefCntAutoPtr.hpp"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/Texture.h"
+#include "DiligentCore/Common/interface/RefCntAutoPtr.hpp"
 
 namespace Diligent
 {
diff --git a/Hydrogent/interface/HnRenderDelegate.hpp b/Hydrogent/interface/HnRenderDelegate.hpp
index 637aa78..3a68358 100644
--- a/Hydrogent/interface/HnRenderDelegate.hpp
+++ b/Hydrogent/interface/HnRenderDelegate.hpp
@@ -35,9 +35,9 @@
 
 #include "pxr/imaging/hd/renderDelegate.h"
 
-#include "../../../DiligentCore/Graphics/GraphicsEngine/interface/RenderDevice.h"
-#include "../../../DiligentCore/Graphics/GraphicsTools/interface/RenderStateCache.h"
-#include "../../../DiligentCore/Common/interface/RefCntAutoPtr.hpp"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/RenderDevice.h"
+#include "DiligentCore/Graphics/GraphicsTools/interface/RenderStateCache.h"
+#include "DiligentCore/Common/interface/RefCntAutoPtr.hpp"
 #include "../../PBR/interface/USD_Renderer.hpp"
 
 #include "entt/entity/registry.hpp"
diff --git a/Hydrogent/interface/HnRenderPassState.hpp b/Hydrogent/interface/HnRenderPassState.hpp
index 8cab519..cc7bacd 100644
--- a/Hydrogent/interface/HnRenderPassState.hpp
+++ b/Hydrogent/interface/HnRenderPassState.hpp
@@ -28,13 +28,13 @@
 
 #include <array>
 
-#include "../../../DiligentCore/Graphics/GraphicsEngine/interface/RasterizerState.h"
-#include "../../../DiligentCore/Graphics/GraphicsEngine/interface/BlendState.h"
-#include "../../../DiligentCore/Graphics/GraphicsEngine/interface/DepthStencilState.h"
-#include "../../../DiligentCore/Graphics/GraphicsEngine/interface/PipelineState.h"
-#include "../../../DiligentCore/Graphics/GraphicsEngine/interface/DeviceContext.h"
-#include "../../../DiligentCore/Common/interface/RefCntAutoPtr.hpp"
-#include "../../../DiligentCore/Common/interface/BasicMath.hpp"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/RasterizerState.h"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/BlendState.h"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/DepthStencilState.h"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/PipelineState.h"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/DeviceContext.h"
+#include "DiligentCore/Common/interface/RefCntAutoPtr.hpp"
+#include "DiligentCore/Common/interface/BasicMath.hpp"
 
 #include "HnTypes.hpp"
 
diff --git a/Hydrogent/interface/HnTextureRegistry.hpp b/Hydrogent/interface/HnTextureRegistry.hpp
index 4bdc5c5..1fbdd29 100644
--- a/Hydrogent/interface/HnTextureRegistry.hpp
+++ b/Hydrogent/interface/HnTextureRegistry.hpp
@@ -35,11 +35,11 @@
 #include "pxr/base/tf/token.h"
 #include "pxr/imaging/hd/types.h"
 
-#include "../../../DiligentCore/Graphics/GraphicsEngine/interface/RenderDevice.h"
-#include "../../../DiligentCore/Graphics/GraphicsEngine/interface/DeviceContext.h"
-#include "../../../DiligentCore/Common/interface/RefCntAutoPtr.hpp"
-#include "../../../DiligentCore/Common/interface/ObjectsRegistry.hpp"
-#include "../../../DiligentTools/TextureLoader/interface/TextureLoader.h"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/RenderDevice.h"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/DeviceContext.h"
+#include "DiligentCore/Common/interface/RefCntAutoPtr.hpp"
+#include "DiligentCore/Common/interface/ObjectsRegistry.hpp"
+#include "DiligentTools/TextureLoader/interface/TextureLoader.h"
 
 namespace Diligent
 {
diff --git a/Hydrogent/interface/HnTypes.hpp b/Hydrogent/interface/HnTypes.hpp
index ae923de..04a438e 100644
--- a/Hydrogent/interface/HnTypes.hpp
+++ b/Hydrogent/interface/HnTypes.hpp
@@ -26,7 +26,7 @@
 
 #pragma once
 
-#include "../../../Common/interface/BasicMath.hpp"
+#include "DiligentCore/Common/interface/BasicMath.hpp"
 
 #include "pxr/usd/usd/stage.h"
 
diff --git a/Hydrogent/interface/Tasks/HnBeginFrameTask.hpp b/Hydrogent/interface/Tasks/HnBeginFrameTask.hpp
index 9b26a77..d7e766e 100644
--- a/Hydrogent/interface/Tasks/HnBeginFrameTask.hpp
+++ b/Hydrogent/interface/Tasks/HnBeginFrameTask.hpp
@@ -36,8 +36,8 @@
 #include "../interface/HnTypes.hpp"
 #include "../interface/HnFrameRenderTargets.hpp"
 
-#include "../../../../DiligentCore/Common/interface/BasicMath.hpp"
-#include "../../../../DiligentCore/Common/interface/Timer.hpp"
+#include "DiligentCore/Common/interface/BasicMath.hpp"
+#include "DiligentCore/Common/interface/Timer.hpp"
 
 namespace Diligent
 {
diff --git a/Hydrogent/interface/Tasks/HnPostProcessTask.hpp b/Hydrogent/interface/Tasks/HnPostProcessTask.hpp
index d34fbcd..6935de5 100644
--- a/Hydrogent/interface/Tasks/HnPostProcessTask.hpp
+++ b/Hydrogent/interface/Tasks/HnPostProcessTask.hpp
@@ -31,14 +31,14 @@
 #include "HnTask.hpp"
 #include "HnTypes.hpp"
 
-#include "../../../../DiligentCore/Graphics/GraphicsEngine/interface/PipelineState.h"
-#include "../../../../DiligentCore/Graphics/GraphicsEngine/interface/PipelineResourceSignature.h"
-#include "../../../../DiligentCore/Graphics/GraphicsEngine/interface/ShaderResourceBinding.h"
-#include "../../../../DiligentCore/Graphics/GraphicsEngine/interface/Buffer.h"
-#include "../../../../DiligentCore/Graphics/GraphicsTools/interface/RenderStateCache.h"
-#include "../../../../DiligentCore/Common/interface/RefCntAutoPtr.hpp"
-#include "../../../../DiligentCore/Common/interface/BasicMath.hpp"
-#include "../../../../DiligentCore/Graphics/GraphicsEngine/interface/GraphicsTypesX.hpp"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/PipelineState.h"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/PipelineResourceSignature.h"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/ShaderResourceBinding.h"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/Buffer.h"
+#include "DiligentCore/Graphics/GraphicsTools/interface/RenderStateCache.h"
+#include "DiligentCore/Common/interface/RefCntAutoPtr.hpp"
+#include "DiligentCore/Common/interface/BasicMath.hpp"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/GraphicsTypesX.hpp"
 #include "../../PBR/interface/PBR_Renderer.hpp"
 #include "../../PostProcess/ScreenSpaceAmbientOcclusion/interface/ScreenSpaceAmbientOcclusion.hpp"
 #include "../../PostProcess/ScreenSpaceReflection/interface/ScreenSpaceReflection.hpp"
diff --git a/Hydrogent/interface/Tasks/HnProcessSelectionTask.hpp b/Hydrogent/interface/Tasks/HnProcessSelectionTask.hpp
index fc66e58..e6201d2 100644
--- a/Hydrogent/interface/Tasks/HnProcessSelectionTask.hpp
+++ b/Hydrogent/interface/Tasks/HnProcessSelectionTask.hpp
@@ -30,11 +30,11 @@
 
 #include "HnTask.hpp"
 
-#include "../../../../DiligentCore/Graphics/GraphicsEngine/interface/PipelineState.h"
-#include "../../../../DiligentCore/Graphics/GraphicsEngine/interface/ShaderResourceBinding.h"
-#include "../../../../DiligentCore/Graphics/GraphicsEngine/interface/Buffer.h"
-#include "../../../../DiligentCore/Common/interface/RefCntAutoPtr.hpp"
-#include "../../../../DiligentCore/Common/interface/BasicMath.hpp"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/PipelineState.h"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/ShaderResourceBinding.h"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/Buffer.h"
+#include "DiligentCore/Common/interface/RefCntAutoPtr.hpp"
+#include "DiligentCore/Common/interface/BasicMath.hpp"
 
 namespace Diligent
 {
diff --git a/Hydrogent/interface/Tasks/HnReadRprimIdTask.hpp b/Hydrogent/interface/Tasks/HnReadRprimIdTask.hpp
index 60595a7..1f77ea0 100644
--- a/Hydrogent/interface/Tasks/HnReadRprimIdTask.hpp
+++ b/Hydrogent/interface/Tasks/HnReadRprimIdTask.hpp
@@ -30,7 +30,7 @@
 
 #include <memory>
 
-#include "../../../../DiligentCore/Graphics/GraphicsTools/interface/GPUCompletionAwaitQueue.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/GPUCompletionAwaitQueue.hpp"
 
 namespace Diligent
 {
diff --git a/Hydrogent/interface/Tasks/HnRenderBoundBoxTask.hpp b/Hydrogent/interface/Tasks/HnRenderBoundBoxTask.hpp
index eea7d5d..44e59b5 100644
--- a/Hydrogent/interface/Tasks/HnRenderBoundBoxTask.hpp
+++ b/Hydrogent/interface/Tasks/HnRenderBoundBoxTask.hpp
@@ -30,7 +30,7 @@
 
 #include <memory>
 
-#include "../../../../DiligentCore/Common/interface/BasicMath.hpp"
+#include "DiligentCore/Common/interface/BasicMath.hpp"
 
 namespace Diligent
 {
diff --git a/Hydrogent/interface/Tasks/HnRenderRprimsTask.hpp b/Hydrogent/interface/Tasks/HnRenderRprimsTask.hpp
index 5e097f6..7fbde16 100644
--- a/Hydrogent/interface/Tasks/HnRenderRprimsTask.hpp
+++ b/Hydrogent/interface/Tasks/HnRenderRprimsTask.hpp
@@ -29,7 +29,7 @@
 #include "HnTask.hpp"
 #include "HnTypes.hpp"
 
-#include "../../../Common/interface/BasicMath.hpp"
+#include "DiligentCore/Common/interface/BasicMath.hpp"
 
 #include "pxr/imaging/hd/renderPass.h"
 
diff --git a/Hydrogent/interface/Tasks/HnRenderShadowsTask.hpp b/Hydrogent/interface/Tasks/HnRenderShadowsTask.hpp
index f834865..43e61ef 100644
--- a/Hydrogent/interface/Tasks/HnRenderShadowsTask.hpp
+++ b/Hydrogent/interface/Tasks/HnRenderShadowsTask.hpp
@@ -32,10 +32,10 @@
 
 #include "../interface/HnRenderPassState.hpp"
 
-#include "../../../../DiligentCore/Graphics/GraphicsEngine/interface/GraphicsTypes.h"
-#include "../../../../DiligentCore/Graphics/GraphicsEngine/interface/PipelineState.h"
-#include "../../../../DiligentCore/Graphics/GraphicsEngine/interface/Buffer.h"
-#include "../../../../DiligentCore/Common/interface/RefCntAutoPtr.hpp"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/GraphicsTypes.h"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/PipelineState.h"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/Buffer.h"
+#include "DiligentCore/Common/interface/RefCntAutoPtr.hpp"
 
 namespace Diligent
 {
diff --git a/Hydrogent/interface/Tasks/HnTask.hpp b/Hydrogent/interface/Tasks/HnTask.hpp
index 4d341c7..794f6db 100644
--- a/Hydrogent/interface/Tasks/HnTask.hpp
+++ b/Hydrogent/interface/Tasks/HnTask.hpp
@@ -31,7 +31,7 @@
 #include "pxr/imaging/hd/task.h"
 #include "pxr/imaging/hd/renderIndex.h"
 
-#include "../../../../DiligentCore/Platforms/Basic/interface/DebugUtilities.hpp"
+#include "DiligentCore/Platforms/Basic/interface/DebugUtilities.hpp"
 
 namespace Diligent
 {
diff --git a/Hydrogent/interface/Tasks/HnTaskManager.hpp b/Hydrogent/interface/Tasks/HnTaskManager.hpp
index fba0132..42611c3 100644
--- a/Hydrogent/interface/Tasks/HnTaskManager.hpp
+++ b/Hydrogent/interface/Tasks/HnTaskManager.hpp
@@ -31,7 +31,7 @@
 
 #include "Tasks/HnTask.hpp"
 
-#include "../../../../DiligentCore/Platforms/Basic/interface/DebugUtilities.hpp"
+#include "DiligentCore/Platforms/Basic/interface/DebugUtilities.hpp"
 
 namespace Diligent
 {
diff --git a/Hydrogent/src/HnCamera.cpp b/Hydrogent/src/HnCamera.cpp
index 7a45e7d..2a91440 100644
--- a/Hydrogent/src/HnCamera.cpp
+++ b/Hydrogent/src/HnCamera.cpp
@@ -29,7 +29,7 @@
 #include "HnRenderParam.hpp"
 
 #include "GfTypeConversions.hpp"
-#include "BasicMath.hpp"
+#include "DiligentCore/Common/interface/BasicMath.hpp"
 
 #include "pxr/imaging/hd/sceneDelegate.h"
 
diff --git a/Hydrogent/src/HnLight.cpp b/Hydrogent/src/HnLight.cpp
index 6df4da2..3205e78 100644
--- a/Hydrogent/src/HnLight.cpp
+++ b/Hydrogent/src/HnLight.cpp
@@ -37,7 +37,7 @@
 #include "pxr/imaging/hd/sceneDelegate.h"
 
 #include "GfTypeConversions.hpp"
-#include "BasicMath.hpp"
+#include "DiligentCore/Common/interface/BasicMath.hpp"
 
 namespace Diligent
 {
diff --git a/Hydrogent/src/HnMaterialNetwork.cpp b/Hydrogent/src/HnMaterialNetwork.cpp
index 11d516c..87d3dfc 100644
--- a/Hydrogent/src/HnMaterialNetwork.cpp
+++ b/Hydrogent/src/HnMaterialNetwork.cpp
@@ -29,7 +29,7 @@
 
 #include "DebugUtilities.hpp"
 #include "StringTools.hpp"
-#include "GraphicsAccessories.hpp"
+#include "DiligentCore/Graphics/GraphicsAccessories/interface/GraphicsAccessories.hpp"
 
 #include "pxr/base/gf/vec3f.h"
 #include "pxr/base/gf/vec4f.h"
diff --git a/Hydrogent/src/HnMesh.cpp b/Hydrogent/src/HnMesh.cpp
index a3e63eb..a9ae651 100644
--- a/Hydrogent/src/HnMesh.cpp
+++ b/Hydrogent/src/HnMesh.cpp
@@ -35,7 +35,7 @@
 #include "GfTypeConversions.hpp"
 
 #include "DebugUtilities.hpp"
-#include "GraphicsTypesX.hpp"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/GraphicsTypesX.hpp"
 #include "GLTFResourceManager.hpp"
 #include "EngineMemory.h"
 
diff --git a/Hydrogent/src/HnRenderBuffer.cpp b/Hydrogent/src/HnRenderBuffer.cpp
index 710762b..8d52777 100644
--- a/Hydrogent/src/HnRenderBuffer.cpp
+++ b/Hydrogent/src/HnRenderBuffer.cpp
@@ -29,7 +29,7 @@
 #include "HnRenderDelegate.hpp"
 
 #include "DebugUtilities.hpp"
-#include "GraphicsAccessories.hpp"
+#include "DiligentCore/Graphics/GraphicsAccessories/interface/GraphicsAccessories.hpp"
 
 namespace Diligent
 {
diff --git a/Hydrogent/src/HnRenderDelegate.cpp b/Hydrogent/src/HnRenderDelegate.cpp
index 8e2c7e7..4d6df47 100644
--- a/Hydrogent/src/HnRenderDelegate.cpp
+++ b/Hydrogent/src/HnRenderDelegate.cpp
@@ -35,10 +35,10 @@
 #include "HnShadowMapManager.hpp"
 
 #include "DebugUtilities.hpp"
-#include "GraphicsUtilities.h"
+#include "DiligentCore/Graphics/GraphicsTools/interface/GraphicsUtilities.h"
 #include "HnRenderBuffer.hpp"
-#include "Align.hpp"
-#include "PlatformMisc.hpp"
+#include "DiligentCore/Common/interface/Align.hpp"
+#include "DiligentCore/Platforms/interface/PlatformMisc.hpp"
 #include "GLTFResourceManager.hpp"
 
 #include "pxr/imaging/hd/material.h"
diff --git a/Hydrogent/src/HnRenderPass.cpp b/Hydrogent/src/HnRenderPass.cpp
index 212495d..4ce4669 100644
--- a/Hydrogent/src/HnRenderPass.cpp
+++ b/Hydrogent/src/HnRenderPass.cpp
@@ -40,8 +40,8 @@
 
 #include "USD_Renderer.hpp"
 #include "GLTF_PBR_Renderer.hpp"
-#include "MapHelper.hpp"
-#include "ScopedDebugGroup.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/MapHelper.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/ScopedDebugGroup.hpp"
 #include "HashUtils.hpp"
 
 namespace Diligent
diff --git a/Hydrogent/src/HnShaderSourceFactory.cpp b/Hydrogent/src/HnShaderSourceFactory.cpp
index 0805052..e0c9ff5 100644
--- a/Hydrogent/src/HnShaderSourceFactory.cpp
+++ b/Hydrogent/src/HnShaderSourceFactory.cpp
@@ -25,10 +25,10 @@
  */
 
 #include "HnShaderSourceFactory.hpp"
-#include "ShaderSourceFactoryUtils.h"
+#include "DiligentCore/Graphics/GraphicsTools/interface/ShaderSourceFactoryUtils.h"
 
 #include "Utilities/interface/DiligentFXShaderSourceStreamFactory.hpp"
-#include "ShaderSourceFactoryUtils.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/ShaderSourceFactoryUtils.hpp"
 
 namespace Diligent
 {
diff --git a/Hydrogent/src/HnTextureRegistry.cpp b/Hydrogent/src/HnTextureRegistry.cpp
index dce5859..b553bf6 100644
--- a/Hydrogent/src/HnTextureRegistry.cpp
+++ b/Hydrogent/src/HnTextureRegistry.cpp
@@ -30,7 +30,7 @@
 #include "GLTFResourceManager.hpp"
 #include "USD_Renderer.hpp"
 #include "HnTextureIdentifier.hpp"
-#include "GraphicsAccessories.hpp"
+#include "DiligentCore/Graphics/GraphicsAccessories/interface/GraphicsAccessories.hpp"
 
 #include <mutex>
 
diff --git a/Hydrogent/src/Tasks/HnBeginFrameTask.cpp b/Hydrogent/src/Tasks/HnBeginFrameTask.cpp
index 3a8fbea..4e8f549 100644
--- a/Hydrogent/src/Tasks/HnBeginFrameTask.cpp
+++ b/Hydrogent/src/Tasks/HnBeginFrameTask.cpp
@@ -36,9 +36,9 @@
 #include "HnShadowMapManager.hpp"
 
 #include "DebugUtilities.hpp"
-#include "GraphicsAccessories.hpp"
-#include "MapHelper.hpp"
-#include "ScopedDebugGroup.hpp"
+#include "DiligentCore/Graphics/GraphicsAccessories/interface/GraphicsAccessories.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/MapHelper.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/ScopedDebugGroup.hpp"
 #include "GLTF_PBR_Renderer.hpp"
 
 namespace Diligent
diff --git a/Hydrogent/src/Tasks/HnCopySelectionDepthTask.cpp b/Hydrogent/src/Tasks/HnCopySelectionDepthTask.cpp
index e1cace4..8973a13 100644
--- a/Hydrogent/src/Tasks/HnCopySelectionDepthTask.cpp
+++ b/Hydrogent/src/Tasks/HnCopySelectionDepthTask.cpp
@@ -29,7 +29,7 @@
 #include "HnFrameRenderTargets.hpp"
 
 #include "DebugUtilities.hpp"
-#include "ScopedDebugGroup.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/ScopedDebugGroup.hpp"
 
 namespace Diligent
 {
diff --git a/Hydrogent/src/Tasks/HnPostProcessTask.cpp b/Hydrogent/src/Tasks/HnPostProcessTask.cpp
index a851801..9fedbbf 100644
--- a/Hydrogent/src/Tasks/HnPostProcessTask.cpp
+++ b/Hydrogent/src/Tasks/HnPostProcessTask.cpp
@@ -34,13 +34,13 @@
 
 #include "DebugUtilities.hpp"
 #include "TextureView.h"
-#include "RenderStateCache.hpp"
-#include "ShaderMacroHelper.hpp"
-#include "CommonlyUsedStates.h"
-#include "GraphicsUtilities.h"
+#include "DiligentCore/Graphics/GraphicsTools/interface/RenderStateCache.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/ShaderMacroHelper.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/CommonlyUsedStates.h"
+#include "DiligentCore/Graphics/GraphicsTools/interface/GraphicsUtilities.h"
 #include "VectorFieldRenderer.hpp"
 #include "ToneMapping.hpp"
-#include "ScopedDebugGroup.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/ScopedDebugGroup.hpp"
 
 namespace Diligent
 {
diff --git a/Hydrogent/src/Tasks/HnProcessSelectionTask.cpp b/Hydrogent/src/Tasks/HnProcessSelectionTask.cpp
index a162a01..7ba2b82 100644
--- a/Hydrogent/src/Tasks/HnProcessSelectionTask.cpp
+++ b/Hydrogent/src/Tasks/HnProcessSelectionTask.cpp
@@ -32,10 +32,10 @@
 #include "HnShaderSourceFactory.hpp"
 
 #include "DebugUtilities.hpp"
-#include "CommonlyUsedStates.h"
-#include "GraphicsUtilities.h"
-#include "MapHelper.hpp"
-#include "ScopedDebugGroup.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/CommonlyUsedStates.h"
+#include "DiligentCore/Graphics/GraphicsTools/interface/GraphicsUtilities.h"
+#include "DiligentCore/Graphics/GraphicsTools/interface/MapHelper.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/ScopedDebugGroup.hpp"
 
 namespace Diligent
 {
diff --git a/Hydrogent/src/Tasks/HnReadRprimIdTask.cpp b/Hydrogent/src/Tasks/HnReadRprimIdTask.cpp
index 62fd064..1583b65 100644
--- a/Hydrogent/src/Tasks/HnReadRprimIdTask.cpp
+++ b/Hydrogent/src/Tasks/HnReadRprimIdTask.cpp
@@ -30,7 +30,7 @@
 #include "HnTokens.hpp"
 
 #include "DebugUtilities.hpp"
-#include "ScopedDebugGroup.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/ScopedDebugGroup.hpp"
 
 namespace Diligent
 {
diff --git a/Hydrogent/src/Tasks/HnRenderBoundBoxTask.cpp b/Hydrogent/src/Tasks/HnRenderBoundBoxTask.cpp
index aedb80c..a4a7b7b 100644
--- a/Hydrogent/src/Tasks/HnRenderBoundBoxTask.cpp
+++ b/Hydrogent/src/Tasks/HnRenderBoundBoxTask.cpp
@@ -35,7 +35,7 @@
 
 #include "BoundBoxRenderer.hpp"
 #include "DebugUtilities.hpp"
-#include "ScopedDebugGroup.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/ScopedDebugGroup.hpp"
 
 namespace Diligent
 {
diff --git a/Hydrogent/src/Tasks/HnRenderEnvMapTask.cpp b/Hydrogent/src/Tasks/HnRenderEnvMapTask.cpp
index 52b3c2a..f2c2e38 100644
--- a/Hydrogent/src/Tasks/HnRenderEnvMapTask.cpp
+++ b/Hydrogent/src/Tasks/HnRenderEnvMapTask.cpp
@@ -34,7 +34,7 @@
 #include "USD_Renderer.hpp"
 
 #include "DebugUtilities.hpp"
-#include "ScopedDebugGroup.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/ScopedDebugGroup.hpp"
 
 namespace Diligent
 {
@@ -42,8 +42,8 @@ namespace Diligent
 namespace HLSL
 {
 
-#include "Shaders/Common/public/ShaderDefinitions.fxh"
-#include "Shaders/PostProcess/ToneMapping/public/ToneMappingStructures.fxh"
+#include "../../../Shaders/Common/public/ShaderDefinitions.fxh"
+#include "../../../Shaders/PostProcess/ToneMapping/public/ToneMappingStructures.fxh"
 
 } // namespace HLSL
 
diff --git a/Hydrogent/src/Tasks/HnRenderShadowsTask.cpp b/Hydrogent/src/Tasks/HnRenderShadowsTask.cpp
index acd4f44..0acc462 100644
--- a/Hydrogent/src/Tasks/HnRenderShadowsTask.cpp
+++ b/Hydrogent/src/Tasks/HnRenderShadowsTask.cpp
@@ -32,8 +32,8 @@
 #include "HnRenderParam.hpp"
 #include "HnLight.hpp"
 #include "HnShadowMapManager.hpp"
-#include "CommonlyUsedStates.h"
-#include "GraphicsUtilities.h"
+#include "DiligentCore/Graphics/GraphicsTools/interface/CommonlyUsedStates.h"
+#include "DiligentCore/Graphics/GraphicsTools/interface/GraphicsUtilities.h"
 
 namespace Diligent
 {
diff --git a/PBR/CMakeLists.txt b/PBR/CMakeLists.txt
index 0824dc2..0d05274 100644
--- a/PBR/CMakeLists.txt
+++ b/PBR/CMakeLists.txt
@@ -17,4 +17,10 @@ target_sources(DiligentFX PRIVATE ${SOURCE} ${INCLUDE})
 target_include_directories(DiligentFX
 PUBLIC
     "${CMAKE_CURRENT_SOURCE_DIR}/interface"
+PRIVATE
+    "${diligentcore_INCLUDEDIR}/DiligentCore/Graphics/HLSL2GLSLConverterLib/interface"
+    "${diligentcore_INCLUDEDIR}/DiligentCore/Common/interface"
+    "${diligentcore_INCLUDEDIR}/DiligentCore/Graphics/GraphicsEngine/interface"
+    "${CMAKE_CURRENT_SOURCE_DIR}/../DiligentCore-2.5.6/Graphics/HLSL2GLSLConverterLib/include"
+    "${CMAKE_CURRENT_SOURCE_DIR}/../DiligentCore-2.5.6/Graphics/ShaderTools/include"
 )
diff --git a/PBR/interface/GLTF_PBR_Renderer.hpp b/PBR/interface/GLTF_PBR_Renderer.hpp
index b8271ca..830b8ef 100644
--- a/PBR/interface/GLTF_PBR_Renderer.hpp
+++ b/PBR/interface/GLTF_PBR_Renderer.hpp
@@ -32,7 +32,7 @@
 #include <vector>
 #include <array>
 
-#include "../../../DiligentTools/AssetLoader/interface/GLTFLoader.hpp"
+#include "DiligentTools/AssetLoader/interface/GLTFLoader.hpp"
 
 namespace Diligent
 {
diff --git a/PBR/interface/PBR_Renderer.hpp b/PBR/interface/PBR_Renderer.hpp
index 1b8fbc6..b17782d 100644
--- a/PBR/interface/PBR_Renderer.hpp
+++ b/PBR/interface/PBR_Renderer.hpp
@@ -31,14 +31,14 @@
 #include <functional>
 #include <array>
 
-#include "../../../DiligentCore/Platforms/Basic/interface/DebugUtilities.hpp"
-#include "../../../DiligentCore/Graphics/GraphicsEngine/interface/DeviceContext.h"
-#include "../../../DiligentCore/Graphics/GraphicsEngine/interface/RenderDevice.h"
-#include "../../../DiligentCore/Graphics/GraphicsEngine/interface/GraphicsTypesX.hpp"
-#include "../../../DiligentCore/Graphics/GraphicsTools/interface/RenderStateCache.hpp"
-#include "../../../DiligentCore/Graphics/GraphicsTools/interface/ShaderMacroHelper.hpp"
-#include "../../../DiligentCore/Common/interface/RefCntAutoPtr.hpp"
-#include "../../../DiligentCore/Common/interface/HashUtils.hpp"
+#include "DiligentCore/Platforms/Basic/interface/DebugUtilities.hpp"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/DeviceContext.h"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/RenderDevice.h"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/GraphicsTypesX.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/RenderStateCache.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/ShaderMacroHelper.hpp"
+#include "DiligentCore/Common/interface/RefCntAutoPtr.hpp"
+#include "DiligentCore/Common/interface/HashUtils.hpp"
 
 namespace Diligent
 {
diff --git a/PBR/src/GLTF_PBR_Renderer.cpp b/PBR/src/GLTF_PBR_Renderer.cpp
index 3292ad8..77f2945 100644
--- a/PBR/src/GLTF_PBR_Renderer.cpp
+++ b/PBR/src/GLTF_PBR_Renderer.cpp
@@ -32,10 +32,10 @@
 #include <algorithm>
 #include <cmath>
 
-#include "BasicMath.hpp"
-#include "MapHelper.hpp"
-#include "GraphicsAccessories.hpp"
-#include "GLTFLoader.hpp"
+#include "DiligentCore/Common/interface/BasicMath.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/MapHelper.hpp"
+#include "DiligentCore/Graphics/GraphicsAccessories/interface/GraphicsAccessories.hpp"
+#include "DiligentTools/AssetLoader/interface/GLTFLoader.hpp"
 
 namespace Diligent
 {
@@ -52,7 +52,7 @@ GLTF_PBR_Renderer::ALPHA_MODE GLTF_PBR_Renderer::GltfAlphaModeToAlphaMode(GLTF::
 namespace HLSL
 {
 
-#include "Shaders/PBR/public/PBR_Structures.fxh"
+#include "../../../Shaders/PBR/public/PBR_Structures.fxh"
 
 } // namespace HLSL
 
diff --git a/PBR/src/PBR_Renderer.cpp b/PBR/src/PBR_Renderer.cpp
index 286e2d8..7d39ba8 100644
--- a/PBR/src/PBR_Renderer.cpp
+++ b/PBR/src/PBR_Renderer.cpp
@@ -29,19 +29,19 @@
 #include <array>
 #include <vector>
 
-#include "RenderStateCache.hpp"
-#include "GraphicsUtilities.h"
-#include "CommonlyUsedStates.h"
-#include "BasicMath.hpp"
-#include "MapHelper.hpp"
-#include "GraphicsAccessories.hpp"
-#include "PlatformMisc.hpp"
-#include "TextureUtilities.h"
+#include "DiligentCore/Graphics/GraphicsTools/interface/RenderStateCache.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/GraphicsUtilities.h"
+#include "DiligentCore/Graphics/GraphicsTools/interface/CommonlyUsedStates.h"
+#include "DiligentCore/Common/interface/BasicMath.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/MapHelper.hpp"
+#include "DiligentCore/Graphics/GraphicsAccessories/interface/GraphicsAccessories.hpp"
+#include "DiligentCore/Platforms/interface/PlatformMisc.hpp"
+#include "DiligentTools/TextureLoader/interface/TextureUtilities.h"
 #include "Utilities/interface/DiligentFXShaderSourceStreamFactory.hpp"
-#include "ShaderSourceFactoryUtils.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/ShaderSourceFactoryUtils.hpp"
 
 #if HLSL2GLSL_CONVERTER_SUPPORTED
-#    include "../include/HLSL2GLSLConverterImpl.hpp"
+#    include "HLSL2GLSLConverterImpl.hpp"
 #endif
 
 namespace Diligent
diff --git a/PBR/src/USD_Renderer.cpp b/PBR/src/USD_Renderer.cpp
index 07f22e7..4c0f240 100644
--- a/PBR/src/USD_Renderer.cpp
+++ b/PBR/src/USD_Renderer.cpp
@@ -30,9 +30,9 @@
 #include <unordered_set>
 #include <functional>
 
-#include "RenderStateCache.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/RenderStateCache.hpp"
 #include "Utilities/interface/DiligentFXShaderSourceStreamFactory.hpp"
-#include "ShaderSourceFactoryUtils.h"
+#include "DiligentCore/Graphics/GraphicsTools/interface/ShaderSourceFactoryUtils.h"
 
 namespace Diligent
 {
diff --git a/PostProcess/Bloom/interface/Bloom.hpp b/PostProcess/Bloom/interface/Bloom.hpp
index 9269811..1220889 100644
--- a/PostProcess/Bloom/interface/Bloom.hpp
+++ b/PostProcess/Bloom/interface/Bloom.hpp
@@ -29,11 +29,11 @@
 #include <vector>
 #include <memory>
 
-#include "../../../../DiligentCore/Common/interface/Timer.hpp"
-#include "../../../../DiligentCore/Graphics/GraphicsEngine/interface/RenderDevice.h"
-#include "../../../../DiligentCore/Graphics/GraphicsTools/interface/RenderStateCache.h"
-#include "../../../../DiligentCore/Common/interface/RefCntAutoPtr.hpp"
-#include "../../../../DiligentCore/Common/interface/BasicMath.hpp"
+#include "DiligentCore/Common/interface/Timer.hpp"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/RenderDevice.h"
+#include "DiligentCore/Graphics/GraphicsTools/interface/RenderStateCache.h"
+#include "DiligentCore/Common/interface/RefCntAutoPtr.hpp"
+#include "DiligentCore/Common/interface/BasicMath.hpp"
 
 #include "PostProcess/Common/interface/PostFXRenderTechnique.hpp"
 #include "PostProcess/Common/interface/PostFXContext.hpp"
diff --git a/PostProcess/Bloom/src/Bloom.cpp b/PostProcess/Bloom/src/Bloom.cpp
index bda1835..41f80b5 100644
--- a/PostProcess/Bloom/src/Bloom.cpp
+++ b/PostProcess/Bloom/src/Bloom.cpp
@@ -26,16 +26,16 @@
 
 
 #include "Bloom.hpp"
-#include "CommonlyUsedStates.h"
-#include "RenderStateCache.hpp"
-#include "GraphicsUtilities.h"
-#include "MapHelper.hpp"
-#include "ScopedDebugGroup.hpp"
-#include "GraphicsTypesX.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/CommonlyUsedStates.h"
+#include "DiligentCore/Graphics/GraphicsTools/interface/RenderStateCache.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/GraphicsUtilities.h"
+#include "DiligentCore/Graphics/GraphicsTools/interface/MapHelper.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/ScopedDebugGroup.hpp"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/GraphicsTypesX.hpp"
 #include "imgui.h"
-#include "ImGuiUtils.hpp"
+#include "DiligentTools/Imgui/interface/ImGuiUtils.hpp"
 #include "ScreenSpaceReflection.hpp"
-#include "ShaderMacroHelper.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/ShaderMacroHelper.hpp"
 
 #include <numeric>
 
diff --git a/PostProcess/Common/interface/PostFXContext.hpp b/PostProcess/Common/interface/PostFXContext.hpp
index 7d2983a..1c68265 100644
--- a/PostProcess/Common/interface/PostFXContext.hpp
+++ b/PostProcess/Common/interface/PostFXContext.hpp
@@ -29,11 +29,11 @@
 #include <array>
 #include <unordered_map>
 
-#include "../../../../DiligentCore/Graphics/GraphicsEngine/interface/RenderDevice.h"
-#include "../../../../DiligentCore/Graphics/GraphicsTools/interface/RenderStateCache.h"
-#include "../../../../DiligentCore/Graphics/GraphicsTools/interface/ResourceRegistry.hpp"
-#include "../../../../DiligentCore/Common/interface/RefCntAutoPtr.hpp"
-#include "../../../../DiligentCore/Common/interface/BasicMath.hpp"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/RenderDevice.h"
+#include "DiligentCore/Graphics/GraphicsTools/interface/RenderStateCache.h"
+#include "DiligentCore/Graphics/GraphicsTools/interface/ResourceRegistry.hpp"
+#include "DiligentCore/Common/interface/RefCntAutoPtr.hpp"
+#include "DiligentCore/Common/interface/BasicMath.hpp"
 
 #include "PostFXRenderTechnique.hpp"
 
diff --git a/PostProcess/Common/interface/PostFXRenderTechnique.hpp b/PostProcess/Common/interface/PostFXRenderTechnique.hpp
index 9adc761..dea3906 100644
--- a/PostProcess/Common/interface/PostFXRenderTechnique.hpp
+++ b/PostProcess/Common/interface/PostFXRenderTechnique.hpp
@@ -28,10 +28,10 @@
 
 #include <vector>
 
-#include "../../../../DiligentCore/Graphics/GraphicsEngine/interface/RenderDevice.h"
-#include "../../../../DiligentCore/Graphics/GraphicsTools/interface/RenderStateCache.h"
-#include "../../../../DiligentCore/Common/interface/RefCntAutoPtr.hpp"
-#include "../../../../DiligentCore/Common/interface/BasicMath.hpp"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/RenderDevice.h"
+#include "DiligentCore/Graphics/GraphicsTools/interface/RenderStateCache.h"
+#include "DiligentCore/Common/interface/RefCntAutoPtr.hpp"
+#include "DiligentCore/Common/interface/BasicMath.hpp"
 
 namespace Diligent
 {
diff --git a/PostProcess/Common/src/PostFXContext.cpp b/PostProcess/Common/src/PostFXContext.cpp
index 78fd798..883414a 100644
--- a/PostProcess/Common/src/PostFXContext.cpp
+++ b/PostProcess/Common/src/PostFXContext.cpp
@@ -26,13 +26,13 @@
 
 #include "PostFXContext.hpp"
 
-#include "CommonlyUsedStates.h"
-#include "GraphicsTypesX.hpp"
-#include "GraphicsUtilities.h"
-#include "MapHelper.hpp"
-#include "RenderStateCache.hpp"
-#include "ScopedDebugGroup.hpp"
-#include "ShaderMacroHelper.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/CommonlyUsedStates.h"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/GraphicsTypesX.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/GraphicsUtilities.h"
+#include "DiligentCore/Graphics/GraphicsTools/interface/MapHelper.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/RenderStateCache.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/ScopedDebugGroup.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/ShaderMacroHelper.hpp"
 
 namespace Diligent
 {
diff --git a/PostProcess/Common/src/PostFXRenderTechnique.cpp b/PostProcess/Common/src/PostFXRenderTechnique.cpp
index 7d8540a..d7e8b1f 100644
--- a/PostProcess/Common/src/PostFXRenderTechnique.cpp
+++ b/PostProcess/Common/src/PostFXRenderTechnique.cpp
@@ -25,7 +25,7 @@
  */
 
 #include "PostFXRenderTechnique.hpp"
-#include "RenderStateCache.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/RenderStateCache.hpp"
 #include "Utilities/interface/DiligentFXShaderSourceStreamFactory.hpp"
 
 namespace Diligent
diff --git a/PostProcess/DepthOfField/interface/DepthOfField.hpp b/PostProcess/DepthOfField/interface/DepthOfField.hpp
index f98091a..9c77ddc 100644
--- a/PostProcess/DepthOfField/interface/DepthOfField.hpp
+++ b/PostProcess/DepthOfField/interface/DepthOfField.hpp
@@ -28,11 +28,11 @@
 #include <unordered_map>
 #include <memory>
 
-#include "../../../../DiligentCore/Common/interface/Timer.hpp"
-#include "../../../../DiligentCore/Graphics/GraphicsEngine/interface/RenderDevice.h"
-#include "../../../../DiligentCore/Graphics/GraphicsTools/interface/RenderStateCache.h"
-#include "../../../../DiligentCore/Common/interface/RefCntAutoPtr.hpp"
-#include "../../../../DiligentCore/Common/interface/BasicMath.hpp"
+#include "DiligentCore/Common/interface/Timer.hpp"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/RenderDevice.h"
+#include "DiligentCore/Graphics/GraphicsTools/interface/RenderStateCache.h"
+#include "DiligentCore/Common/interface/RefCntAutoPtr.hpp"
+#include "DiligentCore/Common/interface/BasicMath.hpp"
 
 #include "PostProcess/Common/interface/PostFXRenderTechnique.hpp"
 #include "PostProcess/Common/interface/PostFXContext.hpp"
diff --git a/PostProcess/DepthOfField/src/DepthOfField.cpp b/PostProcess/DepthOfField/src/DepthOfField.cpp
index 1541cdb..9724bba 100644
--- a/PostProcess/DepthOfField/src/DepthOfField.cpp
+++ b/PostProcess/DepthOfField/src/DepthOfField.cpp
@@ -26,16 +26,16 @@
 
 
 #include "DepthOfField.hpp"
-#include "CommonlyUsedStates.h"
-#include "RenderStateCache.hpp"
-#include "GraphicsUtilities.h"
-#include "MapHelper.hpp"
-#include "ScopedDebugGroup.hpp"
-#include "GraphicsTypesX.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/CommonlyUsedStates.h"
+#include "DiligentCore/Graphics/GraphicsTools/interface/RenderStateCache.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/GraphicsUtilities.h"
+#include "DiligentCore/Graphics/GraphicsTools/interface/MapHelper.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/ScopedDebugGroup.hpp"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/GraphicsTypesX.hpp"
 #include "imgui.h"
-#include "ImGuiUtils.hpp"
+#include "DiligentTools/Imgui/interface/ImGuiUtils.hpp"
 #include "ScreenSpaceReflection.hpp"
-#include "ShaderMacroHelper.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/ShaderMacroHelper.hpp"
 
 namespace Diligent
 {
diff --git a/PostProcess/EpipolarLightScattering/interface/EpipolarLightScattering.hpp b/PostProcess/EpipolarLightScattering/interface/EpipolarLightScattering.hpp
index b9b8000..f9de83c 100644
--- a/PostProcess/EpipolarLightScattering/interface/EpipolarLightScattering.hpp
+++ b/PostProcess/EpipolarLightScattering/interface/EpipolarLightScattering.hpp
@@ -26,15 +26,15 @@
  */
 #pragma once
 
-#include "../../../../DiligentCore/Graphics/GraphicsEngine/interface/RenderDevice.h"
-#include "../../../../DiligentCore/Graphics/GraphicsEngine/interface/DeviceContext.h"
-#include "../../../../DiligentCore/Graphics/GraphicsEngine/interface/Buffer.h"
-#include "../../../../DiligentCore/Graphics/GraphicsEngine/interface/Texture.h"
-#include "../../../../DiligentCore/Graphics/GraphicsEngine/interface/BufferView.h"
-#include "../../../../DiligentCore/Graphics/GraphicsEngine/interface/TextureView.h"
-#include "../../../../DiligentCore/Graphics/GraphicsTools/interface/RenderStateCache.h"
-#include "../../../../DiligentCore/Common/interface/RefCntAutoPtr.hpp"
-#include "../../../../DiligentCore/Common/interface/BasicMath.hpp"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/RenderDevice.h"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/DeviceContext.h"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/Buffer.h"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/Texture.h"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/BufferView.h"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/TextureView.h"
+#include "DiligentCore/Graphics/GraphicsTools/interface/RenderStateCache.h"
+#include "DiligentCore/Common/interface/RefCntAutoPtr.hpp"
+#include "DiligentCore/Common/interface/BasicMath.hpp"
 
 namespace Diligent
 {
diff --git a/PostProcess/EpipolarLightScattering/src/EpipolarLightScattering.cpp b/PostProcess/EpipolarLightScattering/src/EpipolarLightScattering.cpp
index 1e6d846..0fe771e 100644
--- a/PostProcess/EpipolarLightScattering/src/EpipolarLightScattering.cpp
+++ b/PostProcess/EpipolarLightScattering/src/EpipolarLightScattering.cpp
@@ -31,14 +31,14 @@
 #include <cstring>
 
 #include "EpipolarLightScattering.hpp"
-#include "ShaderMacroHelper.hpp"
-#include "GraphicsUtilities.h"
-#include "GraphicsAccessories.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/ShaderMacroHelper.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/GraphicsUtilities.h"
+#include "DiligentCore/Graphics/GraphicsAccessories/interface/GraphicsAccessories.hpp"
 #include "Utilities/interface/DiligentFXShaderSourceStreamFactory.hpp"
-#include "MapHelper.hpp"
-#include "CommonlyUsedStates.h"
-#include "Align.hpp"
-#include "RenderStateCache.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/MapHelper.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/CommonlyUsedStates.h"
+#include "DiligentCore/Common/interface/Align.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/RenderStateCache.hpp"
 
 #define _USE_MATH_DEFINES
 #include <math.h>
diff --git a/PostProcess/ScreenSpaceAmbientOcclusion/interface/ScreenSpaceAmbientOcclusion.hpp b/PostProcess/ScreenSpaceAmbientOcclusion/interface/ScreenSpaceAmbientOcclusion.hpp
index 74085f6..3f1f892 100644
--- a/PostProcess/ScreenSpaceAmbientOcclusion/interface/ScreenSpaceAmbientOcclusion.hpp
+++ b/PostProcess/ScreenSpaceAmbientOcclusion/interface/ScreenSpaceAmbientOcclusion.hpp
@@ -29,11 +29,11 @@
 #include <vector>
 #include <memory>
 
-#include "../../../../DiligentCore/Common/interface/Timer.hpp"
-#include "../../../../DiligentCore/Graphics/GraphicsEngine/interface/RenderDevice.h"
-#include "../../../../DiligentCore/Graphics/GraphicsTools/interface/RenderStateCache.h"
-#include "../../../../DiligentCore/Common/interface/RefCntAutoPtr.hpp"
-#include "../../../../DiligentCore/Common/interface/BasicMath.hpp"
+#include "DiligentCore/Common/interface/Timer.hpp"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/RenderDevice.h"
+#include "DiligentCore/Graphics/GraphicsTools/interface/RenderStateCache.h"
+#include "DiligentCore/Common/interface/RefCntAutoPtr.hpp"
+#include "DiligentCore/Common/interface/BasicMath.hpp"
 
 #include "PostProcess/Common/interface/PostFXRenderTechnique.hpp"
 #include "PostProcess/Common/interface/PostFXContext.hpp"
diff --git a/PostProcess/ScreenSpaceAmbientOcclusion/src/ScreenSpaceAmbientOcclusion.cpp b/PostProcess/ScreenSpaceAmbientOcclusion/src/ScreenSpaceAmbientOcclusion.cpp
index ba596a2..b1a387e 100644
--- a/PostProcess/ScreenSpaceAmbientOcclusion/src/ScreenSpaceAmbientOcclusion.cpp
+++ b/PostProcess/ScreenSpaceAmbientOcclusion/src/ScreenSpaceAmbientOcclusion.cpp
@@ -25,16 +25,16 @@
  */
 
 #include "ScreenSpaceAmbientOcclusion.hpp"
-#include "CommonlyUsedStates.h"
-#include "RenderStateCache.hpp"
-#include "GraphicsUtilities.h"
-#include "MapHelper.hpp"
-#include "ScopedDebugGroup.hpp"
-#include "GraphicsTypesX.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/CommonlyUsedStates.h"
+#include "DiligentCore/Graphics/GraphicsTools/interface/RenderStateCache.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/GraphicsUtilities.h"
+#include "DiligentCore/Graphics/GraphicsTools/interface/MapHelper.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/ScopedDebugGroup.hpp"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/GraphicsTypesX.hpp"
 #include "imgui.h"
-#include "ImGuiUtils.hpp"
+#include "DiligentTools/Imgui/interface/ImGuiUtils.hpp"
 #include "ScreenSpaceReflection.hpp"
-#include "ShaderMacroHelper.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/ShaderMacroHelper.hpp"
 
 namespace Diligent
 {
diff --git a/PostProcess/ScreenSpaceReflection/interface/ScreenSpaceReflection.hpp b/PostProcess/ScreenSpaceReflection/interface/ScreenSpaceReflection.hpp
index 7f1e350..2279036 100644
--- a/PostProcess/ScreenSpaceReflection/interface/ScreenSpaceReflection.hpp
+++ b/PostProcess/ScreenSpaceReflection/interface/ScreenSpaceReflection.hpp
@@ -30,12 +30,12 @@
 #include <unordered_map>
 #include <memory>
 
-#include "../../../../DiligentCore/Common/interface/Timer.hpp"
-#include "../../../../DiligentCore/Graphics/GraphicsEngine/interface/RenderDevice.h"
-#include "../../../../DiligentCore/Graphics/GraphicsTools/interface/RenderStateCache.h"
-#include "../../../../DiligentCore/Graphics/GraphicsTools/interface/ResourceRegistry.hpp"
-#include "../../../../DiligentCore/Common/interface/RefCntAutoPtr.hpp"
-#include "../../../../DiligentCore/Common/interface/BasicMath.hpp"
+#include "DiligentCore/Common/interface/Timer.hpp"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/RenderDevice.h"
+#include "DiligentCore/Graphics/GraphicsTools/interface/RenderStateCache.h"
+#include "DiligentCore/Graphics/GraphicsTools/interface/ResourceRegistry.hpp"
+#include "DiligentCore/Common/interface/RefCntAutoPtr.hpp"
+#include "DiligentCore/Common/interface/BasicMath.hpp"
 
 #include "PostProcess/Common/interface/PostFXContext.hpp"
 #include "PostProcess/Common/interface/PostFXRenderTechnique.hpp"
diff --git a/PostProcess/ScreenSpaceReflection/src/ScreenSpaceReflection.cpp b/PostProcess/ScreenSpaceReflection/src/ScreenSpaceReflection.cpp
index 4c5fca4..6f9af51 100644
--- a/PostProcess/ScreenSpaceReflection/src/ScreenSpaceReflection.cpp
+++ b/PostProcess/ScreenSpaceReflection/src/ScreenSpaceReflection.cpp
@@ -25,15 +25,15 @@
  */
 
 #include "ScreenSpaceReflection.hpp"
-#include "RenderStateCache.hpp"
-#include "CommonlyUsedStates.h"
-#include "GraphicsUtilities.h"
-#include "ScopedDebugGroup.hpp"
-#include "ShaderMacroHelper.hpp"
-#include "GraphicsTypesX.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/RenderStateCache.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/CommonlyUsedStates.h"
+#include "DiligentCore/Graphics/GraphicsTools/interface/GraphicsUtilities.h"
+#include "DiligentCore/Graphics/GraphicsTools/interface/ScopedDebugGroup.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/ShaderMacroHelper.hpp"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/GraphicsTypesX.hpp"
 
 #include "imgui.h"
-#include "ImGuiUtils.hpp"
+#include "DiligentTools/Imgui/interface/ImGuiUtils.hpp"
 
 namespace Diligent
 {
diff --git a/PostProcess/SuperResolution/interface/SuperResolution.hpp b/PostProcess/SuperResolution/interface/SuperResolution.hpp
index 7696f21..5f1892a 100644
--- a/PostProcess/SuperResolution/interface/SuperResolution.hpp
+++ b/PostProcess/SuperResolution/interface/SuperResolution.hpp
@@ -29,10 +29,10 @@
 #include <vector>
 #include <memory>
 
-#include "../../../../DiligentCore/Graphics/GraphicsEngine/interface/RenderDevice.h"
-#include "../../../../DiligentCore/Graphics/GraphicsTools/interface/RenderStateCache.h"
-#include "../../../../DiligentCore/Common/interface/RefCntAutoPtr.hpp"
-#include "../../../../DiligentCore/Common/interface/BasicMath.hpp"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/RenderDevice.h"
+#include "DiligentCore/Graphics/GraphicsTools/interface/RenderStateCache.h"
+#include "DiligentCore/Common/interface/RefCntAutoPtr.hpp"
+#include "DiligentCore/Common/interface/BasicMath.hpp"
 
 #include "PostProcess/Common/interface/PostFXRenderTechnique.hpp"
 #include "PostProcess/Common/interface/PostFXContext.hpp"
diff --git a/PostProcess/SuperResolution/src/SuperResolution.cpp b/PostProcess/SuperResolution/src/SuperResolution.cpp
index 6799b95..7d812f7 100644
--- a/PostProcess/SuperResolution/src/SuperResolution.cpp
+++ b/PostProcess/SuperResolution/src/SuperResolution.cpp
@@ -25,11 +25,11 @@
  */
 
 #include "SuperResolution.hpp"
-#include "CommonlyUsedStates.h"
+#include "DiligentCore/Graphics/GraphicsTools/interface/CommonlyUsedStates.h"
 #include "imgui.h"
-#include "RenderStateCache.hpp"
-#include "ScopedDebugGroup.hpp"
-#include "ShaderMacroHelper.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/RenderStateCache.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/ScopedDebugGroup.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/ShaderMacroHelper.hpp"
 
 namespace Diligent
 {
diff --git a/PostProcess/TemporalAntiAliasing/interface/TemporalAntiAliasing.hpp b/PostProcess/TemporalAntiAliasing/interface/TemporalAntiAliasing.hpp
index 6dcb328..6bb89f1 100644
--- a/PostProcess/TemporalAntiAliasing/interface/TemporalAntiAliasing.hpp
+++ b/PostProcess/TemporalAntiAliasing/interface/TemporalAntiAliasing.hpp
@@ -29,11 +29,11 @@
 #include <unordered_map>
 #include <memory>
 
-#include "../../../../DiligentCore/Graphics/GraphicsEngine/interface/RenderDevice.h"
-#include "../../../../DiligentCore/Graphics/GraphicsTools/interface/RenderStateCache.h"
-#include "../../../../DiligentCore/Graphics/GraphicsTools/interface/ResourceRegistry.hpp"
-#include "../../../../DiligentCore/Common/interface/RefCntAutoPtr.hpp"
-#include "../../../../DiligentCore/Common/interface/BasicMath.hpp"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/RenderDevice.h"
+#include "DiligentCore/Graphics/GraphicsTools/interface/RenderStateCache.h"
+#include "DiligentCore/Graphics/GraphicsTools/interface/ResourceRegistry.hpp"
+#include "DiligentCore/Common/interface/RefCntAutoPtr.hpp"
+#include "DiligentCore/Common/interface/BasicMath.hpp"
 
 #include "PostProcess/Common/interface/PostFXContext.hpp"
 #include "PostProcess/Common/interface/PostFXRenderTechnique.hpp"
diff --git a/PostProcess/TemporalAntiAliasing/src/TemporalAntiAliasing.cpp b/PostProcess/TemporalAntiAliasing/src/TemporalAntiAliasing.cpp
index 570cad6..497abe6 100644
--- a/PostProcess/TemporalAntiAliasing/src/TemporalAntiAliasing.cpp
+++ b/PostProcess/TemporalAntiAliasing/src/TemporalAntiAliasing.cpp
@@ -25,16 +25,16 @@
  */
 
 #include "imgui.h"
-#include "ImGuiUtils.hpp"
+#include "DiligentTools/Imgui/interface/ImGuiUtils.hpp"
 
 #include "TemporalAntiAliasing.hpp"
 
-#include "CommonlyUsedStates.h"
-#include "GraphicsTypesX.hpp"
-#include "MapHelper.hpp"
-#include "RenderStateCache.hpp"
-#include "ScopedDebugGroup.hpp"
-#include "ShaderMacroHelper.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/CommonlyUsedStates.h"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/GraphicsTypesX.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/MapHelper.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/RenderStateCache.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/ScopedDebugGroup.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/ShaderMacroHelper.hpp"
 
 namespace Diligent
 {
diff --git a/Utilities/interface/DiligentFXShaderSourceStreamFactory.hpp b/Utilities/interface/DiligentFXShaderSourceStreamFactory.hpp
index c2b98a8..0ecfd77 100644
--- a/Utilities/interface/DiligentFXShaderSourceStreamFactory.hpp
+++ b/Utilities/interface/DiligentFXShaderSourceStreamFactory.hpp
@@ -27,8 +27,8 @@
 
 #pragma once
 
-#include "Shader.h"
-#include "RefCntAutoPtr.hpp"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/Shader.h"
+#include "DiligentCore/Common/interface/RefCntAutoPtr.hpp"
 
 namespace Diligent
 {
diff --git a/Utilities/src/DiligentFXShaderSourceStreamFactory.cpp b/Utilities/src/DiligentFXShaderSourceStreamFactory.cpp
index e787ea8..9898753 100644
--- a/Utilities/src/DiligentFXShaderSourceStreamFactory.cpp
+++ b/Utilities/src/DiligentFXShaderSourceStreamFactory.cpp
@@ -26,7 +26,7 @@
  */
 
 #include "../interface/DiligentFXShaderSourceStreamFactory.hpp"
-#include "ShaderSourceFactoryUtils.h"
+#include "DiligentCore/Graphics/GraphicsTools/interface/ShaderSourceFactoryUtils.h"
 
 namespace Diligent
 {
