diff --git a/include/webdriverxx/capabilities.h b/include/webdriverxx/capabilities.h
index b23f269..c87ba36 100644
--- a/include/webdriverxx/capabilities.h
+++ b/include/webdriverxx/capabilities.h
@@ -180,6 +180,7 @@ struct Capabilities : JsonObject { // copyable
 	WEBDRIVERXX_PROPERTY(BrowserConnectionEnabled, "browserConnectionEnabled", bool)
 	WEBDRIVERXX_PROPERTY(WebStorageEnabled, "webStorageEnabled", bool)
 	WEBDRIVERXX_PROPERTY(AcceptSslCerts, "acceptSslCerts", bool)
+	WEBDRIVERXX_PROPERTY(AcceptInsecureCerts, "acceptInsecureCerts", bool)
 	WEBDRIVERXX_PROPERTY(Rotatable, "rotatable", bool)
 	WEBDRIVERXX_PROPERTY(NativeEvents, "nativeEvents", bool)
 	WEBDRIVERXX_PROPERTY(Proxy, "proxy", Proxy)
