diff --git a/cartographer/common/configuration_file_resolver.cc b/cartographer/common/configuration_file_resolver.cc
index 9774caf..b8b55ba 100644
--- a/cartographer/common/configuration_file_resolver.cc
+++ b/cartographer/common/configuration_file_resolver.cc
@@ -20,7 +20,6 @@
 #include <iostream>
 #include <streambuf>
 
-#include "cartographer/common/config.h"
 #include "glog/logging.h"
 
 namespace cartographer {
@@ -29,7 +28,6 @@ namespace common {
 ConfigurationFileResolver::ConfigurationFileResolver(
     const std::vector<std::string>& configuration_files_directories)
     : configuration_files_directories_(configuration_files_directories) {
-  configuration_files_directories_.push_back(kConfigurationFilesDirectory);
 }
 
 std::string ConfigurationFileResolver::GetFullPathOrDie(
