diff --git a/source/data/makedata.mak b/source/data/makedata.mak
index 286c18a4d7..3836239ec5 100644
--- a/source/data/makedata.mak
+++ b/source/data/makedata.mak
@@ -33,7 +33,7 @@ ICU_LIB_TARGET=$(DLL_OUTPUT)\$(U_ICUDATA_NAME).dll
 !ENDIF
 !MESSAGE ICU data make path is $(ICUMAKE)
 
-!IF [py -3 -c "exit(0)"]!=0
+!IF [python -c "exit(0)"]!=0
 !MESSAGE Information: Unable to find Python 3. Data will fail to build from source.
 !ENDIF
 
@@ -283,7 +283,7 @@ $(TOOLS_TS): "$(ICUTOOLS)\genrb\$(CFGTOOLS)\genrb.exe" "$(ICUTOOLS)\gencnval\$(C
 $(COREDATA_TS):
 	@cd "$(ICUSRCDATA)"
 	set PYTHONPATH=$(ICUP)\source\python;%PYTHONPATH%
-	py -3 -B -m icutools.databuilder \
+	python -B -m icutools.databuilder \
 		--mode windows-exec \
 		--src_dir "$(ICUSRCDATA)" \
 		--tool_dir "$(ICUTOOLS)" \
diff --git a/source/test/testdata/testdata.mak b/source/test/testdata/testdata.mak
index 2809efd0ca..c20ea05361 100644
--- a/source/test/testdata/testdata.mak
+++ b/source/test/testdata/testdata.mak
@@ -35,7 +35,7 @@ CREATE_DIRS :
 "$(TESTDATAOUT)\testdata.dat" :
 	@echo Building test data
 	set PYTHONPATH=$(ICUP)\source\python;%PYTHONPATH%
-	py -3 -B -m icutools.databuilder \
+	python -B -m icutools.databuilder \
 		--mode windows-exec \
 		--tool_dir "$(ICUTOOLS)" \
 		--tool_cfg "$(CFG)" \
