diff --git a/bindings/cxx/chip.cpp b/bindings/cxx/chip.cpp
index 54c85da..737c14f 100644
--- a/bindings/cxx/chip.cpp
+++ b/bindings/cxx/chip.cpp
@@ -6,6 +6,7 @@
  */
 
 #include <functional>
+#include <gpiod.h>
 #include <gpiod.hpp>
 #include <map>
 #include <system_error>
diff --git a/bindings/cxx/iter.cpp b/bindings/cxx/iter.cpp
index b1acc8c..26291cd 100644
--- a/bindings/cxx/iter.cpp
+++ b/bindings/cxx/iter.cpp
@@ -5,6 +5,7 @@
  * Copyright (C) 2017-2018 Bartosz Golaszewski <bartekgola@gmail.com>
  */
 
+#include <gpiod.h>
 #include <gpiod.hpp>
 #include <system_error>
 
diff --git a/bindings/cxx/line.cpp b/bindings/cxx/line.cpp
index 859ff55..2e9da4f 100644
--- a/bindings/cxx/line.cpp
+++ b/bindings/cxx/line.cpp
@@ -5,6 +5,7 @@
  * Copyright (C) 2017-2018 Bartosz Golaszewski <bartekgola@gmail.com>
  */
 
+#include <gpiod.h>
 #include <gpiod.hpp>
 #include <array>
 #include <map>
diff --git a/bindings/cxx/line_bulk.cpp b/bindings/cxx/line_bulk.cpp
index e77baa2..3c8a5a0 100644
--- a/bindings/cxx/line_bulk.cpp
+++ b/bindings/cxx/line_bulk.cpp
@@ -5,6 +5,7 @@
  * Copyright (C) 2017-2018 Bartosz Golaszewski <bartekgola@gmail.com>
  */
 
+#include <gpiod.h>
 #include <gpiod.hpp>
 #include <map>
 #include <system_error>
