diff --git a/AssetLoader/interface/DXSDKMeshLoader.hpp b/AssetLoader/interface/DXSDKMeshLoader.hpp
index e52c545..c4a50f7 100644
--- a/AssetLoader/interface/DXSDKMeshLoader.hpp
+++ b/AssetLoader/interface/DXSDKMeshLoader.hpp
@@ -29,14 +29,14 @@
 
 #include <vector>
 
-#include "../../../DiligentCore/Primitives/interface/BasicTypes.h"
-#include "../../../DiligentCore/Graphics/GraphicsEngine/interface/RenderDevice.h"
-#include "../../../DiligentCore/Graphics/GraphicsEngine/interface/DeviceContext.h"
-#include "../../../DiligentCore/Graphics/GraphicsEngine/interface/Buffer.h"
-#include "../../../DiligentCore/Graphics/GraphicsEngine/interface/Texture.h"
-#include "../../../DiligentCore/Graphics/GraphicsEngine/interface/TextureView.h"
-#include "../../../DiligentCore/Common/interface/BasicMath.hpp"
-#include "../../../DiligentCore/Common/interface/RefCntAutoPtr.hpp"
+#include "DiligentCore/Primitives/interface/BasicTypes.h"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/RenderDevice.h"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/DeviceContext.h"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/Buffer.h"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/Texture.h"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/TextureView.h"
+#include "DiligentCore/Common/interface/BasicMath.hpp"
+#include "DiligentCore/Common/interface/RefCntAutoPtr.hpp"
 
 namespace Diligent
 {
diff --git a/AssetLoader/interface/GLTFBuilder.hpp b/AssetLoader/interface/GLTFBuilder.hpp
index 2644476..ec166a2 100644
--- a/AssetLoader/interface/GLTFBuilder.hpp
+++ b/AssetLoader/interface/GLTFBuilder.hpp
@@ -33,7 +33,7 @@
 #include <string>
 
 #include "GLTFLoader.hpp"
-#include "GraphicsAccessories.hpp"
+#include "DiligentCore/Graphics/GraphicsAccessories/interface/GraphicsAccessories.hpp"
 
 namespace Diligent
 {
diff --git a/AssetLoader/interface/GLTFLoader.hpp b/AssetLoader/interface/GLTFLoader.hpp
index 3b480f7..6e30f26 100644
--- a/AssetLoader/interface/GLTFLoader.hpp
+++ b/AssetLoader/interface/GLTFLoader.hpp
@@ -38,13 +38,13 @@
 #include <limits>
 #include <algorithm>
 
-#include "../../../DiligentCore/Platforms/interface/PlatformMisc.hpp"
-#include "../../../DiligentCore/Graphics/GraphicsEngine/interface/RenderDevice.h"
-#include "../../../DiligentCore/Graphics/GraphicsEngine/interface/DeviceContext.h"
-#include "../../../DiligentCore/Graphics/GraphicsEngine/interface/GraphicsTypesX.hpp"
-#include "../../../DiligentCore/Common/interface/RefCntAutoPtr.hpp"
-#include "../../../DiligentCore/Common/interface/AdvancedMath.hpp"
-#include "../../../DiligentCore/Common/interface/STDAllocator.hpp"
+#include "DiligentCore/Platforms/interface/PlatformMisc.hpp"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/RenderDevice.h"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/DeviceContext.h"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/GraphicsTypesX.hpp"
+#include "DiligentCore/Common/interface/RefCntAutoPtr.hpp"
+#include "DiligentCore/Common/interface/AdvancedMath.hpp"
+#include "DiligentCore/Common/interface/STDAllocator.hpp"
 #include "GLTFResourceManager.hpp"
 
 namespace tinygltf
diff --git a/AssetLoader/interface/GLTFResourceManager.hpp b/AssetLoader/interface/GLTFResourceManager.hpp
index 3c50661..143f8ea 100644
--- a/AssetLoader/interface/GLTFResourceManager.hpp
+++ b/AssetLoader/interface/GLTFResourceManager.hpp
@@ -32,13 +32,13 @@
 #include <unordered_map>
 #include <atomic>
 
-#include "../../../DiligentCore/Graphics/GraphicsEngine/interface/RenderDevice.h"
-#include "../../../DiligentCore/Graphics/GraphicsEngine/interface/DeviceContext.h"
-#include "../../../DiligentCore/Common/interface/RefCntAutoPtr.hpp"
-#include "../../../DiligentCore/Common/interface/ObjectBase.hpp"
-#include "../../../DiligentCore/Graphics/GraphicsTools/interface/BufferSuballocator.h"
-#include "../../../DiligentCore/Graphics/GraphicsTools/interface/DynamicTextureAtlas.h"
-#include "../../../DiligentCore/Graphics/GraphicsTools/interface/VertexPoolX.hpp"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/RenderDevice.h"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/DeviceContext.h"
+#include "DiligentCore/Common/interface/RefCntAutoPtr.hpp"
+#include "DiligentCore/Common/interface/ObjectBase.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/BufferSuballocator.h"
+#include "DiligentCore/Graphics/GraphicsTools/interface/DynamicTextureAtlas.h"
+#include "DiligentCore/Graphics/GraphicsTools/interface/VertexPoolX.hpp"
 
 namespace Diligent
 {
diff --git a/AssetLoader/src/DXSDKMeshLoader.cpp b/AssetLoader/src/DXSDKMeshLoader.cpp
index ba49ce9..9a4790c 100644
--- a/AssetLoader/src/DXSDKMeshLoader.cpp
+++ b/AssetLoader/src/DXSDKMeshLoader.cpp
@@ -30,11 +30,11 @@
 #include <cfloat>
 
 #include "DXSDKMeshLoader.hpp"
-#include "DataBlobImpl.hpp"
-#include "RefCntAutoPtr.hpp"
-#include "FileWrapper.hpp"
+#include "DiligentCore/Common/interface/DataBlobImpl.hpp"
+#include "DiligentCore/Common/interface/RefCntAutoPtr.hpp"
+#include "DiligentCore/Common/interface/FileWrapper.hpp"
 #include "TextureUtilities.h"
-#include "GraphicsAccessories.hpp"
+#include "DiligentCore/Graphics/GraphicsAccessories/interface/GraphicsAccessories.hpp"
 
 namespace Diligent
 {
diff --git a/AssetLoader/src/GLTFBuilder.cpp b/AssetLoader/src/GLTFBuilder.cpp
index ca92d43..fc22cb6 100644
--- a/AssetLoader/src/GLTFBuilder.cpp
+++ b/AssetLoader/src/GLTFBuilder.cpp
@@ -26,7 +26,7 @@
 
 #include "GLTFBuilder.hpp"
 #include "GLTFLoader.hpp"
-#include "GraphicsAccessories.hpp"
+#include "DiligentCore/Graphics/GraphicsAccessories/interface/GraphicsAccessories.hpp"
 
 namespace Diligent
 {
diff --git a/AssetLoader/src/GLTFLoader.cpp b/AssetLoader/src/GLTFLoader.cpp
index 3eece7f..1cabc3b 100644
--- a/AssetLoader/src/GLTFLoader.cpp
+++ b/AssetLoader/src/GLTFLoader.cpp
@@ -31,20 +31,20 @@
 #include <limits>
 
 #include "GLTFLoader.hpp"
-#include "MapHelper.hpp"
-#include "CommonlyUsedStates.h"
-#include "DataBlobImpl.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/MapHelper.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/CommonlyUsedStates.h"
+#include "DiligentCore/Common/interface/DataBlobImpl.hpp"
 #include "Image.h"
-#include "FileSystem.hpp"
-#include "FileWrapper.hpp"
-#include "GraphicsAccessories.hpp"
+#include "DiligentCore/Platforms/interface/FileSystem.hpp"
+#include "DiligentCore/Common/interface/FileWrapper.hpp"
+#include "DiligentCore/Graphics/GraphicsAccessories/interface/GraphicsAccessories.hpp"
 #include "TextureLoader.h"
 #include "TextureUtilities.h"
-#include "GraphicsUtilities.h"
-#include "Align.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/GraphicsUtilities.h"
+#include "DiligentCore/Common/interface/Align.hpp"
 #include "GLTFBuilder.hpp"
-#include "FixedLinearAllocator.hpp"
-#include "DefaultRawMemoryAllocator.hpp"
+#include "DiligentCore/Common/interface/FixedLinearAllocator.hpp"
+#include "DiligentCore/Common/interface/DefaultRawMemoryAllocator.hpp"
 
 #define TINYGLTF_IMPLEMENTATION
 #define TINYGLTF_NO_STB_IMAGE
@@ -53,7 +53,7 @@
 #if defined(_MSC_VER) && defined(TINYGLTF_ENABLE_DRACO)
 #    pragma warning(disable : 4127) // warning C4127: conditional expression is constant
 #endif
-#include "../../ThirdParty/tinygltf/tiny_gltf.h"
+#include "tiny_gltf.h"
 
 namespace Diligent
 {
diff --git a/AssetLoader/src/GLTFResourceManager.cpp b/AssetLoader/src/GLTFResourceManager.cpp
index 0b1d39f..bdb7f36 100644
--- a/AssetLoader/src/GLTFResourceManager.cpp
+++ b/AssetLoader/src/GLTFResourceManager.cpp
@@ -29,9 +29,9 @@
 
 #include <algorithm>
 
-#include "DefaultRawMemoryAllocator.hpp"
-#include "Align.hpp"
-#include "GraphicsAccessories.hpp"
+#include "DiligentCore/Common/interface/DefaultRawMemoryAllocator.hpp"
+#include "DiligentCore/Common/interface/Align.hpp"
+#include "DiligentCore/Graphics/GraphicsAccessories/interface/GraphicsAccessories.hpp"
 
 namespace Diligent
 {
