diff --git a/Imgui/interface/ImGuiDiligentRenderer.hpp b/Imgui/interface/ImGuiDiligentRenderer.hpp
index f3578fc..3ef3af6 100644
--- a/Imgui/interface/ImGuiDiligentRenderer.hpp
+++ b/Imgui/interface/ImGuiDiligentRenderer.hpp
@@ -28,10 +28,10 @@
 #pragma once
 
 #include <memory>
-#include "../../../DiligentCore/Primitives/interface/BasicTypes.h"
-#include "../../../DiligentCore/Common/interface/BasicMath.hpp"
-#include "../../../DiligentCore/Common/interface/RefCntAutoPtr.hpp"
-#include "../../../DiligentCore/Graphics/GraphicsEngine/interface/GraphicsTypes.h"
+#include "DiligentCore/Primitives/interface/BasicTypes.h"
+#include "DiligentCore/Common/interface/BasicMath.hpp"
+#include "DiligentCore/Common/interface/RefCntAutoPtr.hpp"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/GraphicsTypes.h"
 #include "imgui.h"
 
 struct ImDrawData;
diff --git a/Imgui/interface/ImGuiImplDiligent.hpp b/Imgui/interface/ImGuiImplDiligent.hpp
index 8f66d50..3a35a62 100644
--- a/Imgui/interface/ImGuiImplDiligent.hpp
+++ b/Imgui/interface/ImGuiImplDiligent.hpp
@@ -28,7 +28,7 @@
 #pragma once
 
 #include <memory>
-#include "../../../DiligentCore/Primitives/interface/BasicTypes.h"
+#include "DiligentCore/Primitives/interface/BasicTypes.h"
 
 namespace Diligent
 {
diff --git a/Imgui/interface/ImGuiUtils.hpp b/Imgui/interface/ImGuiUtils.hpp
index 2e8c8b3..26c525b 100644
--- a/Imgui/interface/ImGuiUtils.hpp
+++ b/Imgui/interface/ImGuiUtils.hpp
@@ -33,7 +33,7 @@
 #include <memory>
 #include <vector>
 
-#include "../../../DiligentCore/Platforms/Basic/interface/DebugUtilities.hpp"
+#include "DiligentCore/Platforms/Basic/interface/DebugUtilities.hpp"
 
 namespace ImGui
 {
diff --git a/Imgui/src/ImGuiDiligentRenderer.cpp b/Imgui/src/ImGuiDiligentRenderer.cpp
index 3efc3b0..93fff92 100644
--- a/Imgui/src/ImGuiDiligentRenderer.cpp
+++ b/Imgui/src/ImGuiDiligentRenderer.cpp
@@ -28,10 +28,10 @@
 #include <cstddef>
 #include "ImGuiDiligentRenderer.hpp"
 #include "ImGuiImplDiligent.hpp"
-#include "RenderDevice.h"
-#include "DeviceContext.h"
-#include "MapHelper.hpp"
-#include "GraphicsAccessories.hpp"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/RenderDevice.h"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/DeviceContext.h"
+#include "DiligentCore/Graphics/GraphicsTools/interface/MapHelper.hpp"
+#include "DiligentCore/Graphics/GraphicsAccessories/interface/GraphicsAccessories.hpp"
 
 namespace Diligent
 {
@@ -752,7 +752,8 @@ void ImGuiDiligentRenderer::CreateFontsTexture()
     VERIFY_EXPR(m_pTextureVar != nullptr);
 
     // Store our identifier
-    IO.Fonts->TexID = (ImTextureID)m_pFontSRV;
+    // Fix build https://github.com/DiligentGraphics/DiligentTools/issues/240
+    IO.Fonts->TexID = reinterpret_cast<ImTextureID>(m_pFontSRV.RawPtr());
 }
 
 float4 ImGuiDiligentRenderer::TransformClipRect(const ImVec2& DisplaySize, const float4& rect) const
diff --git a/Imgui/src/ImGuiImplAndroid.cpp b/Imgui/src/ImGuiImplAndroid.cpp
index 435b6a4..996d771 100644
--- a/Imgui/src/ImGuiImplAndroid.cpp
+++ b/Imgui/src/ImGuiImplAndroid.cpp
@@ -28,8 +28,8 @@
 #include "imgui.h"
 
 #include "ImGuiImplAndroid.hpp"
-#include "GraphicsTypes.h"
-#include "DebugUtilities.hpp"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/GraphicsTypes.h"
+#include "DiligentCore/Platforms/Basic/interface/DebugUtilities.hpp"
 
 namespace Diligent
 {
diff --git a/Imgui/src/ImGuiImplDiligent.cpp b/Imgui/src/ImGuiImplDiligent.cpp
index 7ec7560..6b3ee4a 100644
--- a/Imgui/src/ImGuiImplDiligent.cpp
+++ b/Imgui/src/ImGuiImplDiligent.cpp
@@ -29,11 +29,11 @@
 #include "imgui.h"
 #include "ImGuiImplDiligent.hpp"
 #include "ImGuiDiligentRenderer.hpp"
-#include "RenderDevice.h"
-#include "DeviceContext.h"
-#include "RefCntAutoPtr.hpp"
-#include "BasicMath.hpp"
-#include "MapHelper.hpp"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/RenderDevice.h"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/DeviceContext.h"
+#include "DiligentCore/Common/interface/RefCntAutoPtr.hpp"
+#include "DiligentCore/Common/interface/BasicMath.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/MapHelper.hpp"
 
 namespace Diligent
 {
diff --git a/Imgui/src/ImGuiImplLinuxX11.cpp b/Imgui/src/ImGuiImplLinuxX11.cpp
index cb0b262..6cbf3ee 100644
--- a/Imgui/src/ImGuiImplLinuxX11.cpp
+++ b/Imgui/src/ImGuiImplLinuxX11.cpp
@@ -47,7 +47,7 @@
 
 #include "imgui.h"
 
-#include "DebugUtilities.hpp"
+#include "DiligentCore/Platforms/Basic/interface/DebugUtilities.hpp"
 
 namespace Diligent
 {
diff --git a/Imgui/src/ImGuiImplLinuxXCB.cpp b/Imgui/src/ImGuiImplLinuxXCB.cpp
index 3c22933..91684fc 100644
--- a/Imgui/src/ImGuiImplLinuxXCB.cpp
+++ b/Imgui/src/ImGuiImplLinuxXCB.cpp
@@ -32,7 +32,7 @@
 #include <X11/keysym.h>
 #include "xcb_keysyms/xcb_keysyms.h"
 
-#include "DebugUtilities.hpp"
+#include "DiligentCore/Platforms/Basic/interface/DebugUtilities.hpp"
 
 namespace Diligent
 {
diff --git a/Imgui/src/ImGuiImplMacOS.mm b/Imgui/src/ImGuiImplMacOS.mm
index 018fc4a..d9ad5e9 100644
--- a/Imgui/src/ImGuiImplMacOS.mm
+++ b/Imgui/src/ImGuiImplMacOS.mm
@@ -23,7 +23,7 @@
 
 #include "imgui.h"
 #include "ImGuiImplMacOS.hpp"
-#include "../../ThirdParty/imgui_v1.85/imgui_impl_osx_v1.85.h"
+#include "backends/imgui_impl_osx.h"
 #import <Cocoa/Cocoa.h>
 
 namespace Diligent
diff --git a/Imgui/src/ImGuiImplUWP.cpp b/Imgui/src/ImGuiImplUWP.cpp
index 1feac56..852aa95 100644
--- a/Imgui/src/ImGuiImplUWP.cpp
+++ b/Imgui/src/ImGuiImplUWP.cpp
@@ -25,10 +25,10 @@
  *  of the possibility of such damages.
  */
 
-#include "WinHPreface.h"
+#include "DiligentCore/Platforms/Win32/interface/WinHPreface.h"
 #include <wrl.h>
 #include <wrl/client.h>
-#include "WinHPostface.h"
+#include "DiligentCore/Platforms/Win32/interface/WinHPostface.h"
 
 #include "imgui.h"
 #include "ImGuiImplUWP.hpp"
diff --git a/Imgui/src/ImGuiImplWin32.cpp b/Imgui/src/ImGuiImplWin32.cpp
index 4be4d63..ca5baec 100644
--- a/Imgui/src/ImGuiImplWin32.cpp
+++ b/Imgui/src/ImGuiImplWin32.cpp
@@ -25,15 +25,15 @@
  *  of the possibility of such damages.
  */
 
-#include "WinHPreface.h"
+#include "DiligentCore/Platforms/Win32/interface/WinHPreface.h"
 #include <Windows.h>
-#include "WinHPostface.h"
+#include "DiligentCore/Platforms/Win32/interface/WinHPostface.h"
 
-#include "GraphicsTypes.h"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/GraphicsTypes.h"
 #include "imgui.h"
 #include "ImGuiImplWin32.hpp"
 #include "backends/imgui_impl_win32.h"
-#include "DebugUtilities.hpp"
+#include "DiligentCore/Platforms/Basic/interface/DebugUtilities.hpp"
 
 IMGUI_IMPL_API LRESULT ImGui_ImplWin32_WndProcHandler(HWND hwnd, UINT msg, WPARAM wParam, LPARAM lParam);
 
