diff --git a/RenderStatePackager/include/ParsingEnvironment.hpp b/RenderStatePackager/include/ParsingEnvironment.hpp
index 496253e..b96fae2 100644
--- a/RenderStatePackager/include/ParsingEnvironment.hpp
+++ b/RenderStatePackager/include/ParsingEnvironment.hpp
@@ -29,10 +29,10 @@
 #include <vector>
 #include <memory>
 
-#include "ThreadPool.hpp"
-#include "RefCntAutoPtr.hpp"
-#include "ArchiverFactory.h"
-#include "ArchiverFactoryLoader.h"
+#include "DiligentCore/Common/interface/ThreadPool.hpp"
+#include "DiligentCore/Common/interface/RefCntAutoPtr.hpp"
+#include "DiligentCore/Graphics/Archiver/interface/ArchiverFactory.h"
+#include "DiligentCore/Graphics/Archiver/interface/ArchiverFactoryLoader.h"
 #include "RenderStatePackager.hpp"
 
 namespace Diligent
diff --git a/RenderStatePackager/include/RenderStatePackager.hpp b/RenderStatePackager/include/RenderStatePackager.hpp
index e3e132d..61cfe3b 100644
--- a/RenderStatePackager/include/RenderStatePackager.hpp
+++ b/RenderStatePackager/include/RenderStatePackager.hpp
@@ -29,12 +29,12 @@
 #include <vector>
 #include <unordered_map>
 
-#include "Archiver.h"
-#include "ThreadPool.hpp"
-#include "RefCntAutoPtr.hpp"
-#include "SerializationDevice.h"
+#include "DiligentCore/Graphics/Archiver/interface/Archiver.h"
+#include "DiligentCore/Common/interface/ThreadPool.hpp"
+#include "DiligentCore/Common/interface/RefCntAutoPtr.hpp"
+#include "DiligentCore/Graphics/Archiver/interface/SerializationDevice.h"
 #include "RenderStateNotationParser.h"
-#include "HashUtils.hpp"
+#include "DiligentCore/Common/interface/HashUtils.hpp"
 
 namespace Diligent
 {
diff --git a/RenderStatePackager/src/ParsingEnvironment.cpp b/RenderStatePackager/src/ParsingEnvironment.cpp
index fc6a016..2a8f68e 100644
--- a/RenderStatePackager/src/ParsingEnvironment.cpp
+++ b/RenderStatePackager/src/ParsingEnvironment.cpp
@@ -25,10 +25,10 @@
  */
 
 #include "ParsingEnvironment.hpp"
-#include "DynamicLinearAllocator.hpp"
-#include "DefaultRawMemoryAllocator.hpp"
-#include "DataBlobImpl.hpp"
-#include "FileWrapper.hpp"
+#include "DiligentCore/Common/interface/DynamicLinearAllocator.hpp"
+#include "DiligentCore/Common/interface/DefaultRawMemoryAllocator.hpp"
+#include "DiligentCore/Common/interface/DataBlobImpl.hpp"
+#include "DiligentCore/Common/interface/FileWrapper.hpp"
 
 namespace Diligent
 {
diff --git a/RenderStatePackager/src/RenderStatePackager.cpp b/RenderStatePackager/src/RenderStatePackager.cpp
index 9499958..3e777fa 100644
--- a/RenderStatePackager/src/RenderStatePackager.cpp
+++ b/RenderStatePackager/src/RenderStatePackager.cpp
@@ -29,13 +29,13 @@
 #include <deque>
 #include <sstream>
 
-#include "GraphicsAccessories.hpp"
-#include "BasicMath.hpp"
-#include "DefaultRawMemoryAllocator.hpp"
-#include "DynamicLinearAllocator.hpp"
-#include "SerializedPipelineState.h"
-#include "FileSystem.hpp"
-#include "FileWrapper.hpp"
+#include "DiligentCore/Graphics/GraphicsAccessories/interface/GraphicsAccessories.hpp"
+#include "DiligentCore/Common/interface/BasicMath.hpp"
+#include "DiligentCore/Common/interface/DefaultRawMemoryAllocator.hpp"
+#include "DiligentCore/Common/interface/DynamicLinearAllocator.hpp"
+#include "DiligentCore/Graphics/Archiver/interface/SerializedPipelineState.h"
+#include "DiligentCore/Platforms/interface/FileSystem.hpp"
+#include "DiligentCore/Common/interface/FileWrapper.hpp"
 
 namespace Diligent
 {
diff --git a/RenderStatePackager/src/main.cpp b/RenderStatePackager/src/main.cpp
index 759ae30..c15d5fb 100644
--- a/RenderStatePackager/src/main.cpp
+++ b/RenderStatePackager/src/main.cpp
@@ -24,7 +24,7 @@
  *  of the possibility of such damages.
  */
 
-#include "FileWrapper.hpp"
+#include "DiligentCore/Common/interface/FileWrapper.hpp"
 #include "RenderStateNotationParser.h"
 #include "ParsingEnvironment.hpp"
 #include "args.hxx"
