diff --git a/RenderStateNotation/include/RenderStateNotationLoaderImpl.hpp b/RenderStateNotation/include/RenderStateNotationLoaderImpl.hpp
index d9636c3..ff6233a 100644
--- a/RenderStateNotation/include/RenderStateNotationLoaderImpl.hpp
+++ b/RenderStateNotation/include/RenderStateNotationLoaderImpl.hpp
@@ -29,10 +29,10 @@
 #include <unordered_map>
 
 #include "RenderStateNotationLoader.h"
-#include "RefCntAutoPtr.hpp"
-#include "ObjectBase.hpp"
-#include "HashUtils.hpp"
-#include "RenderStateCache.hpp"
+#include "DiligentCore/Common/interface/RefCntAutoPtr.hpp"
+#include "DiligentCore/Common/interface/ObjectBase.hpp"
+#include "DiligentCore/Common/interface/HashUtils.hpp"
+#include "DiligentCore/Graphics/GraphicsTools/interface/RenderStateCache.hpp"
 
 namespace Diligent
 {
diff --git a/RenderStateNotation/include/RenderStateNotationParserImpl.hpp b/RenderStateNotation/include/RenderStateNotationParserImpl.hpp
index 92294b2..b748bc0 100644
--- a/RenderStateNotation/include/RenderStateNotationParserImpl.hpp
+++ b/RenderStateNotation/include/RenderStateNotationParserImpl.hpp
@@ -30,10 +30,10 @@
 #include <unordered_set>
 
 #include "RenderStateNotationParser.h"
-#include "RefCntAutoPtr.hpp"
-#include "ObjectBase.hpp"
-#include "DynamicLinearAllocator.hpp"
-#include "HashUtils.hpp"
+#include "DiligentCore/Common/interface/RefCntAutoPtr.hpp"
+#include "DiligentCore/Common/interface/ObjectBase.hpp"
+#include "DiligentCore/Common/interface/DynamicLinearAllocator.hpp"
+#include "DiligentCore/Common/interface/HashUtils.hpp"
 
 namespace Diligent
 {
diff --git a/RenderStateNotation/include/pch.h b/RenderStateNotation/include/pch.h
index 2da62b8..23eb430 100644
--- a/RenderStateNotation/include/pch.h
+++ b/RenderStateNotation/include/pch.h
@@ -26,12 +26,12 @@
 
 #pragma once
 
-#include "json.hpp"
+#include "nlohmann/json.hpp"
 
-#include "BasicMath.hpp"
-#include "RenderDevice.h"
-#include "DynamicLinearAllocator.hpp"
-#include "StringTools.hpp"
+#include "DiligentCore/Common/interface/BasicMath.hpp"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/RenderDevice.h"
+#include "DiligentCore/Common/interface/DynamicLinearAllocator.hpp"
+#include "DiligentCore/Common/interface/StringTools.hpp"
 
 #include "generated/CommonParser.hpp"
 #include "generated/GraphicsTypesParser.hpp"
diff --git a/RenderStateNotation/interface/RenderStateNotationLoader.h b/RenderStateNotation/interface/RenderStateNotationLoader.h
index a53f741..e8aefe7 100644
--- a/RenderStateNotation/interface/RenderStateNotationLoader.h
+++ b/RenderStateNotation/interface/RenderStateNotationLoader.h
@@ -31,12 +31,12 @@
 /// \file
 /// Defines Diligent::IRenderStateNotationLoader interface
 #include "RenderStateNotationParser.h"
-#include "../../../DiligentCore/Graphics/GraphicsTools/interface/RenderStateCache.h"
+#include "DiligentCore/Graphics/GraphicsTools/interface/RenderStateCache.h"
 
 
 DILIGENT_BEGIN_NAMESPACE(Diligent)
 
-#include "../../../DiligentCore/Primitives/interface/DefineRefMacro.h"
+#include "DiligentCore/Primitives/interface/DefineRefMacro.h"
 
 /// Render state notation loader initialization info.
 struct RenderStateNotationLoaderCreateInfo
@@ -196,13 +196,13 @@ typedef struct LoadPipelineStateInfo LoadPipelineStateInfo;
 
 // clang-format on
 
-#include "../../../DiligentCore/Primitives/interface/UndefRefMacro.h"
+#include "DiligentCore/Primitives/interface/UndefRefMacro.h"
 
 // {FD9B12C5-3BC5-4729-A2B4-924DF374B3D3}
 static DILIGENT_CONSTEXPR INTERFACE_ID IID_RenderStateNotationLoader = {0xFD9B12C5, 0x3BC5, 0x4729, {0xA2, 0xB4, 0x92, 0x4D, 0xF3, 0x74, 0xB3, 0xD3}};
 
 #define DILIGENT_INTERFACE_NAME IRenderStateNotationLoader
-#include "../../../DiligentCore/Primitives/interface/DefineInterfaceHelperMacros.h"
+#include "DiligentCore/Primitives/interface/DefineInterfaceHelperMacros.h"
 
 #define IRenderStateNotationLoaderInclusiveMethods \
     IObjectInclusiveMethods;                       \
@@ -269,7 +269,7 @@ DILIGENT_BEGIN_INTERFACE(IRenderStateNotationLoader, IObject)
 };
 DILIGENT_END_INTERFACE
 
-#include "../../../DiligentCore/Primitives/interface/UndefInterfaceHelperMacros.h"
+#include "DiligentCore/Primitives/interface/UndefInterfaceHelperMacros.h"
 
 #if DILIGENT_C_INTERFACE
 
@@ -283,11 +283,11 @@ DILIGENT_END_INTERFACE
 
 #endif
 
-#include "../../../DiligentCore/Primitives/interface/DefineGlobalFuncHelperMacros.h"
+#include "DiligentCore/Primitives/interface/DefineGlobalFuncHelperMacros.h"
 
 void DILIGENT_GLOBAL_FUNCTION(CreateRenderStateNotationLoader)(const RenderStateNotationLoaderCreateInfo REF CreateInfo,
                                                                IRenderStateNotationLoader**                  ppLoader);
 
-#include "../../../DiligentCore/Primitives/interface/UndefGlobalFuncHelperMacros.h"
+#include "DiligentCore/Primitives/interface/UndefGlobalFuncHelperMacros.h"
 
 DILIGENT_END_NAMESPACE // namespace Diligent
diff --git a/RenderStateNotation/interface/RenderStateNotationParser.h b/RenderStateNotation/interface/RenderStateNotationParser.h
index c6a887d..59c418b 100644
--- a/RenderStateNotation/interface/RenderStateNotationParser.h
+++ b/RenderStateNotation/interface/RenderStateNotationParser.h
@@ -30,7 +30,7 @@
 
 /// \file
 /// Defines Diligent::IRenderStateNotationParser interface
-#include "../../../DiligentCore/Graphics/GraphicsEngine/interface/RenderDevice.h"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/RenderDevice.h"
 
 DILIGENT_BEGIN_NAMESPACE(Diligent)
 
@@ -392,7 +392,7 @@ typedef struct RenderStateNotationParserCreateInfo RenderStateNotationParserCrea
 static DILIGENT_CONSTEXPR INTERFACE_ID IID_RenderStateNotationParser = {0x355AC9F7, 0x5D9D, 0x423D, {0xAE, 0x35, 0x80, 0xE0, 0x02, 0x8D, 0xE1, 0x7E}};
 
 #define DILIGENT_INTERFACE_NAME IRenderStateNotationParser
-#include "../../../DiligentCore/Primitives/interface/DefineInterfaceHelperMacros.h"
+#include "DiligentCore/Primitives/interface/DefineInterfaceHelperMacros.h"
 
 #define IRenderStateNotationParserInclusiveMethods \
     IObjectInclusiveMethods;                       \
@@ -555,7 +555,7 @@ DILIGENT_BEGIN_INTERFACE(IRenderStateNotationParser, IObject)
 };
 DILIGENT_END_INTERFACE
 
-#include "../../../DiligentCore/Primitives/interface/UndefInterfaceHelperMacros.h"
+#include "DiligentCore/Primitives/interface/UndefInterfaceHelperMacros.h"
 
 #if DILIGENT_C_INTERFACE
 
@@ -577,12 +577,12 @@ DILIGENT_END_INTERFACE
 
 #endif
 
-#include "../../../DiligentCore/Primitives/interface/DefineGlobalFuncHelperMacros.h"
+#include "DiligentCore/Primitives/interface/DefineGlobalFuncHelperMacros.h"
 
 void DILIGENT_GLOBAL_FUNCTION(CreateRenderStateNotationParser)(const RenderStateNotationParserCreateInfo REF CreateInfo,
                                                                IRenderStateNotationParser**                  pParser);
 
 
-#include "../../../DiligentCore/Primitives/interface/UndefGlobalFuncHelperMacros.h"
+#include "DiligentCore/Primitives/interface/UndefGlobalFuncHelperMacros.h"
 
 DILIGENT_END_NAMESPACE // namespace Diligent
diff --git a/RenderStateNotation/src/RenderStateNotationLoaderImpl.cpp b/RenderStateNotation/src/RenderStateNotationLoaderImpl.cpp
index 00dbb73..5961798 100644
--- a/RenderStateNotation/src/RenderStateNotationLoaderImpl.cpp
+++ b/RenderStateNotation/src/RenderStateNotationLoaderImpl.cpp
@@ -25,9 +25,9 @@
  */
 
 #include "RenderStateNotationLoaderImpl.hpp"
-#include "DefaultRawMemoryAllocator.hpp"
-#include "CallbackWrapper.hpp"
-#include "DynamicLinearAllocator.hpp"
+#include "DiligentCore/Common/interface/DefaultRawMemoryAllocator.hpp"
+#include "DiligentCore/Common/interface/CallbackWrapper.hpp"
+#include "DiligentCore/Common/interface/DynamicLinearAllocator.hpp"
 
 namespace Diligent
 {
diff --git a/RenderStateNotation/src/RenderStateNotationParserImpl.cpp b/RenderStateNotation/src/RenderStateNotationParserImpl.cpp
index 22775b5..49e6045 100644
--- a/RenderStateNotation/src/RenderStateNotationParserImpl.cpp
+++ b/RenderStateNotation/src/RenderStateNotationParserImpl.cpp
@@ -32,10 +32,10 @@
 #include <functional>
 #include <array>
 
-#include "DataBlobImpl.hpp"
-#include "FileWrapper.hpp"
-#include "DefaultRawMemoryAllocator.hpp"
-#include "GraphicsAccessories.hpp"
+#include "DiligentCore/Common/interface/DataBlobImpl.hpp"
+#include "DiligentCore/Common/interface/FileWrapper.hpp"
+#include "DiligentCore/Common/interface/DefaultRawMemoryAllocator.hpp"
+#include "DiligentCore/Graphics/GraphicsAccessories/interface/GraphicsAccessories.hpp"
 
 namespace Diligent
 {
