diff --git a/Tests/DiligentToolsGPUTest/src/RenderStateNotationLoaderTest.cpp b/Tests/DiligentToolsGPUTest/src/RenderStateNotationLoaderTest.cpp
index b38dd3d..c29b367 100644
--- a/Tests/DiligentToolsGPUTest/src/RenderStateNotationLoaderTest.cpp
+++ b/Tests/DiligentToolsGPUTest/src/RenderStateNotationLoaderTest.cpp
@@ -25,7 +25,7 @@
  */
 
 #include "gtest/gtest.h"
-#include "RefCntAutoPtr.hpp"
+#include "DiligentCore/Common/interface/RefCntAutoPtr.hpp"
 #include "RenderStateNotationLoader.h"
 #include "DefaultShaderSourceStreamFactory.h"
 #include "RenderStateCache.h"
diff --git a/Tests/DiligentToolsGPUTest/src/RenderStatePackagerGPUTest.cpp b/Tests/DiligentToolsGPUTest/src/RenderStatePackagerGPUTest.cpp
index 021a787..096a8c2 100644
--- a/Tests/DiligentToolsGPUTest/src/RenderStatePackagerGPUTest.cpp
+++ b/Tests/DiligentToolsGPUTest/src/RenderStatePackagerGPUTest.cpp
@@ -25,7 +25,7 @@
  */
 
 #include "gtest/gtest.h"
-#include "RefCntAutoPtr.hpp"
+#include "DiligentCore/Common/interface/RefCntAutoPtr.hpp"
 #include "DefaultShaderSourceStreamFactory.h"
 #include "RenderStatePackager.hpp"
 #include "ParsingEnvironment.hpp"
diff --git a/Tests/DiligentToolsTest/include/DRSNLoader.hpp b/Tests/DiligentToolsTest/include/DRSNLoader.hpp
index 9a1882a..fb3b8a1 100644
--- a/Tests/DiligentToolsTest/include/DRSNLoader.hpp
+++ b/Tests/DiligentToolsTest/include/DRSNLoader.hpp
@@ -26,13 +26,13 @@
 
 #include <type_traits>
 
-#include "json.hpp"
-#include "BasicMath.hpp"
-#include "DynamicLinearAllocator.hpp"
-#include "FileWrapper.hpp"
-#include "DataBlobImpl.hpp"
-#include "DefaultRawMemoryAllocator.hpp"
-#include "RenderDevice.h"
+#include "nlohmann/json.hpp"
+#include "DiligentCore/Common/interface/BasicMath.hpp"
+#include "DiligentCore/Common/interface/DynamicLinearAllocator.hpp"
+#include "DiligentCore/Common/interface/FileWrapper.hpp"
+#include "DiligentCore/Common/interface/DataBlobImpl.hpp"
+#include "DiligentCore/Common/interface/DefaultRawMemoryAllocator.hpp"
+#include "DiligentCore/Graphics/GraphicsEngine/interface/RenderDevice.h"
 #include "generated/CommonParser.hpp"
 #include "generated/GraphicsTypesParser.hpp"
 #include "generated/BlendStateParser.hpp"
diff --git a/Tests/DiligentToolsTest/src/CommandLineParserTest.cpp b/Tests/DiligentToolsTest/src/CommandLineParserTest.cpp
index 3b6f259..523b488 100644
--- a/Tests/DiligentToolsTest/src/CommandLineParserTest.cpp
+++ b/Tests/DiligentToolsTest/src/CommandLineParserTest.cpp
@@ -24,7 +24,7 @@
  *  of the possibility of such damages.
  */
 
-#include "../../../NativeApp/include/CommandLineParser.hpp"
+#include "NativeApp/include/CommandLineParser.hpp"
 
 #include "gtest/gtest.h"
 
diff --git a/Tests/DiligentToolsTest/src/JPEGCodecTest.cpp b/Tests/DiligentToolsTest/src/JPEGCodecTest.cpp
index 324d06c..a9facb7 100644
--- a/Tests/DiligentToolsTest/src/JPEGCodecTest.cpp
+++ b/Tests/DiligentToolsTest/src/JPEGCodecTest.cpp
@@ -29,7 +29,7 @@
 
 #include "gtest/gtest.h"
 
-#include "DataBlobImpl.hpp"
+#include "DiligentCore/Common/interface/DataBlobImpl.hpp"
 
 #include <cmath>
 
diff --git a/Tests/DiligentToolsTest/src/PNGCodecTest.cpp b/Tests/DiligentToolsTest/src/PNGCodecTest.cpp
index ac4da10..254404f 100644
--- a/Tests/DiligentToolsTest/src/PNGCodecTest.cpp
+++ b/Tests/DiligentToolsTest/src/PNGCodecTest.cpp
@@ -32,7 +32,7 @@
 
 #include <vector>
 
-#include "DataBlobImpl.hpp"
+#include "DiligentCore/Common/interface/DataBlobImpl.hpp"
 
 using namespace Diligent;
 
diff --git a/Tests/DiligentToolsTest/src/RenderStateNotationParser/RenderStateNotationParserTest.cpp b/Tests/DiligentToolsTest/src/RenderStateNotationParser/RenderStateNotationParserTest.cpp
index 7c73030..11b225e 100644
--- a/Tests/DiligentToolsTest/src/RenderStateNotationParser/RenderStateNotationParserTest.cpp
+++ b/Tests/DiligentToolsTest/src/RenderStateNotationParser/RenderStateNotationParserTest.cpp
@@ -25,7 +25,7 @@
  */
 
 #include "gtest/gtest.h"
-#include "RefCntAutoPtr.hpp"
+#include "DiligentCore/Common/interface/RefCntAutoPtr.hpp"
 #include "RenderStateNotationParser.h"
 #include "DefaultShaderSourceStreamFactory.h"
 #include "TestingEnvironment.hpp"
diff --git a/Tests/DiligentToolsTest/src/RenderStatePackager/RenderStatePackagerTest.cpp b/Tests/DiligentToolsTest/src/RenderStatePackager/RenderStatePackagerTest.cpp
index 6d2378d..618e4b8 100644
--- a/Tests/DiligentToolsTest/src/RenderStatePackager/RenderStatePackagerTest.cpp
+++ b/Tests/DiligentToolsTest/src/RenderStatePackager/RenderStatePackagerTest.cpp
@@ -32,9 +32,9 @@
 #include "RenderStatePackager.hpp"
 #include "ParsingEnvironment.hpp"
 #include "TestingEnvironment.hpp"
-#include "FileSystem.hpp"
-#include "BasicMath.hpp"
-#include "GraphicsAccessories.hpp"
+#include "DiligentCore/Platforms/interface/FileSystem.hpp"
+#include "DiligentCore/Common/interface/BasicMath.hpp"
+#include "DiligentCore/Graphics/GraphicsAccessories/interface/GraphicsAccessories.hpp"
 
 using namespace Diligent;
 using namespace Diligent::Testing;
diff --git a/Tests/DiligentToolsTest/src/TextureUtilitiesTest.cpp b/Tests/DiligentToolsTest/src/TextureUtilitiesTest.cpp
index bfed06c..2b36469 100644
--- a/Tests/DiligentToolsTest/src/TextureUtilitiesTest.cpp
+++ b/Tests/DiligentToolsTest/src/TextureUtilitiesTest.cpp
@@ -30,7 +30,7 @@
 
 #include "gtest/gtest.h"
 
-#include "GraphicsAccessories.hpp"
+#include "DiligentCore/Graphics/GraphicsAccessories/interface/GraphicsAccessories.hpp"
 
 using namespace Diligent;
 
diff --git a/Tests/IncludeTest/BaseStructAlignmentTest.cpp b/Tests/IncludeTest/BaseStructAlignmentTest.cpp
index 26bad65..541901d 100644
--- a/Tests/IncludeTest/BaseStructAlignmentTest.cpp
+++ b/Tests/IncludeTest/BaseStructAlignmentTest.cpp
@@ -24,7 +24,7 @@
  *  of the possibility of such damages.
  */
 
-#include "../../../DiligentCore/Primitives/interface/CheckBaseStructAlignment.hpp"
+#include "DiligentCore/Primitives/interface/CheckBaseStructAlignment.hpp"
 #include "RenderStateNotation/interface/RenderStateNotationParser.h"
 
 namespace Diligent
