diff --git a/CMakeLists.txt b/CMakeLists.txt
index ee48f6c..7eb7dde 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -216,6 +216,12 @@ if (BUILD_SHARED_LIBS)
   if (MSVC)
     add_definitions (-DSOXR_DLL)
   endif ()
+else ()
+  if (MINGW)
+    add_definitions (-DSOXR_DLL)
+    add_definitions (-Dsoxr_EXPORTS)
+    add_definitions (-Dsoxr_lsr_EXPORTS)
+  endif ()
 endif ()
 
 if (CMAKE_BUILD_TYPE STREQUAL "None") # As used by some distros.
