diff --git a/cmake/abseil-cpp.cmake b/cmake/abseil-cpp.cmake
index a4e9d22..5e5d4d5 100644
--- a/cmake/abseil-cpp.cmake
+++ b/cmake/abseil-cpp.cmake
@@ -37,7 +37,7 @@ elseif(protobuf_ABSL_PROVIDER STREQUAL "package")
 endif()
 set(_protobuf_FIND_ABSL "if(NOT TARGET absl::strings)\n  find_package(absl CONFIG)\nendif()")
 
-if (BUILD_SHARED_LIBS AND MSVC)
+if (FALSE)
   # On MSVC Abseil is bundled into a single DLL.
   # This condition is necessary as of abseil 20230125.3 when abseil is consumed via add_subdirectory,
   # the abseil_dll target  is named abseil_dll, while if abseil is consumed via find_package, the target
