diff --git a/CMakeLists.txt b/CMakeLists.txt
index 303fcae..0a80adb 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -8,7 +8,8 @@ SET(VERSION_MAJOR 4)
 SET(VERSION_MINOR 0)
 SET(FTGL_SOVERSION 1)
 
-FIND_PACKAGE(Freetype REQUIRED) # if it fails, check this: https://bugs.launchpad.net/ubuntu/+source/cmake/+bug/826988
+include(FindPkgConfig)
+pkg_check_modules(FREETYPE REQUIRED freetype2)
 FIND_PACKAGE(OpenGL   REQUIRED)
 
 INCLUDE_DIRECTORIES(${PROJECT_SOURCE_DIR}/src)
