diff --git a/aui.core/CMakeLists.txt b/aui.core/CMakeLists.txt
index 6aa1b07..5cb497f 100644
--- a/aui.core/CMakeLists.txt
+++ b/aui.core/CMakeLists.txt
@@ -6,7 +6,9 @@ option(AUI_ENABLE_ASAN "Enable AddressSanitizer" OFF)
 option(AUI_ENABLE_DEATH_TESTS "Enable GTest death tests" ON)
 
 aui_module(aui.core EXPORT aui)
-aui_enable_tests(aui.core)
+if (NOT AUIB_DISABLE)
+    aui_enable_tests(aui.core)
+endif ()
 aui_enable_benchmarks(aui.core)
 
 
@@ -49,8 +51,12 @@ if (NOT WIN32)
     endif()
 
     if (AUI_PLATFORM_LINUX)
-        auib_import(backtrace https://github.com/aui-framework/libbacktrace
-                    VERSION a5a32e3cc8e22a9fc40689a884b032c52ee3f88e)
+        if (AUIB_DISABLE)
+            find_package(libbacktrace CONFIG REQUIRED)
+        else ()
+            auib_import(backtrace https://github.com/aui-framework/libbacktrace
+                        VERSION a5a32e3cc8e22a9fc40689a884b032c52ee3f88e)
+        endif ()
         set(_use_backtrace TRUE)
     else()
 
@@ -84,13 +90,17 @@ endif()
 
 if(_use_backtrace)
     message(STATUS "Stacktrace backend: backtrace")
-    aui_link(aui.core PRIVATE backtrace)
+    if (AUIB_DISABLE)
+        aui_link(aui.core PRIVATE libbacktrace::libbacktrace)
+    else ()
+        aui_link(aui.core PRIVATE backtrace)
+    endif ()
     target_compile_definitions(aui.core PRIVATE AUI_USE_BACKTRACE=1)
 elseif(_use_unwind)
     message(STATUS "Stacktrace backend: unwind")
     target_compile_definitions(aui.core PRIVATE AUI_USE_UNWIND=1)
 elseif(WIN32)
-    aui_link(aui.core PRIVATE dbghelp shlwapi.lib)
+    aui_link(aui.core PRIVATE dbghelp shell32 shlwapi.lib)
     message(STATUS "Stacktrace backend: dbghelp")
 else()
     message(STATUS "Stacktrace backend: none")
@@ -146,17 +156,23 @@ if (AUI_SHARED_PTR_FIND_INSTANCES)
 endif()
 
 
-# [auib_import examples]
-auib_import(fmt https://github.com/fmtlib/fmt
-            VERSION 9.1.0
-            CMAKE_ARGS -DFMT_TEST=FALSE -DFMT_DOC=FALSE)
+if (AUIB_DISABLE)
+    find_package(fmt CONFIG REQUIRED)
+    find_package(range-v3 CONFIG REQUIRED)
+    aui_link(aui.core PUBLIC fmt::fmt range-v3::range-v3)
+else ()
+    # [auib_import examples]
+    auib_import(fmt https://github.com/fmtlib/fmt
+                VERSION 9.1.0
+                CMAKE_ARGS -DFMT_TEST=FALSE -DFMT_DOC=FALSE)
 
-auib_import(range-v3 https://github.com/ericniebler/range-v3
-        VERSION 0.12.0
-        CMAKE_ARGS -DRANGE_V3_DOCS=OFF -DRANGE_V3_TESTS=OFF -DRANGE_V3_EXAMPLES=OFF -DRANGE_V3_PERF=OFF -DRANGE_V3_HEADER_CHECKS=OFF)
-# [auib_import examples]
+    auib_import(range-v3 https://github.com/ericniebler/range-v3
+            VERSION 0.12.0
+            CMAKE_ARGS -DRANGE_V3_DOCS=OFF -DRANGE_V3_TESTS=OFF -DRANGE_V3_EXAMPLES=OFF -DRANGE_V3_PERF=OFF -DRANGE_V3_HEADER_CHECKS=OFF)
+    # [auib_import examples]
 
-aui_link(aui.core PUBLIC fmt::fmt-header-only range-v3::range-v3)
+    aui_link(aui.core PUBLIC fmt::fmt-header-only range-v3::range-v3)
+endif ()
 
 # segfault functionality
 if (NOT MSVC AND AUI_CATCH_UNHANDLED)
