diff --git a/aui.audio/CMakeLists.txt b/aui.audio/CMakeLists.txt
index 21e441538..96d151626 100644
--- a/aui.audio/CMakeLists.txt
+++ b/aui.audio/CMakeLists.txt
@@ -5,26 +5,43 @@ add_subdirectory(3rdparty/ogg)
 aui_link(aui.audio PUBLIC aui::core)
 
 if (AUI_PLATFORM_ANDROID)
-    auib_import(oboe https://github.com/google/oboe ADD_SUBDIRECTORY VERSION main)
-    aui_link(aui.audio PUBLIC oboe $<LINK_ONLY:log> $<LINK_ONLY:OpenSLES>)
+    if (AUIB_DISABLE)
+        find_package(oboe CONFIG REQUIRED)
+        aui_link(aui.audio PUBLIC oboe::oboe $<LINK_ONLY:log> $<LINK_ONLY:OpenSLES>)
+    else ()
+        auib_import(oboe https://github.com/google/oboe ADD_SUBDIRECTORY VERSION main)
+        aui_link(aui.audio PUBLIC oboe $<LINK_ONLY:log> $<LINK_ONLY:OpenSLES>)
 
-    install(
-            TARGETS oboe
-            EXPORT aui
-            ARCHIVE       DESTINATION "aui.audio/lib"
-            LIBRARY       DESTINATION "aui.audio/lib"
-            RUNTIME       DESTINATION "aui.audio/bin"
-            PUBLIC_HEADER DESTINATION "aui.audio/include"
-            INCLUDES      DESTINATION "aui.audio/include"
-    )
+        install(
+                TARGETS oboe
+                EXPORT aui
+                ARCHIVE       DESTINATION "aui.audio/lib"
+                LIBRARY       DESTINATION "aui.audio/lib"
+                RUNTIME       DESTINATION "aui.audio/bin"
+                PUBLIC_HEADER DESTINATION "aui.audio/include"
+                INCLUDES      DESTINATION "aui.audio/include"
+        )
+    endif ()
 elseif (AUI_PLATFORM_LINUX)
-    aui_link(aui.audio PUBLIC pulse)
+    if (AUIB_DISABLE)
+        include(FindPkgConfig)
+        pkg_check_modules(PulseAudio REQUIRED IMPORTED_TARGET libpulse)
+        pkg_check_modules(PulseAudioSimple REQUIRED IMPORTED_TARGET libpulse-simple)
+        aui_link(aui.audio PRIVATE PkgConfig::PulseAudio PkgConfig::PulseAudioSimple)
+    else ()
+        aui_link(aui.audio PUBLIC pulse)
+    endif ()
 elseif (AUI_PLATFORM_WIN)
-    aui_link(aui.audio PUBLIC Winmm dsound.lib dxguid.lib)
+    aui_link(aui.audio PUBLIC Winmm dsound dxguid)
 endif()
 
-auib_import(Opus https://github.com/xiph/opus.git VERSION v1.4)
-aui_link(aui.audio PRIVATE Opus::opus)
+if (AUIB_DISABLE)
+    find_package(Opus CONFIG REQUIRED)
+    aui_link(aui.audio PRIVATE Opus::opus)
+else ()
+    auib_import(Opus https://github.com/xiph/opus.git VERSION v1.4)
+    aui_link(aui.audio PRIVATE Opus::opus)
+endif ()
 
 if (AUI_PLATFORM_IOS OR AUI_PLATFORM_MACOS)
     auib_use_system_libs_begin()
@@ -42,9 +59,15 @@ if (AUI_PLATFORM_MACOS)
 					   "-framework QuartzCore")
 endif()
 
-auib_import(soxr https://github.com/aui-framework/soxr
-            ADD_SUBDIRECTORY
-            VERSION adacc686124ac3568432f3dca388878963a005ee
-            )
+if (AUIB_DISABLE)
+    include(FindPkgConfig)
+    pkg_check_modules(soxr REQUIRED IMPORTED_TARGET soxr)
+    aui_link(aui.audio PRIVATE PkgConfig::soxr)
+else ()
+    auib_import(soxr https://github.com/aui-framework/soxr
+                ADD_SUBDIRECTORY
+                VERSION adacc686124ac3568432f3dca388878963a005ee
+                )
 
-aui_link(aui.audio PRIVATE soxr)
+    aui_link(aui.audio PRIVATE soxr)
+endif ()
