diff --git a/aui.views/src/AUI/GL/gl.h b/aui.views/src/AUI/GL/gl.h
index 64209c6..062af7b 100644
--- a/aui.views/src/AUI/GL/gl.h
+++ b/aui.views/src/AUI/GL/gl.h
@@ -10,7 +10,9 @@
  */
 
 #pragma once
+#ifndef __cplusplus
 extern "C" {
+#endif
 #if AUI_PLATFORM_ANDROID
 #include <GLES3/gl3.h>
 #include <GLES3/gl31.h>
@@ -27,4 +29,6 @@ extern "C" {
 #include <GL/glew.h>
 #include <GL/gl.h>
 #endif
+#ifndef __cplusplus
 }
+#endif
