diff --git a/netcode/netcode.c b/netcode/netcode.c
index e975614..2327924 100755
--- a/netcode/netcode.c
+++ b/netcode/netcode.c
@@ -142,7 +142,9 @@ void netcode_default_free_function( void * context, void * pointer )
 
 #if NETCODE_PLATFORM == NETCODE_PLATFORM_WINDOWS
 
+    #ifndef NOMINMAX
     #define NOMINMAX
+    #endif
     #define _WINSOCK_DEPRECATED_NO_WARNINGS
     #include <winsock2.h>
     #include <ws2tcpip.h>
