diff --git a/lib/CMakeLists.txt b/lib/CMakeLists.txt
index f4be2be..039ff11 100644
--- a/lib/CMakeLists.txt
+++ b/lib/CMakeLists.txt
@@ -62,7 +62,6 @@ set(nghttp3_SOURCES
   nghttp3_unreachable.c
   nghttp3_settings.c
   nghttp3_callbacks.c
-  sfparse/sfparse.c
 )
 
 set(NGHTTP3_GENERATED_DIR "${CMAKE_CURRENT_BINARY_DIR}/generated")
@@ -87,6 +86,7 @@ install(
 # Public shared library
 if(ENABLE_SHARED_LIB)
   add_library(nghttp3 SHARED ${nghttp3_SOURCES})
+  target_link_libraries(nghttp3 sfparse)
   set_target_properties(nghttp3 PROPERTIES
     COMPILE_FLAGS "${WARNCFLAGS}"
     VERSION ${LT_VERSION} SOVERSION ${LT_SOVERSION}
@@ -105,6 +105,7 @@ endif()
 if(ENABLE_STATIC_LIB)
   # Public static library
   add_library(nghttp3_static STATIC ${nghttp3_SOURCES})
+  target_link_libraries(nghttp3_static sfparse)
   set_target_properties(nghttp3_static PROPERTIES
     COMPILE_FLAGS "${WARNCFLAGS}"
     VERSION ${LT_VERSION} SOVERSION ${LT_SOVERSION}
