diff --git a/configure.ac b/configure.ac
index b3e282786..7db7aa58e 100644
--- a/configure.ac
+++ b/configure.ac
@@ -4286,6 +4286,9 @@ AC_MSG_RESULT([yes])
 ], [
 ac_have_armv82_dotprod_feature=no
 AC_MSG_RESULT([no])
+], [
+ac_have_armv82_dotprod_feature=no
+AC_MSG_RESULT([no])
 ])
 CFLAGS="$save_CFLAGS"
 
@@ -4306,6 +4309,9 @@ AC_MSG_RESULT([yes])
 ], [
 ac_have_armv8_crypto=no
 AC_MSG_RESULT([no])
+], [
+ac_have_armv8_crypto=no
+AC_MSG_RESULT([no])
 ])
 CFLAGS="$save_CFLAGS"
 
