diff --git a/third_party/partition_alloc/src/partition_alloc/BUILD.gn b/third_party/partition_alloc/src/partition_alloc/BUILD.gn
index ba9ed4ea..759d0b8d 100644
--- a/third_party/partition_alloc/src/partition_alloc/BUILD.gn
+++ b/third_party/partition_alloc/src/partition_alloc/BUILD.gn
@@ -41,8 +41,13 @@ partition_alloc_dchecks_are_on =
 # Building PartitionAlloc for Windows component build.
 # Currently use build_with_chromium not to affect any third_party code,
 # but if any third_party code wants to use, remove build_with_chromium.
-use_partition_alloc_as_malloc_on_win_component_build =
-    build_with_chromium && is_win && is_component_build
+if (build_with_chromium && is_win && is_component_build) {
+    use_partition_alloc_as_malloc_on_win_component_build = true
+} else {
+    use_partition_alloc_as_malloc_on_win_component_build = false
+}
+#use_partition_alloc_as_malloc_on_win_component_build =
+#    build_with_chromium && is_win && is_component_build
 
 # TODO(crbug.com/40276913): Split PartitionAlloc into a public and
 # private parts. The public config would include add the "./include" dir and
