diff --git a/newton-4.00/CMakeLists.txt b/newton-4.00/CMakeLists.txt
index c7f63370f..466bc9f72 100644
--- a/newton-4.00/CMakeLists.txt
+++ b/newton-4.00/CMakeLists.txt
@@ -97,30 +97,6 @@ if(UNIX)
 endif ()
 
 if (MSVC)
-	if (PTR_SIZE EQUAL 8)
-		set(CMAKE_INSTALL_PREFIX "win64sdk" CACHE PATH "..." FORCE)
-	else()
-		set(CMAKE_INSTALL_PREFIX "win32sdk" CACHE PATH "..." FORCE)
-	endif ()
-
-	if(NEWTON_STATIC_RUNTIME_LIBRARIES)
-		message ("link to static runtime libraries")
-		set(CMAKE_C_FLAGS_DEBUG "/MTd")
-		set(CMAKE_C_FLAGS_RELWITHDEBINFO "/MT")
-		set(CMAKE_C_FLAGS_RELEASE "/MT")
-		set(CMAKE_CXX_FLAGS_DEBUG "/MTd")
-		set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "/MT")
-		set(CMAKE_CXX_FLAGS_RELEASE "/MT")
-	else()
-		message ("link to dynamic runtime libraries")
-		set(CMAKE_C_FLAGS_DEBUG "/MDd")
-		set(CMAKE_C_FLAGS_RELWITHDEBINFO "/MD")
-		set(CMAKE_C_FLAGS_RELEASE "/MD")
-		set(CMAKE_CXX_FLAGS_DEBUG "/MDd")
-		set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "/MD")
-		set(CMAKE_CXX_FLAGS_RELEASE "/MD")
-	endif()
-
 	if(${CMAKE_VS_PLATFORM_NAME} STREQUAL "x64")
 		message ("build 64 bit")
 	elseif(${CMAKE_VS_PLATFORM_NAME} STREQUAL "Win32")
