diff --git a/source/config/mh-mingw b/source/config/mh-mingw
index 30f6e5b..7cadf9d 100644
--- a/source/config/mh-mingw
+++ b/source/config/mh-mingw
@@ -15,11 +15,6 @@
 # If you prefer to have the DLLs in the lib folder, then set this to NO instead.
 MINGW_MOVEDLLSTOBINDIR = YES
 
-# We install sbin tools into the same bin directory because
-# pkgdata needs some of the tools in sbin, and we can't always depend on
-# icu-config working on Windows.
-sbindir=$(bindir)
-
 ## Commands to generate dependency files
 GEN_DEPS.c=	$(CC) -E -MM $(DEFS) $(CPPFLAGS)
 GEN_DEPS.cc=	$(CXX) -E -MM $(DEFS) $(CPPFLAGS)
@@ -81,6 +76,10 @@ else
 SO_TARGET_VERSION_SUFFIX = 
 endif
 
+## Remove shared library 's'
+STATIC_PREFIX_WHEN_USED = 
+STATIC_PREFIX = 
+
 # Static library prefix and file extension
 LIBSICU = $(LIBPREFIX)$(STATIC_PREFIX)$(ICUPREFIX)
 A = a
diff --git a/source/config/mh-mingw64 b/source/config/mh-mingw64
index fb64c56..40b436a 100644
--- a/source/config/mh-mingw64
+++ b/source/config/mh-mingw64
@@ -15,11 +15,6 @@ MINGW_MOVEDLLSTOBINDIR = YES
 # This file is similar to mh-mingw
 # Any changes made here may also need to be made in mh-mingw
 
-# We install sbin tools into the same bin directory because
-# pkgdata needs some of the tools in sbin, and we can't always depend on
-# icu-config working on Windows.
-sbindir=$(bindir)
-
 ## Commands to generate dependency files
 GEN_DEPS.c=	$(CC) -E -MM $(DEFS) $(CPPFLAGS)
 GEN_DEPS.cc=	$(CXX) -E -MM $(DEFS) $(CPPFLAGS)
@@ -81,6 +76,10 @@ else
 SO_TARGET_VERSION_SUFFIX = 
 endif
 
+## Remove shared library 's'
+STATIC_PREFIX_WHEN_USED = 
+STATIC_PREFIX = 
+
 # Static library prefix and file extension
 LIBSICU = lib$(LIBPREFIX)$(STATIC_PREFIX)$(ICUPREFIX)
 A = a
