diff --git a/CMakeLists.txt b/CMakeLists.txt
index ce1fdd2..7e6ce3e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -21,7 +21,8 @@
 # https://cmake.org/cmake/help/v3.12/module/CMakeDependentOption.html?highlight=cmakedependentoption
 # cmake 3.12 have more better modern c++ support
 
-cmake_minimum_required(VERSION 3.12  FATAL_ERROR)
+cmake_minimum_required(VERSION 3.15  FATAL_ERROR)
+cmake_policy(SET CMP0091 NEW)
 project(nana VERSION      1.7.3
         DESCRIPTION "C++ GUI library"
         HOMEPAGE_URL http://nanapro.org
