diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1656d5a..c207351 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -62,6 +62,9 @@ endif ()
 
 find_package(Boost 1.65 COMPONENTS ${Boost_COMPONENTS} REQUIRED)
 find_package(LIBZIP REQUIRED)
+if (WIN32)
+  find_package(ZLIB REQUIRED)
+endif ()
 
 
 # ==============================================================================
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index bf70918..a6474a6 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -112,7 +112,7 @@ target_link_libraries(fmi4cpp
 )
 
 if(WIN32)
-    target_link_libraries(fmi4cpp PRIVATE "Bcrypt")
+    target_link_libraries(fmi4cpp PRIVATE "Bcrypt" ZLIB::ZLIB)
 elseif(UNIX)
     target_link_libraries(fmi4cpp PRIVATE stdc++fs dl)
 endif()
