diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0aac377..94a44f4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -348,8 +348,8 @@ if(SDL2TTF_INSTALL)
         endif()
         # Only install a SDL2_ttf.pc file in Release mode
         install(CODE "
-        file(COPY_FILE \"${CMAKE_CURRENT_BINARY_DIR}/SDL2_ttf-$<CONFIG>.pc\"
-            \"${CMAKE_CURRENT_BINARY_DIR}/SDL2_ttf.pc\" ONLY_IF_DIFFERENT)
+        file(COPY \"${CMAKE_CURRENT_BINARY_DIR}/SDL2_ttf-$<CONFIG>.pc\"
+            DESTINATION \"${CMAKE_CURRENT_BINARY_DIR}/SDL2_ttf.pc\")
         file(INSTALL DESTINATION \"\${CMAKE_INSTALL_PREFIX}/${PC_DESTDIR}\"
             TYPE FILE
             FILES \"${CMAKE_CURRENT_BINARY_DIR}/SDL2_ttf.pc\")" CONFIG Release)
