diff --git a/src/asm/trampoline-aarch64.asm b/src/asm/trampoline-aarch64.asm
new file mode 100644
index 0000000..fab9ff3
--- /dev/null
+++ b/src/asm/trampoline-aarch64.asm
@@ -0,0 +1,42 @@
+;
+; The MIT License (MIT)
+;
+; Copyright © 2026 Franklin "Snaipe" Mathieu <http://snai.pe/>
+;
+; Permission is hereby granted, free of charge, to any person obtaining a copy
+; of this software and associated documentation files (the "Software"), to deal
+; in the Software without restriction, including without limitation the rights
+; to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+; copies of the Software, and to permit persons to whom the Software is
+; furnished to do so, subject to the following conditions:
+;
+; The above copyright notice and this permission notice shall be included in
+; all copies or substantial portions of the Software.
+;
+; THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+; IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+; FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+; AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+; LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+; OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+; THE SOFTWARE.
+;
+
+    AREA    |.text|, CODE, READONLY, ALIGN=3
+
+    EXPORT  bxfi_trampoline
+    EXPORT  bxfi_trampoline_addr
+    EXPORT  bxfi_trampoline_end
+
+bxfi_trampoline
+    ldr     x16, bxfi_trampoline_addr
+    br      x16
+
+    ALIGN   8
+
+bxfi_trampoline_addr
+    DCQ     0
+
+bxfi_trampoline_end
+
+    END
