diff --git a/meson.build b/meson.build
index 3c261d0ff..5f98981e9 100644
--- a/meson.build
+++ b/meson.build
@@ -192,6 +192,9 @@ test_deps = []
 internal_deps = []
 extra_link_args = []
 
+thread_dep = dependency('threads')
+deps += [thread_dep]
+
 if host_machine.endian() == 'big'
   conf.set('WORDS_BIGENDIAN', 1)
 endif
diff --git a/util/cairo-script/meson.build b/util/cairo-script/meson.build
index 653c19d33..9f90bc8b5 100644
--- a/util/cairo-script/meson.build
+++ b/util/cairo-script/meson.build
@@ -46,17 +46,20 @@ pkgmod.generate(libcairoscript,
 meson.override_dependency('cairo-script-interpreter', libcairoscript_dep)
 
 csi_replay_exe = executable('csi-replay', csi_replay_sources,
+  c_args: pthread_c_args,
   include_directories: [incbase],
   dependencies: deps + [libcairo_dep, libcairoscript_dep],
 )
 
 csi_exec_exe = executable('csi-exec', csi_exec_sources,
+  c_args: pthread_c_args,
   include_directories: [incbase],
   dependencies: deps + [libcairo_dep, libcairoscript_dep],
 )
 
 if feature_conf.get('CAIRO_HAS_SCRIPT_SURFACE', 0) == 1 and conf.get('HAVE_LIBGEN_H', 0) == 1
   csi_trace_exe = executable('csi-trace', csi_trace_sources,
+    c_args: pthread_c_args,
     include_directories: [incbase],
     dependencies: deps + [libcairo_dep, libcairoscript_dep],
   )
