diff --git a/CMakeLists.txt b/CMakeLists.txt
index 009d392..9ff5ee4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -18,6 +18,7 @@ project(
     LANGUAGES C CXX
 )
 
+include(FindPkgConfig)
 include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake OPTIONAL RESULT_VARIABLE CONAN_FILE)
 if(CONAN_FILE)
     conan_basic_setup(TARGETS)
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index a3d309e..814df3d 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -62,8 +62,10 @@ endif()
 
 set_property(CACHE LIBTCOD_LODEPNG PROPERTY STRINGS "vendored" "find_package" "disable")
 if(LIBTCOD_LODEPNG STREQUAL "find_package")
-    find_package(lodepng-c CONFIG REQUIRED)
-    target_link_libraries(${PROJECT_NAME} PRIVATE lodepng-c)
+    pkg_check_modules(lodepng REQUIRED lodepng)
+    target_link_directories(${PROJECT_NAME} PRIVATE ${lodepng_LIBRARY_DIRS})
+    target_link_libraries(${PROJECT_NAME} PRIVATE ${lodepng_LIBRARIES})
+    target_include_directories(${PROJECT_NAME} PRIVATE ${lodepng_INCLUDE_DIRS})
 elseif(LIBTCOD_LODEPNG STREQUAL "vendored")
     message("Will be vendored: LodePNG")
     target_sources(${PROJECT_NAME} PRIVATE "vendor/lodepng.c")
@@ -80,6 +82,8 @@ if(LIBTCOD_UTF8PROC STREQUAL "vendored")
     target_include_directories(${PROJECT_NAME} PRIVATE "vendor/utf8proc")
     target_sources(${PROJECT_NAME} PRIVATE "vendor/utf8proc/utf8proc.c")
 elseif (LIBTCOD_UTF8PROC STREQUAL "find_package")
+    pkg_check_modules(libutf8proc REQUIRED libutf8proc)
+    target_include_directories(${PROJECT_NAME} PRIVATE ${libutf8proc_INCLUDE_DIRS})
     find_package(utf8proc REQUIRED)
     target_link_libraries(${PROJECT_NAME} PRIVATE utf8proc)
 elseif (LIBTCOD_UTF8PROC STREQUAL "vcpkg")
@@ -96,8 +100,8 @@ if(LIBTCOD_STB STREQUAL "vendored")
     message("Will be vendored: stb")
     target_include_directories(${PROJECT_NAME} PRIVATE "vendor/")
 elseif(LIBTCOD_STB STREQUAL "find_package" OR LIBTCOD_STB STREQUAL "vcpkg")
-    find_package(Stb REQUIRED)
-    target_include_directories(${PROJECT_NAME} PRIVATE ${Stb_INCLUDE_DIR})
+    pkg_check_modules(stb REQUIRED stb)
+    target_include_directories(${PROJECT_NAME} PRIVATE ${stb_INCLUDE_DIRS})
 else()
     message(FATAL_ERROR "Unknown option for LIBTCOD_STB: '${LIBTCOD_STB}'")
 endif()
