diff --git a/src/archo/src/zipfile.cpp b/src/archo/src/zipfile.cpp
index 364c855b..c2a4bf65 100644
--- a/src/archo/src/zipfile.cpp
+++ b/src/archo/src/zipfile.cpp
@@ -32,0 +33 @@ constexpr uint64_t kEntryMax = 100000;
+constexpr unsigned int kZipDirectoryAttribute = 0x10;
@@ -190,4 +190,0 @@ bool ZipFile::extractCurrentFile(const std::filesystem::path& path, bool whiny)
-// Create directory
-#if !WIN32
-        const int FILE_ATTRIBUTE_DIRECTORY = 0x10;
-#endif
@@ -195 +192 @@ bool ZipFile::extractCurrentFile(const std::filesystem::path& path, bool whiny)
-        if (finfo.external_fa & FILE_ATTRIBUTE_DIRECTORY ||
+        if (finfo.external_fa & internal::kZipDirectoryAttribute ||
