diff --git a/lib_acl/CMakeLists.txt b/lib_acl/CMakeLists.txt
index 287d5f3..6a8b8b8 100644
--- a/lib_acl/CMakeLists.txt
+++ b/lib_acl/CMakeLists.txt
@@ -202,7 +202,7 @@ endif()
 
 link_directories(${acl_output_path})
 
-if(CMAKE_SYSTEM_NAME MATCHES "Linux" OR CMAKE_SYSTEM_NAME MATCHES "Darwin")
+if(CMAKE_SYSTEM_NAME MATCHES "Linux" OR CMAKE_SYSTEM_NAME MATCHES "Darwin" OR CMAKE_SYSTEM_NAME MATCHES "Android" OR CMAKE_SYSTEM_NAME MATCHES "FreeBSD")
 	install(TARGETS acl_static
 		EXPORT acl_static-targets
 		ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
@@ -253,12 +253,6 @@ if (ACL_BUILD_SHARED MATCHES "YES")
 		set_target_properties(acl_shared PROPERTIES OUTPUT_NAME "acl")
 	endif()
 
-	if(CMAKE_SYSTEM_NAME MATCHES "Linux" OR CMAKE_SYSTEM_NAME MATCHES "Darwin")
-		install(TARGETS acl_shared
-			EXPORT acl_shared-targets
-			LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
-		)
-	endif()
 endif()
 
 ##############################################################################
diff --git a/lib_acl_cpp/CMakeLists.txt b/lib_acl_cpp/CMakeLists.txt
index 08c558e..6f88470 100644
--- a/lib_acl_cpp/CMakeLists.txt
+++ b/lib_acl_cpp/CMakeLists.txt
@@ -276,7 +276,7 @@ endif()
 
 link_directories(${lib_output_path})
 
-if(CMAKE_SYSTEM_NAME MATCHES "Linux" OR CMAKE_SYSTEM_NAME MATCHES "Darwin")
+if(CMAKE_SYSTEM_NAME MATCHES "Linux" OR CMAKE_SYSTEM_NAME MATCHES "Darwin" OR CMAKE_SYSTEM_NAME MATCHES "Android" OR CMAKE_SYSTEM_NAME MATCHES "FreeBSD")
 	install(TARGETS acl_cpp_static
 		EXPORT acl_cpp_static-targets
 		ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
@@ -337,12 +337,6 @@ if (ACL_BUILD_SHARED MATCHES "YES")
 		target_link_libraries(acl_cpp_shared protocol acl ZLIB::ZLIB)
 	endif()
 
-	if(CMAKE_SYSTEM_NAME MATCHES "Linux" OR CMAKE_SYSTEM_NAME MATCHES "Darwin")
-		install(TARGETS acl_cpp_shared
-			EXPORT acl_cpp_shared-targets
-			LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
-		)
-	endif()
 endif()
 
 ##############################################################################
diff --git a/lib_fiber/c/CMakeLists.txt b/lib_fiber/c/CMakeLists.txt
index c784212..2929bee 100644
--- a/lib_fiber/c/CMakeLists.txt
+++ b/lib_fiber/c/CMakeLists.txt
@@ -163,7 +163,7 @@ endif()
 
 link_directories(${lib_output_path})
 
-if(CMAKE_SYSTEM_NAME MATCHES "Linux" OR CMAKE_SYSTEM_NAME MATCHES "Darwin")
+if(CMAKE_SYSTEM_NAME MATCHES "Linux" OR CMAKE_SYSTEM_NAME MATCHES "Darwin" OR CMAKE_SYSTEM_NAME MATCHES "Android" OR CMAKE_SYSTEM_NAME MATCHES "FreeBSD")
 	install(TARGETS fiber_static
 		EXPORT fiber_static-targets
 		ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
@@ -215,12 +215,6 @@ if (ACL_BUILD_SHARED MATCHES "YES")
 		set(CMAKE_SHARED_LINKER_FLAGS "${sys_ldflags}")
 	endif()
 
-	if(CMAKE_SYSTEM_NAME MATCHES "Linux" OR CMAKE_SYSTEM_NAME MATCHES "Darwin")
-		install(TARGETS fiber_shared
-			EXPORT fiber_shared-targets
-			LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
-		)
-	endif()
 endif()
 
 ##############################################################################
diff --git a/lib_fiber/cpp/CMakeLists.txt b/lib_fiber/cpp/CMakeLists.txt
index 4d70233..5c5851b 100644
--- a/lib_fiber/cpp/CMakeLists.txt
+++ b/lib_fiber/cpp/CMakeLists.txt
@@ -152,7 +152,7 @@ endif()
 
 link_directories(${lib_output_path})
 
-if(CMAKE_SYSTEM_NAME MATCHES "Linux" OR CMAKE_SYSTEM_NAME MATCHES "Darwin")
+if(CMAKE_SYSTEM_NAME MATCHES "Linux" OR CMAKE_SYSTEM_NAME MATCHES "Darwin" OR CMAKE_SYSTEM_NAME MATCHES "Android" OR CMAKE_SYSTEM_NAME MATCHES "FreeBSD")
 	install(TARGETS fiber_cpp_static
 		EXPORT fiber_cpp_static-targets
 		ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
@@ -210,12 +210,6 @@ if (ACL_BUILD_SHARED MATCHES "YES")
 		target_link_libraries(fiber_cpp_shared acl_cpp protocol acl fiber ZLIB::ZLIB)
 	endif()
 
-	if(CMAKE_SYSTEM_NAME MATCHES "Linux" OR CMAKE_SYSTEM_NAME MATCHES "Darwin")
-		install(TARGETS fiber_cpp_shared
-			EXPORT fiber_cpp_shared-targets
-			LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
-		)
-	endif()
 endif()
 
 ##############################################################################
diff --git a/lib_protocol/CMakeLists.txt b/lib_protocol/CMakeLists.txt
index b53ee88..9cd89c1 100644
--- a/lib_protocol/CMakeLists.txt
+++ b/lib_protocol/CMakeLists.txt
@@ -153,7 +153,7 @@ endif()
 
 link_directories(${lib_output_path})
 
-if(CMAKE_SYSTEM_NAME MATCHES "Linux" OR CMAKE_SYSTEM_NAME MATCHES "Darwin")
+if(CMAKE_SYSTEM_NAME MATCHES "Linux" OR CMAKE_SYSTEM_NAME MATCHES "Darwin" OR CMAKE_SYSTEM_NAME MATCHES "Android" OR CMAKE_SYSTEM_NAME MATCHES "FreeBSD")
 	install(TARGETS protocol_static
 		EXPORT protocol_static-targets
 		ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
@@ -211,12 +211,6 @@ if (ACL_BUILD_SHARED MATCHES "YES")
 		target_link_libraries(protocol_shared acl ZLIB::ZLIB)
 	endif()
 
-	if(CMAKE_SYSTEM_NAME MATCHES "Linux" OR CMAKE_SYSTEM_NAME MATCHES "Darwin")
-		install(TARGETS protocol_shared
-			EXPORT protocol_shared-targets
-			LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
-		)
-	endif()
 endif()
 
 ##############################################################################
