diff --git a/lib_acl_cpp/CMakeLists.txt b/lib_acl_cpp/CMakeLists.txt
index 844239c..bbc0e7c 100644
--- a/lib_acl_cpp/CMakeLists.txt
+++ b/lib_acl_cpp/CMakeLists.txt
@@ -301,7 +301,7 @@ if (ACL_BUILD_SHARED MATCHES "YES")
 		target_compile_options(acl_cpp_static PRIVATE
 				-fvisibility=hidden -fvisibility-inlines-hidden)
 	elseif (CMAKE_SYSTEM_NAME MATCHES "Darwin")
-		set(sys_ldflags "-shared -liconv -lpthread -ldl")
+		set(sys_ldflags "-shared -lpthread -ldl")
 		target_compile_options(acl_cpp_static PRIVATE
 			-fvisibility=hidden -fvisibility-inlines-hidden)
 		set(CMAKE_MACOSX_RPATH 1)
@@ -329,7 +329,7 @@ if (ACL_BUILD_SHARED MATCHES "YES")
 		set(lc_flags "-Wl,-rpath,${CMAKE_INSTALL_PREFIX}/lib -L${lib_output_path}")
 		set(CMAKE_SHARED_LINKER_FLAGS "${lc_flags} ${sys_ldflags}")
 		set_target_properties(acl_cpp_shared PROPERTIES OUTPUT_NAME "acl_cpp")
-		target_link_libraries(acl_cpp_shared protocol acl ZLIB::ZLIB)
+		target_link_libraries(acl_cpp_shared protocol acl ZLIB::ZLIB Iconv::Iconv)
 	endif()
 
 	if(CMAKE_SYSTEM_NAME MATCHES "Linux" OR CMAKE_SYSTEM_NAME MATCHES "Darwin" OR CMAKE_SYSTEM_NAME MATCHES "Android" OR CMAKE_SYSTEM_NAME MATCHES "FreeBSD")
diff --git a/lib_fiber/cpp/CMakeLists.txt b/lib_fiber/cpp/CMakeLists.txt
index 9aec6df..14df06e 100644
--- a/lib_fiber/cpp/CMakeLists.txt
+++ b/lib_fiber/cpp/CMakeLists.txt
@@ -173,7 +173,7 @@ if (ACL_BUILD_SHARED MATCHES "YES")
 		target_compile_options(fiber_cpp_static PRIVATE
 			-fvisibility=hidden -fvisibility-inlines-hidden)
 	elseif (CMAKE_SYSTEM_NAME MATCHES "Darwin")
-		set(sys_ldflags "-shared -liconv -lpthread -ldl")
+		set(sys_ldflags "-shared -lpthread -ldl")
 		target_compile_options(fiber_cpp_static PRIVATE
 			-fvisibility=hidden -fvisibility-inlines-hidden)
 		set(CMAKE_MACOSX_RPATH 1)
@@ -202,7 +202,7 @@ if (ACL_BUILD_SHARED MATCHES "YES")
 		set_target_properties(fiber_cpp_shared PROPERTIES OUTPUT_NAME "fiber_cpp")
 		set(lc_ldflags "-Wl,-rpath,${CMAKE_INSTALL_PREFIX}/lib -L${lib_output_path}")
 		set(CMAKE_SHARED_LINKER_FLAGS "${lc_ldflags} ${sys_ldflags}")
-		target_link_libraries(fiber_cpp_shared acl_cpp protocol acl fiber ZLIB::ZLIB)
+		target_link_libraries(fiber_cpp_shared acl_cpp protocol acl fiber ZLIB::ZLIB Iconv::Iconv)
 	endif()
 
 	if(CMAKE_SYSTEM_NAME MATCHES "Linux" OR CMAKE_SYSTEM_NAME MATCHES "Darwin" OR CMAKE_SYSTEM_NAME MATCHES "Android" OR CMAKE_SYSTEM_NAME MATCHES "FreeBSD")
