diff --git a/CMake/folly-deps.cmake b/CMake/folly-deps.cmake
--- a/CMake/folly-deps.cmake
+++ b/CMake/folly-deps.cmake
@@ -56,6 +56,9 @@ find_package(Glog CONFIG REQUIRED)
 set(FOLLY_HAVE_LIBGLOG ${GLOG_FOUND})
 list(APPEND FOLLY_LINK_LIBRARIES glog::glog)
 list(APPEND FOLLY_INCLUDE_DIRECTORIES ${GLOG_INCLUDE_DIR})
+if(MSVC)
+  add_compile_definitions(GLOG_NO_ABBREVIATED_SEVERITIES)
+endif()
 
 find_package(Gflags MODULE)
 set(FOLLY_HAVE_LIBGFLAGS ${LIBGFLAGS_FOUND})
