diff -Naurw CMakeLists.txt CMakeLists.txt
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -86,6 +86,7 @@
 add_library(backward ${libtype} backward.cpp)
 target_compile_definitions(backward PUBLIC ${BACKWARD_DEFINITIONS})
 target_include_directories(backward PUBLIC ${BACKWARD_INCLUDE_DIRS})
+target_link_libraries(backward PUBLIC -limagehlp -lmsvcr110 -ldbghelp)
 
 ###############################################################################
 # TESTS
