diff -u -r -N a/crypto/cpuid.c b/crypto/cpuid.c
--- a/crypto/cpuid.c
+++ b/crypto/cpuid.c
@@ -91,6 +91,9 @@
 #define OPENSSL_CPUID_SETUP
 typedef uint64_t IA32CAP;
 
+#if defined(__APPLE__) && defined(__MACH__) && defined(__GNUC__) && __GNUC__ >= 2
+void OPENSSL_cpuid_setup(void) __attribute__((constructor));
+#endif
 void OPENSSL_cpuid_setup(void)
 {
     static int trigger = 0;
diff -u -r -N a/crypto/x86_64cpuid.pl b/crypto/x86_64cpuid.pl
--- a/crypto/x86_64cpuid.pl
+++ b/crypto/x86_64cpuid.pl
@@ -28,11 +28,18 @@
 
 print<<___;
 #include crypto/cryptlib.h
+___
+
+if ($flavour ne "macosx") {
+    print<<___;
 .extern		OPENSSL_cpuid_setup
 .hidden		OPENSSL_cpuid_setup
 .section	.init
 	call	OPENSSL_cpuid_setup
+___
+}
 
+print<<___;
 .hidden	OPENSSL_ia32cap_P
 .comm	OPENSSL_ia32cap_P,40,4	# <--Should match with internal/cryptlib.h OPENSSL_IA32CAP_P_MAX_INDEXES
 .text
