diff --git a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -76,7 +76,7 @@
     include/coro/concepts/promise.hpp
     include/coro/concepts/range_of.hpp
 
-    include/coro/detail/task_self_deleting.hpp / src/detail/task_self_deleting.cpp
+    include/coro/detail/task_self_deleting.hpp src/detail/task_self_deleting.cpp
     include/coro/detail/void_value.hpp
 
     include/coro/attribute.hpp
