diff --git a/PseudoDouble.h b/PseudoDouble.h
index 631ec43..bdd91f9 100644
--- a/PseudoDouble.h
+++ b/PseudoDouble.h
@@ -32,6 +32,7 @@
 #define PSEUDO_DOUBLE_CPP_H
 
 #include <stdexcept>
+#include <string>
 
 #ifndef PD_DO_ERROR_OVERFLOW
 #define PD_DO_ERROR_OVERFLOW throw std::overflow_error("overflow");
diff --git a/pseudo_double.h b/pseudo_double.h
index f1a3522..c5c80a2 100644
--- a/pseudo_double.h
+++ b/pseudo_double.h
@@ -31,6 +31,7 @@
 #define PSEUDO_DOUBLE_H
 
 #include <stdint.h>
+#include <stdbool.h>
 
 #ifdef _MSC_VER
 #include <intrin.h>
