diff --git a/deps/marisa-0.2.6/CMakeLists.txt b/deps/marisa-0.2.6/CMakeLists.txt
index 305f273..1472532 100644
--- a/deps/marisa-0.2.6/CMakeLists.txt
+++ b/deps/marisa-0.2.6/CMakeLists.txt
@@ -17,3 +17,10 @@ set(
 add_library(marisa STATIC ${LIBMARISA_SOURCES})
 set_target_properties(marisa PROPERTIES POSITION_INDEPENDENT_CODE ON)
 source_group(marisa FILES ${LIBMARISA_SOURCES})
+
+if (NOT BUILD_SHARED_LIBS)
+  install(TARGETS marisa
+    RUNTIME DESTINATION bin
+    LIBRARY DESTINATION lib
+    ARCHIVE DESTINATION lib)
+endif()
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 8ec1177..2bc2900 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -104,7 +104,11 @@ endif()
 add_library(libopencc ${LIBOPENCC_SOURCES} ${LIBOPENCC_HEADERS})
 set_target_properties(libopencc PROPERTIES POSITION_INDEPENDENT_CODE ON)
 source_group(libopencc FILES ${LIBOPENCC_SOURCES} ${LIBOPENCC_HEADERS})
-target_link_libraries(libopencc marisa)
+if (NOT BUILD_SHARED_LIBS)
+  target_link_libraries(libopencc PUBLIC marisa)
+else()
+  target_link_libraries(libopencc marisa)
+endif()
 
 GENERATE_EXPORT_HEADER(
   libopencc
