diff --git a/Common/Core/SMP/Common/vtkSMPToolsImpl.h b/Common/Core/SMP/Common/vtkSMPToolsImpl.h
index f336f1fe2d2..8ed35c982d5 100644
--- a/Common/Core/SMP/Common/vtkSMPToolsImpl.h
+++ b/Common/Core/SMP/Common/vtkSMPToolsImpl.h
@@ -62,7 +62,7 @@ public:
   bool GetNestedParallelism() { return this->NestedActivated; }
 
   //--------------------------------------------------------------------------------
-  bool IsParallelScope() { return this->IsParallel; }
+  bool IsParallelScope();
 
   //--------------------------------------------------------------------------------
   bool GetSingleThread();
@@ -118,6 +118,12 @@ private:
   std::atomic<bool> IsParallel{ false };
 };
 
+template <BackendType Backend>
+bool vtkSMPToolsImpl<Backend>::IsParallelScope()
+{
+  return this->IsParallel;
+}
+
 using ExecuteFunctorPtrType = void (*)(void*, vtkIdType, vtkIdType, vtkIdType);
 
 VTK_ABI_NAMESPACE_END
diff --git a/Common/Core/SMP/Common/vtkSMPToolsImpl.h b/Common/Core/SMP/Common/vtkSMPToolsImpl.h
index 8ed35c982d5..d2a4a377adc 100644
--- a/Common/Core/SMP/Common/vtkSMPToolsImpl.h
+++ b/Common/Core/SMP/Common/vtkSMPToolsImpl.h
@@ -43,7 +43,7 @@ const BackendType DefaultBackend = BackendType::OpenMP;
 #endif
 
 template <BackendType Backend>
-class VTKCOMMONCORE_EXPORT vtkSMPToolsImpl
+class vtkSMPToolsImpl
 {
 public:
   //--------------------------------------------------------------------------------
diff --git a/Utilities/octree/octree/octree_node.txx b/Utilities/octree/octree/octree_node.txx
index 6971c7c1199..5f12bb7de76 100644
--- a/Utilities/octree/octree/octree_node.txx
+++ b/Utilities/octree/octree/octree_node.txx
@@ -210,7 +210,7 @@ const octree_node<T_, d_, A_>& octree_node<T_, d_, A_>::operator[](int child) co
   {
     throw std::domain_error("Attempt to access children of an octree leaf node.");
   }
-  return this->_M_chilren[child];
+  return this->m_children[child];
 }
 
 /**\brief Return a reference to a child node.
