diff --git a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -115,9 +115,9 @@
         PRIVATE $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/third_party/mman-win32>)
 endif (WIN32)
 
-add_subdirectory(trantor)
+find_package(Trantor CONFIG REQUIRED)
 
-target_link_libraries(${PROJECT_NAME} PUBLIC trantor)
+target_link_libraries(${PROJECT_NAME} PUBLIC Trantor::Trantor)
 
 if (NOT WIN32)
     if (NOT ${CMAKE_SYSTEM_NAME} STREQUAL "OpenBSD")
