diff --git a/CMakeLists.txt b/CMakeLists.txt
index b8913c2d..97bb7d2d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -14,6 +14,11 @@ list(APPEND CMAKE_MODULE_PATH "${PROJECT_SOURCE_DIR}/cmake/")
 
 add_definitions (-DCMAKE -DVERSION=\"${VERSION}\")
 
+if(CMAKE_SYSTEM_NAME STREQUAL "Windows")
+	add_definitions("-DWIN32")
+	add_definitions("-D_WIN32")
+endif()
+
 if (WIN32)
 	add_definitions("-D_CRT_SECURE_NO_WARNINGS")
 	add_definitions("-D_CRT_NONSTDC_NO_DEPRECATE")
--