diff --git a/Modules/ThirdParty/MINC/src/libminc/libcommon/ParseArgv.c b/Modules/ThirdParty/MINC/src/libminc/libcommon/ParseArgv.c
index 260f574..15d80dd 100644
--- a/Modules/ThirdParty/MINC/src/libminc/libcommon/ParseArgv.c
+++ b/Modules/ThirdParty/MINC/src/libminc/libcommon/ParseArgv.c
@@ -315,7 +315,7 @@ ParseArgv(argcPtr, argv, argTable, flags)
          }
          break;
       case ARGV_FUNC: {
-         int (*handlerProc)() =  (int (*)())(uintptr_t)infoPtr->src;
+         int (*handlerProc)(void *, const char *, char *) = (int (*)(void *, const char *, char *))(uintptr_t)infoPtr->src;
 		
          if ((*handlerProc)(infoPtr->dst, infoPtr->key,
                             argv[srcIndex])) {
@@ -325,7 +325,7 @@ ParseArgv(argcPtr, argv, argTable, flags)
          break;
       }
       case ARGV_GENFUNC: {
-         int (*handlerProc)() = (int (*)())(uintptr_t)infoPtr->src;
+         int (*handlerProc)(void *, const char *, int, char **) = (int (*)(void *, const char *, int, char **))(uintptr_t)infoPtr->src;
 
          argc = (*handlerProc)(infoPtr->dst, infoPtr->key,
                                argc, argv+srcIndex);
