diff --git a/src/plugins/cairo/libdecor-cairo.c b/src/plugins/cairo/libdecor-cairo.c
index 5951e93..c1f7f40 100644
--- a/src/plugins/cairo/libdecor-cairo.c
+++ b/src/plugins/cairo/libdecor-cairo.c
@@ -42,7 +42,7 @@
 #include "cursor-settings.h"
 #include "os-compatibility.h"
 
-#include <cairo/cairo.h>
+#include <cairo.h>
 #include <pango/pangocairo.h>
 
 #include "common/libdecor-cairo-blur.h"
diff --git a/src/plugins/common/libdecor-cairo-blur.h b/src/plugins/common/libdecor-cairo-blur.h
index d48e9dd..25ecfef 100644
--- a/src/plugins/common/libdecor-cairo-blur.h
+++ b/src/plugins/common/libdecor-cairo-blur.h
@@ -1,6 +1,6 @@
 #pragma once
 
-#include <cairo/cairo.h>
+#include <cairo.h>
 
 int
 blur_surface(cairo_surface_t *surface, int margin);
diff --git a/src/plugins/gtk/libdecor-gtk.c b/src/plugins/gtk/libdecor-gtk.c
index bc8290f..d1852e7 100644
--- a/src/plugins/gtk/libdecor-gtk.c
+++ b/src/plugins/gtk/libdecor-gtk.c
@@ -41,7 +41,7 @@
 #include "cursor-settings.h"
 #include "os-compatibility.h"
 
-#include <cairo/cairo.h>
+#include <cairo.h>
 
 #include "common/libdecor-cairo-blur.h"
 #include <poll.h>
