diff --git a/cmake_admin/FindFLAC.cmake b/cmake_admin/FindFLAC.cmake
index 2de2f41..2aa09b6 100644
--- a/cmake_admin/FindFLAC.cmake
+++ b/cmake_admin/FindFLAC.cmake
@@ -61,7 +61,9 @@ else()
   if(NOT TARGET "Ogg::ogg")
     find_package(Ogg QUIET)
   endif()
+  if(HAS_LIBM)
   set(_flac_link_libraries "Ogg::ogg" ${MATH_LIBRARY})
+  endif(HAS_LIBM)
 endif()
 
 if(PC_FLAC++_FOUND)
diff --git a/cmake_admin/FindGLib2.cmake b/cmake_admin/FindGLib2.cmake
index f6fa6f1..cb910f6 100644
--- a/cmake_admin/FindGLib2.cmake
+++ b/cmake_admin/FindGLib2.cmake
@@ -64,6 +64,14 @@ find_library(
   NAMES "gobject-2.0"
   HINTS "${PC_GOBJECT2_LIBDIR}")
 
+find_library(
+  Intl_LIBRARY
+  NAMES "intl")
+
+find_library(
+  Pcre_LIBRARY
+  NAMES "pcre")
+
 # GLib stores its config in lib/glib-2.0/include
 get_filename_component(_glib2_libdir "${GLib2_glib-2_LIBRARY}" PATH)
 find_path(
@@ -102,52 +110,28 @@ find_package_handle_standard_args(
                 "GLib2_INCLUDE_DIRS"
   VERSION_VAR "GLib2_VERSION")
 
+get_filename_component(_intl_path "${Intl_LIBRARY}" DIRECTORY)
+
 # Create the targets
 if(GLib2_glib-2_LIBRARY AND NOT TARGET GLib2::glib-2)
   # Handle transitive dependencies
   if(PC_GLIB2_FOUND)
     get_target_properties_from_pkg_config("${GLib2_glib-2_LIBRARY}" "PC_GLIB2"
                                           "_glib2")
+    list(APPEND _glib2_link_directories "${_intl_path}")
   else()
     find_package(Intl QUIET)
     find_package(Iconv QUIET)
     list(APPEND _glib2_link_libraries "Intl::Intl" "Iconv::Iconv")
-    if(WIN32)
-      list(APPEND _glib2_link_libraries "ws2_32" "winmm")
-    else()
-      list(APPEND _glib2_link_libraries "Threads::Threads")
-    endif()
     list(APPEND _glib2_link_libraries ${MATH_LIBRARY})
-
-    # Glib can link to either PCRE 1 or 2
-    find_library(
-      _pcre2_8bit_library
-      NAMES "pcre2-8"
-      HINTS "${PC_GLIB2_LIBDIR}")
-    if(_pcre2_8bit_library)
-      include(CheckCSourceCompiles)
-      set(_backup_includes ${CMAKE_REQUIRED_INCLUDES})
-      set(_backup_libraries ${CMAKE_REQUIRED_LIBRARIES})
-      set(_backup_libdir ${CMAKE_REQUIRED_LIBRARIES})
-      set(CMAKE_REQUIRED_INCLUDES "${GLib2_INCLUDE_DIRS}")
-      set(CMAKE_REQUIRED_LIBRARIES
-          "${GLib2_glib-2_LIBRARY}" "${_glib2_link_libraries}"
-          "${_pcre2_8bit_library}")
-      check_c_source_compiles(
-        "#include <glib.h>
-    int main(){
-      g_regex_error_quark();
-    }"
-        GLIB2_USES_PCRE2)
-      set(CMAKE_REQUIRED_INCLUDES ${_backup_includes})
-      set(CMAKE_REQUIRED_LIBRARIES ${_backup_libraries})
-    endif()
-    if(GLIB2_USES_PCRE2)
-      list(APPEND _glib2_link_libraries "${_pcre2_8bit_library}")
-    else()
-      list(APPEND _glib2_link_libraries "pcre")
-    endif()
   endif()
+  if(WIN32)
+    list(APPEND _glib2_link_libraries "ws2_32" "winmm")
+  else()
+    list(APPEND _glib2_link_libraries "Threads::Threads")
+  endif()
+  get_filename_component(_pcre_path "${Pcre_LIBRARY}" DIRECTORY)
+  list(APPEND _glib2_link_directories "${_pcre_path}")
 
   # pkg_check_modules consider these as LDFLAGS_OTHER rather instead of
   # libraries
@@ -172,6 +156,7 @@ if(GLib2_gthread-2_LIBRARY AND NOT TARGET GLib2::gthread-2)
   if(PC_GTHREAD2_FOUND)
     get_target_properties_from_pkg_config("${GLib2_gthread-2_LIBRARY}"
                                           "PC_GTHREAD2" "_gthread2")
+    list(APPEND _glib2_link_directories "${_intl_path}")
   else()
     set(_gthread2_link_libraries "Threads::Threads" "GLib2::glib-2")
   endif()
@@ -191,6 +176,7 @@ if(GLib2_gmodule-2_LIBRARY AND NOT TARGET GLib2::gmodule-2)
   if(PC_GMODULE2_FOUND)
     get_target_properties_from_pkg_config("${GLib2_gmodule-2_LIBRARY}"
                                           "PC_GMODULE2" "_gmodule2")
+    list(APPEND _glib2_link_directories "${_intl_path}")
   else()
     set(_gmodule2_link_libraries "GLib2::glib-2")
   endif()
@@ -210,6 +196,7 @@ if(GLib2_gobject-2_LIBRARY AND NOT TARGET GLib2::gobject-2)
   if(PC_GOBJECT2_FOUND)
     get_target_properties_from_pkg_config("${GLib2_gobject-2_LIBRARY}"
                                           "PC_OBJECT2" "_gobject2")
+    list(APPEND _glib2_link_directories "${_intl_path}")
   else()
     find_package(libffi QUIET)
     set(_gobject2_link_libraries "libffi" "GLib2::glib-2")
diff --git a/cmake_admin/FindVorbis.cmake b/cmake_admin/FindVorbis.cmake
index 0865c3e..b4d9786 100644
--- a/cmake_admin/FindVorbis.cmake
+++ b/cmake_admin/FindVorbis.cmake
@@ -67,7 +67,9 @@ else()
   if(NOT TARGET Ogg::ogg)
     find_package(Ogg QUIET)
   endif()
+  if(HAS_LIBM)
   set(_vorbis_link_libraries "Ogg::ogg" ${MATH_LIBRARY})
+  endif(HAS_LIBM)
 endif()
 
 # Extract additional flags if pkg-config is available
