diff --git a/folly/Benchmark.cpp b/folly/Benchmark.cpp
--- a/folly/Benchmark.cpp
+++ b/folly/Benchmark.cpp
@@ -17,6 +17,7 @@
 // @author Andrei Alexandrescu (andrei.alexandrescu@fb.com)
 
 #include <folly/Benchmark.h>
+#include <folly/portability/Windows.h>
 
 #include <algorithm>
 #include <cmath>
@@ -29,19 +30,14 @@
 #include <utility>
 #include <vector>
 
+#include <boost/regex.hpp>
+
 #include <folly/FileUtil.h>
 #include <folly/MapUtil.h>
 #include <folly/String.h>
 #include <folly/container/Foreach.h>
 #include <folly/json.h>
 
-// This needs to be at the end because some versions end up including
-// Windows.h without defining NOMINMAX, which breaks uses
-// of `std::numeric_limits<T>::max()`. We explicitly define NOMINMAX here
-// explicitly instead.
-#define NOMINMAX 1
-#include <boost/regex.hpp>
-
 using namespace std;
 
 FOLLY_GFLAGS_DEFINE_bool(benchmark, false, "Run benchmarks.");
diff --git a/folly/experimental/io/HugePages.cpp b/folly/experimental/io/HugePages.cpp
--- a/folly/experimental/io/HugePages.cpp
+++ b/folly/experimental/io/HugePages.cpp
@@ -27,6 +27,7 @@
 #include <stdexcept>
 #include <system_error>
 
+#include <folly/portability/Windows.h>
 #include <boost/regex.hpp>
 
 #include <folly/Conv.h>
diff --git a/folly/experimental/TestUtil.cpp b/folly/experimental/TestUtil.cpp
--- a/folly/experimental/TestUtil.cpp
+++ b/folly/experimental/TestUtil.cpp
@@ -14,11 +14,14 @@
  * limitations under the License.
  */
 
+#include <folly/portability/Windows.h>
+
 #include <folly/experimental/TestUtil.h>
 
 #include <sys/stat.h>
 #include <sys/types.h>
 
+#include <boost/regex.hpp>
 #include <glog/logging.h>
 
 #include <folly/Exception.h>
@@ -33,11 +36,6 @@
 #include <crtdbg.h> // @manual
 #endif
 
-// This needs to be at the end because some versions end up including
-// Windows.h without defining NOMINMAX, which breaks uses of std::numeric_limits
-// in various headers that we include.
-#include <boost/regex.hpp>
-
 namespace folly {
 namespace test {
 
