--- configure	2023-11-11 00:36:22.000000000 +0100
+++ configure_mingw	2024-02-23 20:20:11.436767000 +0100
@@ -5686,9 +5686,9 @@
         # however ld then forgets what the entry point should be (oops) so we
         # have to manually (re)set it.
         if enabled x86_32; then
-            disabled debug && add_ldexeflags -Wl,--pic-executable,-e,_mainCRTStartup
+            disabled debug && add_ldexeflags -Wl,-e,_mainCRTStartup
         elif enabled x86_64; then
-            disabled debug && add_ldexeflags -Wl,--pic-executable,-e,mainCRTStartup
+            disabled debug && add_ldexeflags -Wl,-e,mainCRTStartup
             check_ldflags -Wl,--high-entropy-va # binutils 2.25
             # Set image base >4GB for extra entropy with HEASLR
             add_ldexeflags -Wl,--image-base,0x140000000
