diff --git a/angelscript/source/as_callfunc_arm64_msvc.asm b/angelscript/source/as_callfunc_arm64_msvc.asm
index ce43c75..1059246 100644
--- a/angelscript/source/as_callfunc_arm64_msvc.asm
+++ b/angelscript/source/as_callfunc_arm64_msvc.asm
@@ -68,7 +68,7 @@ GetHFAReturnDouble PROC
     ALIGN   4
 GetHFAReturnFloat PROC
     adr     x9, |populateFloats|
-    sub     x9, x9, x2 // x9 -= returnSize; (already 4 bytes per return)
+    sub     x9, x9, x2 ; x9 -= returnSize; (already 4 bytes per return)
     br      x9
 
     str     s3, [x1, #0x4]
@@ -185,6 +185,7 @@ CallARM64Ret128 PROC
     ldp     fp, lr, [sp],#0x20
 
     ret ; CallARM64Ret128
+    ENDP
 
     ALIGN   4
 CallARM64RetInMemory PROC
@@ -202,4 +203,4 @@ CallARM64RetInMemory PROC
 
     ret ; CallARM64RetInMemory
 
-    END
+    ENDP
