diff --git a/include/webdriverxx/capabilities.h b/include/webdriverxx/capabilities.h
index ca5cb64..052ad27 100644
--- a/include/webdriverxx/capabilities.h
+++ b/include/webdriverxx/capabilities.h
@@ -2,7 +2,7 @@
 #define WEBDRIVERXX_CAPABILITIES_H
 
 #include "conversions.h"
-#include "picojson.h"
+#include <picojson.h>
 #include <string>
 
 namespace webdriverxx {
diff --git a/include/webdriverxx/client.h b/include/webdriverxx/client.h
index 5548438..d4b6a3e 100644
--- a/include/webdriverxx/client.h
+++ b/include/webdriverxx/client.h
@@ -5,7 +5,7 @@
 #include "capabilities.h"
 #include "detail/resource.h"
 #include "detail/http_connection.h"
-#include "picojson.h"
+#include <picojson.h>
 #include <string>
 #include <vector>
 
diff --git a/include/webdriverxx/conversions.h b/include/webdriverxx/conversions.h
index 8374073..510b5a0 100644
--- a/include/webdriverxx/conversions.h
+++ b/include/webdriverxx/conversions.h
@@ -4,7 +4,7 @@
 #include "types.h"
 #include "detail/error_handling.h"
 #include "detail/meta_tools.h"
-#include "picojson.h"
+#include <picojson.h>
 #include <algorithm>
 
 namespace webdriverxx {
diff --git a/include/webdriverxx/detail/resource.h b/include/webdriverxx/detail/resource.h
index 26e3ed6..185c5e0 100644
--- a/include/webdriverxx/detail/resource.h
+++ b/include/webdriverxx/detail/resource.h
@@ -6,7 +6,7 @@
 #include "shared.h"
 #include "../conversions.h"
 #include "../response_status_code.h"
-#include "../picojson.h"
+#include <picojson.h>
 
 namespace webdriverxx {
 namespace detail {
diff --git a/include/webdriverxx/detail/types.h b/include/webdriverxx/detail/types.h
index f218900..fcbe4e9 100644
--- a/include/webdriverxx/detail/types.h
+++ b/include/webdriverxx/detail/types.h
@@ -3,7 +3,7 @@
 
 #include "../conversions.h"
 #include "../capabilities.h"
-#include "../picojson.h"
+#include <picojson.h>
 #include <string>
 
 #define CXX17_2X ((defined(_MSVC_LANG) && _MSVC_LANG >= 201703L) || __cplusplus >= 201703L)
diff --git a/include/webdriverxx/js_args.h b/include/webdriverxx/js_args.h
index 0944f28..6e90c94 100644
--- a/include/webdriverxx/js_args.h
+++ b/include/webdriverxx/js_args.h
@@ -2,7 +2,7 @@
 #define WEBDRIVERXX_JS_ARGS_H
 
 #include "conversions.h"
-#include "picojson.h"
+#include <picojson.h>
 
 namespace webdriverxx {
 
diff --git a/include/webdriverxx/session.h b/include/webdriverxx/session.h
index ee5410d..24f4cd5 100644
--- a/include/webdriverxx/session.h
+++ b/include/webdriverxx/session.h
@@ -12,7 +12,7 @@
 #include "detail/keyboard.h"
 #include "detail/shared.h"
 #include "detail/factories_impl.h"
-#include "picojson.h"
+#include <picojson.h>
 #include "utils/picobase64.h"
 #include <string>
 #include <fstream>
