diff --git a/src/tools/perf/perftest.c b/src/tools/perf/perftest.c
index a90c998..a099519 100644
--- a/src/tools/perf/perftest.c
+++ b/src/tools/perf/perftest.c
@@ -243,7 +243,11 @@ static void sock_rte_barrier(void *rte_group, void (*progress)(void *arg),
 {
 #if _OPENMP
 #  pragma omp barrier
+#  if _OPENMP >= 202011
+#    pragma omp masked
+#  else
 #  pragma omp master
+#  endif
 #endif
   {
     sock_rte_group_t *group = rte_group;
@@ -577,7 +581,11 @@ static void mpi_rte_barrier(void *rte_group, void (*progress)(void *arg),
 
 #pragma omp barrier
 
+#if _OPENMP >= 202011
+#pragma omp masked
+#else
 #pragma omp master
+#endif
   {
     /*
     * Naive non-blocking barrier implementation over send/recv, to call user
diff --git a/src/ucm/event/event.c b/src/ucm/event/event.c
index 3502c64..368972d 100644
--- a/src/ucm/event/event.c
+++ b/src/ucm/event/event.c
@@ -521,7 +521,9 @@ void ucm_event_handler_remove(ucm_event_handler_t *handler)
 
 static ucs_status_t ucm_event_install(int events)
 {
+#ifdef UCX_SHARED_LIB
     UCS_MODULE_FRAMEWORK_DECLARE(ucm);
+#endif
     ucm_event_installer_t *event_installer;
     int malloc_events;
     ucs_status_t status;
@@ -547,8 +549,10 @@ static ucs_status_t ucm_event_install(int events)
 
     ucm_debug("malloc hooks are ready");
 
+#ifdef UCX_SHARED_LIB
     /* Call extra event installers */
     UCS_MODULE_FRAMEWORK_LOAD(ucm, UCS_MODULE_LOAD_FLAG_NODELETE);
+#endif
     ucs_list_for_each(event_installer, &ucm_event_installer_list, list) {
         status = event_installer->install(events);
         if (status != UCS_OK) {
