diff --git a/src/tools/perf/perftest.c b/src/tools/perf/perftest.c
index a90c998..a099519 100644
--- a/src/tools/perf/perftest.c
+++ b/src/tools/perf/perftest.c
@@ -243,7 +243,11 @@ static void sock_rte_barrier(void *rte_group, void (*progress)(void *arg),
 {
 #if _OPENMP
 #  pragma omp barrier
+#  if _OPENMP >= 202011
+#    pragma omp masked
+#  else
 #  pragma omp master
+#  endif
 #endif
   {
     sock_rte_group_t *group = rte_group;
@@ -577,7 +581,11 @@ static void mpi_rte_barrier(void *rte_group, void (*progress)(void *arg),
 
 #pragma omp barrier
 
+#if _OPENMP >= 202011
+#pragma omp masked
+#else
 #pragma omp master
+#endif
   {
     /*
      * Naive non-blocking barrier implementation over send/recv, to call user
diff --git a/src/tools/profile/Makefile.am b/src/tools/profile/Makefile.am
index a0952c2..2dd667c 100644
--- a/src/tools/profile/Makefile.am
+++ b/src/tools/profile/Makefile.am
@@ -10,3 +10,4 @@ ucx_read_profile_CFLAGS   = $(BASE_CFLAGS)
 ucx_read_profile_SOURCES  = read_profile.c
 ucx_read_profile_LDADD    = \
+    -Wl,--undefined=ucs_init \
     $(abs_top_builddir)/src/ucs/libucs.la
diff --git a/src/tools/profile/Makefile.in b/src/tools/profile/Makefile.in
index b644463..f7edce1 100644
--- a/src/tools/profile/Makefile.in
+++ b/src/tools/profile/Makefile.in
@@ -479,5 +479,6 @@ ucx_read_profile_CFLAGS = $(BASE_CFLAGS)
 ucx_read_profile_SOURCES = read_profile.c
 ucx_read_profile_LDADD = \
+    -Wl,--undefined=ucs_init \
     $(abs_top_builddir)/src/ucs/libucs.la
 
 all: all-am
diff --git a/Makefile.am b/Makefile.am
--- a/Makefile.am
+++ b/Makefile.am
@@ -53 +52,0 @@ SUBDIRS += \
-	test/apps \
diff --git a/Makefile.in b/Makefile.in
--- a/Makefile.in
+++ b/Makefile.in
@@ -617,7 +617,7 @@ top_top_srcdir = @top_top_srcdir@
 @DOCS_ONLY_FALSE@SUBDIRS = src/ucm src/ucs src/uct src/ucp \
 @DOCS_ONLY_FALSE@	src/tools/vfs src/tools/info src/tools/perf \
 @DOCS_ONLY_FALSE@	src/tools/profile bindings/go bindings/java \
-@DOCS_ONLY_FALSE@	test/apps examples cmake config \
+@DOCS_ONLY_FALSE@	cmake config \
 @DOCS_ONLY_FALSE@	$(am__append_1) $(am__append_2)
 @HAVE_LCOV_TRUE@LCOV_DIR = lcov_data
 @HAVE_LCOV_TRUE@LCOV_INFO = lcov.info
