diff --git a/src/npz.cpp b/src/npz.cpp
index 51a25e0..50e0d37 100644
--- a/src/npz.cpp
+++ b/src/npz.cpp
@@ -459,7 +459,7 @@ void npzstringwriter::write_file(const std::string &filename,
 
 void npzstringwriter::close() {
   if (!m_closed) {
-    ::close(m_output, m_entries);
+    (::close)(m_output, m_entries);
     m_closed = true;
   }
 }
@@ -501,7 +501,7 @@ void npzfilewriter::write_file(const std::string &filename,
 
 void npzfilewriter::close() {
   if (!m_closed) {
-    ::close(m_output, m_entries);
+    (::close)(m_output, m_entries);
     m_closed = true;
     m_output.close();
   }
